// @frontify/fondue/dist/components/MultiInput/MultiInput.es.js
import e, { Children as n } from "react";
var d = /* @__PURE__ */ ((a) => (a.Columns = "Columns", a.Spider = "Spider", a))(d || {});
const m = ({
  layout: a,
  spanLastItem: r,
  children: c,
  "data-test-id": l = "multi-input"
}) => {
  const t = n.toArray(c);
  return /* @__PURE__ */ e.createElement("div", { "data-test-id": l, className: "tw-flex tw-gap-2 tw-w-full tw-flex-col" }, a === "Columns" && /* @__PURE__ */ e.createElement("div", { "data-test-id": `${l}-grid-columns`, className: "tw-grid tw-grid-cols-2 tw-gap-2" }, t.map((i, s) => /* @__PURE__ */ e.createElement(
    "div",
    {
      className: r && s === t.length - 1 ? "tw-col-span-2" : "",
      key: s
    },
    i
  ))), a === "Spider" && /* @__PURE__ */ e.createElement("div", { "data-test-id": `${l}-grid-spider`, className: "tw-grid tw-grid-cols-4 tw-gap-2" }, t[0] && /* @__PURE__ */ e.createElement("div", { className: "tw-col-start-2 tw-col-span-2" }, t[0]), t[1] && /* @__PURE__ */ e.createElement("div", { className: "tw-col-start-1 tw-col-span-2" }, t[1]), t[2] && /* @__PURE__ */ e.createElement("div", { className: "tw-col-start-3 tw-col-span-2" }, t[2]), t[3] && /* @__PURE__ */ e.createElement("div", { className: "tw-col-start-2 tw-col-span-2" }, t[3])));
};
m.displayName = "FondueMultiInput";
export {
  m as MultiInput,
  d as MultiInputLayout
};
