// @frontify/fondue/dist/components/AssetInput/MultiAssetPreview/SelectedAssetsThumbnail.es.js
import { merge as a } from "../../../utilities/merge.es.js";
import r from "react";
import { AssetInputSize as l } from "../AssetInput.es.js";
import { AssetThumbnail as o } from "../AssetThumbnail.es.js";
const c = (t) => t.type === "image" || t.type === "logo", n = ({ assets: t }) => {
  const s = t.length, m = t.slice(0, 4);
  if (s < 4)
    for (const [e] of [...Array(4 - s)].entries())
      m.push({
        name: `empty picture ${e + 1}`,
        size: 20,
        type: "image",
        extension: "",
        src: "",
        source: "upload"
      });
  return /* @__PURE__ */ r.createElement("div", { className: "tw-border-black-20 tw-grid tw-grid-cols-2 tw-gap-0.5" }, m.map(
    (e, i) => c(e) ? /* @__PURE__ */ r.createElement(
      "div",
      {
        "data-test-id": "assets-image",
        key: e.name,
        style: e.src ? { backgroundImage: `url(${e.src})` } : {},
        className: a([
          i === 0 ? "tw-rounded-tl" : i === 2 && "tw-rounded-bl",
          "tw-h-11 tw-w-11 tw-bg-black-5"
        ])
      }
    ) : /* @__PURE__ */ r.createElement("div", { className: "tw-h-11 tw-w-11", key: e.src }, /* @__PURE__ */ r.createElement(o, { asset: e, size: l.Small, isMultiAsset: !0 }))
  ));
};
n.displayName = "FondueSelectedAssetsThumbnail";
export {
  n as SelectedAssetsThumbnail
};
