// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/plugins/TextStylePlugins/paragraphPlugin.es.js
import { jsx as l } from "react/jsx-runtime";
import { createPluginFactory as m, createParagraphPlugin as c } from "@udecode/plate";
import { Plugin as u, merge as g, alignmentClassnames as i, getColumnBreakClasses as P, MarkupElement as y } from "@frontify/fondue";
import { TextStyles as s, BlockStyles as h } from "../styles.es.js";
class A extends u {
  constructor({ styles: e = h.p, ...r } = {}) {
    super(s.p, {
      markupElement: new x(),
      label: "Body Text",
      ...r
    }), this.styles = {}, this.styles = e;
  }
  plugins() {
    return [E(this.styles)];
  }
}
const k = "tw-m-0 tw-px-0 tw-py-0", a = ({ element: t, attributes: e, children: r, styles: o }) => {
  const n = t.align, p = g([n && i[n], k, P(t)]);
  return /* @__PURE__ */ l("p", { ...e, className: p, style: o, children: r });
};
class x extends y {
  constructor(e = s.p, r = a) {
    super(e, r);
  }
}
const E = (t) => m({
  ...c(),
  key: s.p,
  isElement: !0,
  component: a
})({
  component: (e) => /* @__PURE__ */ l(a, { ...e, styles: t })
});
export {
  k as PARAGRAPH_CLASSES,
  x as ParagraphMarkupElement,
  a as ParagraphMarkupElementNode,
  A as ParagraphPlugin,
  E as createParagraphPlugin
};
