// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/plugins/TextStylePlugins/quotePlugin.es.js
import { jsx as o } from "react/jsx-runtime";
import { createPluginFactory as a } from "@udecode/plate";
import { Plugin as m, MarkupElement as i, merge as c, alignmentClassnames as p, getColumnBreakClasses as g } from "@frontify/fondue";
import { TextStyles as r, BlockStyles as k } from "../styles.es.js";
const y = "textstyle-quote-plugin";
class h extends m {
  constructor({ styles: e = k.quote, ...s } = {}) {
    super(r.quote, {
      label: "Quote",
      markupElement: new d(),
      ...s
    }), this.styles = {}, this.styles = e;
  }
  plugins() {
    return [q(this.styles)];
  }
}
class d extends i {
  constructor(e = y, s = l) {
    super(e, s);
  }
}
const l = ({ element: t, attributes: e, children: s, styles: u }) => {
  const n = t.align;
  return /* @__PURE__ */ o(
    "blockquote",
    {
      ...e,
      className: c([n && p[n], g(t), "a-quote"]),
      children: /* @__PURE__ */ o("span", { style: u, children: s })
    }
  );
}, q = (t) => a({
  key: r.quote,
  isElement: !0,
  component: l,
  deserializeHtml: {
    rules: [{ validNodeName: ["blockquote", "BLOCKQUOTE"] }]
  }
})({
  component: (e) => /* @__PURE__ */ o(l, { ...e, styles: t })
});
export {
  l as QuoteMarkupElementNode,
  h as QuotePlugin,
  q as createQuotePlugin
};
