// @frontify/app-bridge/dist/index.es.js
import { enableMapSet as e } from "immer";
import { AppBridgeCreateAsset as m } from "./AppBridgeCreateAsset.es.js";
import { HttpClientError as u } from "./errors/HttpClientError.es.js";
import { FetchError as a } from "./errors/NotifyError.es.js";
import { TimeoutReachedError as f } from "./errors/TimeoutReachedError.es.js";
import { closeAssetChooser as n, openAssetChooser as D } from "./registries/commands/AssetChooser.es.js";
import { openAssetViewer as y } from "./registries/commands/AssetViewer.es.js";
import { openNavigationManager as g } from "./registries/commands/NavigationManager.es.js";
import { closeTemplateChooser as C, openTemplateChooser as T } from "./registries/commands/TemplateChooser.es.js";
import { useAssetChooser as d } from "./react/useAssetChooser.es.js";
import { useAssetUpload as h } from "./react/useAssetUpload.es.js";
import { useAssetViewer as b } from "./react/useAssetViewer.es.js";
import { useBlockAssets as U } from "./react/useBlockAssets.es.js";
import { useBlockTemplates as R } from "./react/useBlockTemplates.es.js";
import { useBlockSettings as L } from "./react/useBlockSettings.es.js";
import { useBrandportalLink as j } from "./react/useBrandportalLink.es.js";
import { BulkDownloadState as F, useBulkDownload as G } from "./react/useBulkDownload.es.js";
import { useCategorizedDocumentPages as I } from "./react/useCategorizedDocumentPages.es.js";
import { useColorPalettes as N } from "./react/useColorPalettes.es.js";
import { useColors as z } from "./react/useColors.es.js";
import { useCoverPage as q } from "./react/useCoverPage.es.js";
import { useDocumentCategories as K } from "./react/useDocumentCategories.es.js";
import { useDocumentGroups as W } from "./react/useDocumentGroups.es.js";
import { useDocumentPageTargets as Y } from "./react/useDocumentPageTargets.es.js";
import { useDocumentSection as $ } from "./react/useDocumentSection.es.js";
import { useDocumentTargets as oe } from "./react/useDocumentTargets.es.js";
import { useEditorState as te } from "./react/useEditorState.es.js";
import { useFileInput as pe } from "./react/useFileInput.es.js";
import { useFileUpload as se } from "./react/useFileUpload.es.js";
import { useGroupedDocuments as ie } from "./react/useGroupedDocuments.es.js";
import { useGuidelineActions as xe } from "./react/useGuidelineActions.es.js";
import { useNavigationManager as De } from "./react/useNavigationManager.es.js";
import { usePageTemplateSettings as ye } from "./react/usePageTemplateSettings.es.js";
import { usePrivacySettings as ge } from "./react/usePrivacySettings.es.js";
import { useReadyForPrint as Ce } from "./react/useReadyForPrint.es.js";
import { useTemplateAssets as Pe } from "./react/useTemplateAssets.es.js";
import { useThemeAssets as Se } from "./react/useThemeAssets.es.js";
import { useTemplateChooser as Be } from "./react/useTemplateChooser.es.js";
import { useUncategorizedDocumentPages as ke } from "./react/useUncategorizedDocumentPages.es.js";
import { useUngroupedDocuments as Ee } from "./react/useUngroupedDocuments.es.js";
import { getAppBridgeBlockStub as ve } from "./tests/AppBridgeBlockStub.es.js";
import { getAppBridgePlatformAppStub as Oe } from "./tests/AppBridgePlatformAppStub.es.js";
import { getAppBridgeThemeStub as we } from "./tests/AppBridgeThemeStub.es.js";
import { AssetApiDummy as Ge } from "./tests/AssetApiDummy.es.js";
import { AssetChooserOptionsDummy as Ie } from "./tests/AssetChooserOptionsDummy.es.js";
import { AssetChooserResultDummy as Ne } from "./tests/AssetChooserResultDummy.es.js";
import { AssetDummy as ze } from "./tests/AssetDummy.es.js";
import { BrandportalLinkApiDummy as qe } from "./tests/BrandportalLinkApiDummy.es.js";
import { BrandportalLinkDummy as Ke } from "./tests/BrandportalLinkDummy.es.js";
import { BulkDownloadDummy as We } from "./tests/BulkDownloadDummy.es.js";
import { ColorApiCreateDummy as Ye, ColorApiDummy as Ze, ColorApiPatchDummy as $e } from "./tests/ColorApiDummy.es.js";
import { ColorCreateDummy as oo, ColorDummy as ro, ColorPatchDummy as to } from "./tests/ColorDummy.es.js";
import { ColorPaletteApiCreateDummy as po, ColorPaletteApiDummy as uo, ColorPaletteApiPatchDummy as so } from "./tests/ColorPaletteApiDummy.es.js";
import { ColorPaletteCreateDummy as io, ColorPaletteDummy as fo, ColorPalettePatchDummy as xo } from "./tests/ColorPaletteDummy.es.js";
import { CoverPageApiDummy as Do } from "./tests/CoverPageApiDummy.es.js";
import { CoverPageDummy as yo } from "./tests/CoverPageDummy.es.js";
import { DocumentApiDummy as go } from "./tests/DocumentApiDummy.es.js";
import { DocumentCategoryApiDummy as Co } from "./tests/DocumentCategoryApiDummy.es.js";
import { DocumentCategoryDummy as Po } from "./tests/DocumentCategoryDummy.es.js";
import { DocumentDummy as ho } from "./tests/DocumentDummy.es.js";
import { DocumentGroupApiDummy as bo } from "./tests/DocumentGroupApiDummy.es.js";
import { DocumentGroupDummy as Uo } from "./tests/DocumentGroupDummy.es.js";
import { DocumentPageApiDummy as Ro, DocumentPageDuplicateApiDummy as vo } from "./tests/DocumentPageApiDummy.es.js";
import { DocumentPageDummy as Oo, DocumentPageDuplicateDummy as jo } from "./tests/DocumentPageDummy.es.js";
import { DocumentSectionApiDummy as Fo } from "./tests/DocumentSectionApiDummy.es.js";
import { DocumentSectionDummy as Mo } from "./tests/DocumentSectionDummy.es.js";
import { GuidelineSearchResultApiDummy as Ho } from "./tests/GuidelineSearchResultApiDummy.es.js";
import { GuidelineSearchResultDummy as Vo } from "./tests/GuidelineSearchResultDummy.es.js";
import { HttpUtilResponseDummy as _o } from "./tests/HttpUtilResponseDummy.es.js";
import { DocumentPageTargetsApiDummy as Jo, DocumentTargetsApiDummy as Ko, SingleTargetApiDummy as Qo, UpdateTargetsApiDummy as Wo } from "./tests/TargetsApiDummy.es.js";
import { DocumentPageTargetsDummy as Yo, DocumentTargetsDummy as Zo, UpdateTargetsDummy as $o } from "./tests/TargetsDummy.es.js";
import { TemplateApiLegacyDummy as or } from "./tests/TemplateApiLegacyDummy.es.js";
import { TemplateLegacyDummy as tr } from "./tests/TemplateLegacyDummy.es.js";
import { TemplateDummy as pr } from "./tests/TemplateDummy.es.js";
import { UserApiDummy as sr } from "./tests/UserApiDummy.es.js";
import { UserDummy as ir } from "./tests/UserDummy.es.js";
import { withAppBridgeBlockStubs as xr } from "./tests/withAppBridgeBlockStubs.es.js";
import { withAppBridgeThemeStubs as Dr } from "./tests/withAppBridgeThemeStubs.es.js";
import { LinkSettingsDisplay as yr, LinkSettingsIconPosition as cr, LinkType as gr } from "./types/Document.es.js";
import { DocumentPageVisibility as Cr } from "./types/DocumentPage.es.js";
import { FileExtension as Pr } from "./types/FileExtension.es.js";
import { FileExtensionSets as Sr } from "./types/FileExtensionSets.es.js";
import { FileType as Br } from "./types/FileType.es.js";
import { AssetChooserObjectType as kr, AssetChooserProjectType as Ur, TerrificEvent as Er } from "./types/Terrific.es.js";
import { Topic as vr } from "./types/Topic.es.js";
import { rgbObjectToRgbString as Or, rgbStringToRgbObject as jr } from "./utilities/color.es.js";
import { getDatasetByClassName as Fr, getDatasetByElement as Gr } from "./utilities/dom.es.js";
import { getMimeType as Ir, mimeTypes as Hr } from "./utilities/file.es.js";
import { generateRandomString as Vr } from "./utilities/hash.es.js";
import { HttpClient as _r } from "./utilities/httpClient.es.js";
import { notify as Jr } from "./utilities/notify.es.js";
import { compareObjects as Qr, convertObjectCase as Wr, isObject as Xr, mergeDeep as Yr } from "./utilities/object.es.js";
import { SUBSCRIBE_TIMEOUT as $r, subscribe as et } from "./utilities/subscribe.es.js";
e();
export {
  m as AppBridgeCreateAsset,
  Ge as AssetApiDummy,
  kr as AssetChooserObjectType,
  Ie as AssetChooserOptionsDummy,
  Ur as AssetChooserProjectType,
  Ne as AssetChooserResultDummy,
  ze as AssetDummy,
  qe as BrandportalLinkApiDummy,
  Ke as BrandportalLinkDummy,
  We as BulkDownloadDummy,
  F as BulkDownloadState,
  Ye as ColorApiCreateDummy,
  Ze as ColorApiDummy,
  $e as ColorApiPatchDummy,
  oo as ColorCreateDummy,
  ro as ColorDummy,
  po as ColorPaletteApiCreateDummy,
  uo as ColorPaletteApiDummy,
  so as ColorPaletteApiPatchDummy,
  io as ColorPaletteCreateDummy,
  fo as ColorPaletteDummy,
  xo as ColorPalettePatchDummy,
  to as ColorPatchDummy,
  Do as CoverPageApiDummy,
  yo as CoverPageDummy,
  go as DocumentApiDummy,
  Co as DocumentCategoryApiDummy,
  Po as DocumentCategoryDummy,
  ho as DocumentDummy,
  bo as DocumentGroupApiDummy,
  Uo as DocumentGroupDummy,
  Ro as DocumentPageApiDummy,
  Oo as DocumentPageDummy,
  vo as DocumentPageDuplicateApiDummy,
  jo as DocumentPageDuplicateDummy,
  Jo as DocumentPageTargetsApiDummy,
  Yo as DocumentPageTargetsDummy,
  Cr as DocumentPageVisibility,
  Fo as DocumentSectionApiDummy,
  Mo as DocumentSectionDummy,
  Ko as DocumentTargetsApiDummy,
  Zo as DocumentTargetsDummy,
  a as FetchError,
  Pr as FileExtension,
  Sr as FileExtensionSets,
  Br as FileType,
  Ho as GuidelineSearchResultApiDummy,
  Vo as GuidelineSearchResultDummy,
  _r as HttpClient,
  u as HttpClientError,
  _o as HttpUtilResponseDummy,
  yr as LinkSettingsDisplay,
  cr as LinkSettingsIconPosition,
  gr as LinkType,
  $r as SUBSCRIBE_TIMEOUT,
  Qo as SingleTargetApiDummy,
  or as TemplateApiLegacyDummy,
  pr as TemplateDummy,
  tr as TemplateLegacyDummy,
  Er as TerrificEvent,
  f as TimeoutReachedError,
  vr as Topic,
  Wo as UpdateTargetsApiDummy,
  $o as UpdateTargetsDummy,
  sr as UserApiDummy,
  ir as UserDummy,
  n as closeAssetChooser,
  C as closeTemplateChooser,
  Qr as compareObjects,
  Wr as convertObjectCase,
  Vr as generateRandomString,
  ve as getAppBridgeBlockStub,
  Oe as getAppBridgePlatformAppStub,
  we as getAppBridgeThemeStub,
  Fr as getDatasetByClassName,
  Gr as getDatasetByElement,
  Ir as getMimeType,
  Xr as isObject,
  Yr as mergeDeep,
  Hr as mimeTypes,
  Jr as notify,
  D as openAssetChooser,
  y as openAssetViewer,
  g as openNavigationManager,
  T as openTemplateChooser,
  Or as rgbObjectToRgbString,
  jr as rgbStringToRgbObject,
  et as subscribe,
  d as useAssetChooser,
  h as useAssetUpload,
  b as useAssetViewer,
  U as useBlockAssets,
  L as useBlockSettings,
  R as useBlockTemplates,
  j as useBrandportalLink,
  G as useBulkDownload,
  I as useCategorizedDocumentPages,
  N as useColorPalettes,
  z as useColors,
  q as useCoverPage,
  K as useDocumentCategories,
  W as useDocumentGroups,
  Y as useDocumentPageTargets,
  $ as useDocumentSection,
  oe as useDocumentTargets,
  te as useEditorState,
  pe as useFileInput,
  se as useFileUpload,
  ie as useGroupedDocuments,
  xe as useGuidelineActions,
  De as useNavigationManager,
  ye as usePageTemplateSettings,
  ge as usePrivacySettings,
  Ce as useReadyForPrint,
  Pe as useTemplateAssets,
  Be as useTemplateChooser,
  Se as useThemeAssets,
  ke as useUncategorizedDocumentPages,
  Ee as useUngroupedDocuments,
  xr as withAppBridgeBlockStubs,
  Dr as withAppBridgeThemeStubs
};
