// @frontify/fondue/dist/components/RichTextEditor/Plugins/TextStylePlugin/TextStyles/custom2Plugin.es.js
var a = Object.defineProperty;
var i = (e, t, s) => t in e ? a(e, t, { enumerable: !0, configurable: !0, writable: !0, value: s }) : e[t] = s;
var l = (e, t, s) => (i(e, typeof t != "symbol" ? t + "" : t, s), s);
import { createPluginFactory as c } from "@udecode/plate";
import n from "react";
import { alignmentClassnames as p } from "../../helper.es.js";
import { merge as g } from "../../../../../utilities/merge.es.js";
import { TextStyles as r } from "../types.es.js";
import { Plugin as f } from "../../Plugin.es.js";
import { defaultStyles as y } from "../../../utils/defaultStyles.es.js";
import { MarkupElement as C } from "../../MarkupElement.es.js";
import { getColumnBreakClasses as E } from "../../ColumnBreakPlugin/utils/getColumnBreakClasses.es.js";
const d = "textstyle-custom2-plugin";
class D extends f {
  constructor({ styles: s = y.custom2, ...m } = {}) {
    super(r.custom2, {
      label: "Custom 2",
      markupElement: new k(),
      ...m
    });
    l(this, "styles", {});
    this.styles = s;
  }
  plugins() {
    return [x(this.styles)];
  }
}
class k extends C {
  constructor(t = d, s = u) {
    super(t, s);
  }
}
const u = ({ element: e, attributes: t, children: s, styles: m }) => {
  const o = e.align;
  return /* @__PURE__ */ n.createElement(
    "p",
    {
      ...t,
      className: g([o && p[o], E(e)]),
      style: m
    },
    s
  );
}, x = (e) => c({
  key: r.custom2,
  isElement: !0,
  deserializeHtml: {
    rules: [{ validClassName: r.custom2 }]
  }
})({
  component: (t) => /* @__PURE__ */ n.createElement(u, { ...t, styles: e })
});
export {
  D as Custom2Plugin
};
