// @frontify/fondue/dist/components/RichTextEditor/Plugins/TextStylePlugin/TextStyles/imageCaptionPlugin.es.js
var l = Object.defineProperty;
var p = (t, e, a) => e in t ? l(t, e, { enumerable: !0, configurable: !0, writable: !0, value: a }) : t[e] = a;
var s = (t, e, a) => (p(t, typeof e != "symbol" ? e + "" : e, a), a);
import { createPluginFactory as g } from "@udecode/plate";
import i from "react";
import { alignmentClassnames as u } from "../../helper.es.js";
import { merge as c } from "../../../../../utilities/merge.es.js";
import { TextStyles as o } from "../types.es.js";
import { Plugin as C } from "../../Plugin.es.js";
import { defaultStyles as f } from "../../../utils/defaultStyles.es.js";
import { MarkupElement as y } from "../../MarkupElement.es.js";
import { getColumnBreakClasses as E } from "../../ColumnBreakPlugin/utils/getColumnBreakClasses.es.js";
const d = "textstyle-imageCaption-plugin";
class B extends C {
  constructor({ styles: a = f.imageCaption, ...m } = {}) {
    super(o.imageCaption, {
      label: "Image Caption",
      markupElement: new k(),
      ...m
    });
    s(this, "styles", {});
    this.styles = a;
  }
  plugins() {
    return [I(this.styles)];
  }
}
class k extends y {
  constructor(e = d, a = n) {
    super(e, a);
  }
}
const n = ({ element: t, attributes: e, children: a, styles: m }) => {
  const r = t.align;
  return /* @__PURE__ */ i.createElement(
    "p",
    {
      ...e,
      className: c([r && u[r], E(t)]),
      style: m
    },
    a
  );
}, I = (t) => g({
  key: o.imageCaption,
  isElement: !0,
  component: n,
  deserializeHtml: {
    rules: [{ validClassName: o.imageCaption }]
  }
})({
  component: (e) => /* @__PURE__ */ i.createElement(n, { ...e, styles: t })
});
export {
  B as ImageCaptionPlugin
};
