// @frontify/fondue/dist/components/RichTextEditor/Plugins/TextStylePlugin/TextStyles/quotePlugin.es.js
var a = Object.defineProperty;
var i = (t, e, o) => e in t ? a(t, e, { enumerable: !0, configurable: !0, writable: !0, value: o }) : t[e] = o;
var m = (t, e, o) => (i(t, typeof e != "symbol" ? e + "" : e, o), o);
import { createPluginFactory as c } from "@udecode/plate";
import n from "react";
import { alignmentClassnames as p } from "../../helper.es.js";
import { merge as g } from "../../../../../utilities/merge.es.js";
import { TextStyles as u } from "../types.es.js";
import { Plugin as f } from "../../Plugin.es.js";
import { defaultStyles as y } from "../../../utils/defaultStyles.es.js";
import { MarkupElement as k } from "../../MarkupElement.es.js";
import { getColumnBreakClasses as E } from "../../ColumnBreakPlugin/utils/getColumnBreakClasses.es.js";
const d = "textstyle-quote-plugin";
class T extends f {
  constructor({ styles: o = y.quote, ...r } = {}) {
    super(u.quote, {
      label: "Quote",
      markupElement: new q(),
      ...r
    });
    m(this, "styles", {});
    this.styles = o;
  }
  plugins() {
    return [Q(this.styles)];
  }
}
class q extends k {
  constructor(e = d, o = s) {
    super(e, o);
  }
}
const s = ({ element: t, attributes: e, children: o, styles: r }) => {
  const l = t.align;
  return /* @__PURE__ */ n.createElement(
    "blockquote",
    {
      ...e,
      className: g([l && p[l], E(t)]),
      style: r
    },
    o
  );
}, Q = (t) => c({
  key: u.quote,
  isElement: !0,
  component: s,
  deserializeHtml: {
    rules: [{ validNodeName: ["blockquote", "BLOCKQUOTE"] }]
  }
})({
  component: (e) => /* @__PURE__ */ n.createElement(s, { ...e, styles: t })
});
export {
  s as QuoteMarkupElementNode,
  T as QuotePlugin,
  Q as createQuotePlugin
};
