// @frontify/fondue/dist/typography/Text/Text.es.js
import { merge as m } from "../../utilities/merge.es.js";
import d from "react";
import { decorationMap as s, wordBreakMap as p, whitespaceMap as c, overflowMap as f, displayMap as u } from "../shared/records.es.js";
const g = {
  default: "tw-font-regular",
  strong: "tw-font-medium",
  "x-strong": "tw-font-bold"
}, b = {
  "x-small": "tw-text-body-x-small",
  small: "tw-text-body-small",
  medium: "tw-text-body-medium",
  large: "tw-text-body-large"
}, M = {
  default: "tw-text-text",
  weak: "tw-text-text-weak",
  "x-weak": "tw-text-text-x-weak",
  disabled: "tw-text-text-disabled",
  negative: "tw-text-text-negative",
  positive: "tw-text-text-positive",
  warning: "tw-text-text-warning",
  interactive: "tw-text-text-interactive"
}, v = ({
  children: a,
  as: o = "span",
  weight: x = "default",
  size: i = "medium",
  color: l = "default",
  decoration: n = "none",
  wordBreak: w = "normal",
  whitespace: r = "normal",
  overflow: t = "visible",
  display: e
}) => /* @__PURE__ */ d.createElement(
  o,
  {
    "data-test-id": "text",
    className: m([
      "tw-font-body tw-max-w-full",
      g[x],
      b[i],
      M[l],
      s[n],
      p[w],
      t !== "truncate" && c[r],
      f[t],
      e && u[e]
    ])
  },
  a
);
v.displayName = "FondueText";
export {
  v as Text
};
