// @frontify/fondue/dist/components/RichTextEditor/Plugins/ListPlugin/OrderedListPlugin/OrderedListMarkupElement.es.js
import { usePlateEditorRef as s, ELEMENT_OL as c, findNodePath as l, getNodeAncestors as p, ELEMENT_LI as a } from "@udecode/plate";
import i from "react";
import { MarkupElement as u } from "../../MarkupElement.es.js";
const m = [
  "[&>li>p]:before:tw-content-[counter(count,decimal)_'._']",
  "[&>li>p]:before:tw-content-[counter(count,_lower-alpha)_'._']",
  "[&>li>p]:before:tw-content-[counter(count,lower-roman)_'._']"
], d = (e, o) => {
  const t = l(e, o);
  if (!t)
    return 0;
  const r = p(e, t);
  return Array.from(r).filter((n) => n[0].type === a).length;
}, f = (e) => `tw-list-none tw-pl-[10px] tw-mb-[10px] tw-ml-[15px] [&>li>p]:before:tw-pr-1 [&>li>p]:before:tw-tabular-nums ${m[e % 3]}`, E = { counterReset: "count" }, L = ({
  attributes: e,
  children: o,
  element: t
}) => {
  const r = s(), n = d(r, t);
  return /* @__PURE__ */ i.createElement("ol", { className: f(n), ...e, style: E }, o);
};
class N extends u {
  constructor(o = c, t = L) {
    super(o, t);
  }
}
export {
  E as OL_STYLES,
  N as OrderedListMarkupElement,
  L as OrderedListMarkupElementNode,
  f as getOrderedListClasses
};
