// @frontify/fondue/dist/components/Flex/Flex.es.js
import A from "react";
import { merge as M } from "../../utilities/merge.es.js";
import { SPACING_VALUES as t, PADDING_VALUES_MAP as n, MARGIN_VALUES_MAP as r } from "../../utilities/dimensions.es.js";
import { Box as j } from "../Box/Box.es.js";
const N = {
  row: "tw-flex-row",
  column: "tw-flex-col",
  "row-reverse": "tw-flex-row-reverse",
  "column-reverse": "tw-flex-col-reverse"
}, _ = {
  nowrap: "tw-flex-nowrap",
  wrap: "tw-flex-wrap",
  "wrap-reverse": "tw-flex-wrap-reverse"
}, h = {
  start: "tw-justify-start",
  end: "tw-justify-end",
  center: "tw-justify-center",
  between: "tw-justify-between",
  around: "tw-justify-around",
  evenly: "tw-justify-evenly"
}, C = {
  start: "tw-items-start",
  end: "tw-items-end",
  center: "tw-items-center",
  baseline: "tw-items-baseline",
  stretch: "tw-items-stretch"
}, E = {
  start: "tw-content-start",
  end: "tw-content-end",
  center: "tw-content-center",
  normal: "tw-content-normal",
  between: "tw-content-between",
  around: "tw-content-around",
  evenly: "tw-content-evenly",
  baseline: "tw-content-baseline",
  stretch: "tw-content-stretch"
}, I = ({
  direction: s = "row",
  wrap: o = "nowrap",
  justify: a = "center",
  alignItems: w = "center",
  alignContent: c = "center",
  "data-test-id": i = "fondue-flex",
  children: l,
  bg: m,
  color: p,
  as: f = "div",
  margin: d = 0,
  padding: e = 0,
  minWidth: u,
  maxWidth: x,
  maxHeight: y,
  minHeight: v
}) => {
  const g = t.includes(e) ? n[e] : n[0], b = t.includes(e) ? r[d] : r[0];
  return /* @__PURE__ */ A.createElement(
    j,
    {
      "data-test-id": i,
      className: M([
        "tw-flex",
        "tw-gap-1",
        N[s],
        _[o],
        h[a],
        C[w],
        E[c],
        m,
        p,
        g,
        b
      ]),
      style: {
        maxWidth: x,
        minWidth: u,
        maxHeight: y,
        minHeight: v
      },
      as: f
    },
    l
  );
};
I.displayName = "FondueFlex";
export {
  I as Flex
};
