// @frontify/fondue/dist/components/Tooltip/BrightHeader.es.js
import t, { cloneElement as o } from "react";
import { merge as r } from "../../utilities/merge.es.js";
import { IconSize as n } from "../../foundation/Icon/IconSize.es.js";
import i from "../../foundation/Icon/Generated/IconExclamationMarkCircle.es.js";
import a from "../../foundation/Icon/Generated/IconExclamationMarkTriangle.es.js";
import m from "../../foundation/Icon/Generated/IconCheckMark.es.js";
import s from "../../foundation/Icon/Generated/IconDocument.es.js";
var g = /* @__PURE__ */ ((e) => (e.Information = "Information", e.Warning = "Warning", e.Tip = "Tip", e.Note = "Note", e))(g || {});
const c = {
  Information: "tw-bg-box-selected-strong",
  Warning: "tw-bg-box-negative-strong-pressed",
  Tip: "tw-bg-box-positive-strong-pressed",
  Note: "tw-bg-box-warning-strong"
}, E = {
  Information: "before:tw-bg-box-selected-strong",
  Warning: "before:tw-bg-box-negative-strong-pressed",
  Tip: "before:tw-bg-box-positive-strong-pressed",
  Note: "before:tw-bg-box-warning-strong"
}, b = {
  Information: /* @__PURE__ */ t.createElement(i, null),
  Warning: /* @__PURE__ */ t.createElement(a, null),
  Tip: /* @__PURE__ */ t.createElement(m, null),
  Note: /* @__PURE__ */ t.createElement(s, null)
}, l = ({ headerStyle: e }) => /* @__PURE__ */ t.createElement(
  "div",
  {
    "data-test-id": "bright-header",
    className: r([
      "tw-h-9 tw-relative tw-z-[120000] tw-flex tw-justify-center tw-items-center tw-rounded-t-md tw-dark tw-text-text",
      c[e]
    ])
  },
  o(b[e], { size: n.Size20 })
);
l.displayName = "FondueBrightHeader";
export {
  l as BrightHeader,
  g as BrightHeaderStyle,
  E as brightHeaderArrowBackgroundColors,
  c as brightHeaderBackgroundColors
};
