// @frontify/fondue/dist/components/AddBlockButton/AddBlockButton.es.js
import { IconSize as p } from "../../foundation/Icon/IconSize.es.js";
import { useButton as w } from "@react-aria/button";
import { useFocusRing as d } from "@react-aria/focus";
import { mergeProps as g } from "@react-aria/utils";
import { FOCUS_STYLE as u } from "../../utilities/focusStyle.es.js";
import { merge as b } from "../../utilities/merge.es.js";
import r, { useRef as f } from "react";
import v from "../../foundation/Icon/Generated/IconPlus.es.js";
var h = /* @__PURE__ */ ((t) => (t.Horizontal = "Horizontal", t.Vertical = "Vertical", t))(h || {}), x = /* @__PURE__ */ ((t) => (t.Left = "Left", t.Top = "Top", t.Bottom = "Bottom", t.Right = "Right", t))(x || {});
const R = {
  Horizontal: "tw-rotate-0",
  Vertical: "tw-rotate-90"
}, s = {
  Left: "tw-rotate-180",
  Top: "tw-rotate-[270deg]",
  Bottom: "tw-rotate-90",
  Right: "tw-rotate-0"
}, z = ({
  onClick: t,
  title: a,
  direction: e,
  orientation: o,
  "data-test-id": n = "add-block-button"
}) => {
  const { isFocusVisible: l, focusProps: m } = d(), i = f(null), { buttonProps: c } = w({ onPress: () => t() }, i);
  return /* @__PURE__ */ r.createElement(
    "button",
    {
      ...g(c, m),
      title: a,
      "data-test-id": n,
      className: b([
        "tw-group tw-leading-none tw-rounded-sm tw-outline-none",
        l && u,
        o && !e && R[o],
        e && s[e],
        !e && !o && s.Right
      ])
    },
    /* @__PURE__ */ r.createElement(
      "span",
      {
        className: "tw-text-box-selected-strong-inverse tw-bg-box-selected-strong tw-rounded tw-inline-flex tw-items-center tw-w-7 tw-h-6 tw-relative tw-p-1 tw-transition-colors hover:tw-bg-box-selected-strong-hover hover:tw-text-box-selected-strong-inverse-hover group-active:tw-bg-box-selected-strong-pressed group-active:tw-text-bg-box-selected-strong-inverse-pressed",
        style: {
          clipPath: `path(
                        "M27.3333 10.9369L21.7246 2C20.5 0.5 19.5 0 18 0H4C1.79086 0 0 1.79086 0 4V20C0 22.2091 1.79086 24 4 24H18C19.5 24 20.5 23.5 21.7246 22L27.3333 13.0631C27.7412 12.4131 27.7412 11.5869 27.3333 10.9369Z"
                    )`
        }
      },
      /* @__PURE__ */ r.createElement(v, { size: p.Size16 })
    )
  );
};
z.displayName = "FondueAddBlockButton";
export {
  z as AddBlockButton,
  h as AddBlockButtonDirection,
  x as ButtonDirection
};
