// @frontify/fondue/dist/components/DialogFooter/DialogFooter.es.js
import t from "react";
import { dialogPaddingMap as s } from "../../types/dialog.es.js";
import c from "../../foundation/Icon/Generated/IconArrowLeft.es.js";
import { merge as p } from "../../utilities/merge.es.js";
import { Box as o } from "../Box/Box.es.js";
import { Button as m } from "../Button/Button.es.js";
import { ButtonEmphasis as f, ButtonSize as d } from "../Button/ButtonTypes.es.js";
const w = ({
  actionButtons: i,
  children: n,
  backButton: r,
  padding: l = "small",
  "data-test-id": e = "fondue-dialog-footer"
}) => /* @__PURE__ */ t.createElement(
  o,
  {
    "data-test-id": e,
    className: p([
      "tw-border-t tw-border-t-line tw-flex tw-items-center tw-justify-between tw-gap-x-3",
      s[l]
    ])
  },
  r && /* @__PURE__ */ t.createElement(
    m,
    {
      "data-test-id": `${e}-back-button`,
      ...r,
      emphasis: f.Default,
      icon: /* @__PURE__ */ t.createElement(c, null)
    }
  ),
  /* @__PURE__ */ t.createElement(o, { className: "tw-grow", "data-test-id": `${e}-content` }, n),
  /* @__PURE__ */ t.createElement(o, { className: "tw-flex tw-gap-x-3", "data-test-id": `${e}-action-buttons` }, i.map((a) => /* @__PURE__ */ t.createElement(m, { key: `${e}-button-${a.children}`, ...a, size: d.Medium })))
);
w.displayName = "FondueDialogFooter";
export {
  w as DialogFooter
};
