// @frontify/app-bridge/dist/tests/DocumentPageApiDummy.es.js
import { DocumentPageVisibility as l } from "../types/DocumentPage.es.js";
class t {
  static with(e) {
    return {
      id: e,
      creator: 9,
      created: "2022-08-15T11:47:57.000+00:00",
      modifier: null,
      modified: null,
      document_id: 145,
      category_id: 29,
      parent_id: null,
      valid_from: "2022-08-15T11:47:57.000+00:00",
      valid_to: null,
      title: `Document Page Dummy ${e}`,
      slug: `document-page-dummy-${e}`,
      sort: 1,
      link_type: "INTERNAL",
      link_url: null,
      view_count: 0,
      visibility: l.Everyone,
      change_processed: null,
      change_processed_by: null,
      change_skipped: null,
      change_skipped_by: null,
      change_comment: null,
      change_comment_by: null,
      change_title: null,
      targets: null,
      permanent_link: `/r/document-page-${e}`
    };
  }
  static withFields(e) {
    return {
      ...t.with(e.id),
      ...e
    };
  }
}
class u {
  static with(e) {
    return {
      page: {
        id: e,
        link_type: "INTERNAL",
        name: "Document page duplicate dummy",
        sections: [],
        url: `document-page-duplicate-dummy-${e}`,
        visibility: "EVERYONE"
      }
    };
  }
}
export {
  t as DocumentPageApiDummy,
  u as DocumentPageDuplicateApiDummy
};
