// @frontify/fondue/dist/components/LinkChooser/utils/useManualComboBoxHandlers.es.js
import { chain as t } from "@react-aria/utils";
const B = ({ inputProps: f, inputRef: d, popoverRef: y, state: e }, { onOpen: c, onClose: u, onNavigate: E, onSelect: K }) => {
  const { onClick: m, onTouchEnd: p, onBlur: T, onFocus: g, onKeyDown: a, ...h } = f;
  return {
    ...h,
    onClick: t(m, c),
    onTouchEnd: t(p, c),
    onBlur: t(T, (o) => {
      var n, s, i;
      (n = y.current) != null && n.contains(o.relatedTarget) || ((s = o.relatedTarget) == null ? void 0 : s.dataset.comboBoxControl) === "true" && e.isOpen ? (i = d.current) == null || i.focus() : u();
    }),
    onFocus: t(g, c),
    onKeyDown: (o) => {
      const n = ["Enter", "Tab"], s = ["ArrowRight", "ArrowLeft", ...n], i = e.selectionManager.focusedKey, r = e.collection.getItem(i), { key: l } = o;
      if (e.isOpen && n.includes(l))
        if (r && r.parentKey !== "search") {
          E(r.key);
          return;
        } else if (r)
          K(r.key);
        else {
          u();
          return;
        }
      else
        e.isOpen && l === "Escape" ? u() : !e.isOpen && !s.includes(l) && c();
      a && a(o);
    }
  };
};
export {
  B as useManualComboBoxEventHandlers
};
