// @frontify/fondue/dist/components/Badge/Badge.es.js
import { IconSize as p } from "../../foundation/Icon/IconSize.es.js";
import { FOCUS_VISIBLE_STYLE as u } from "../../utilities/focusStyle.es.js";
import { merge as s } from "../../utilities/merge.es.js";
import r, { cloneElement as y } from "react";
import { BadgeStatusIcon as N } from "./BadgeStatusIcon.es.js";
import { getStyleClasses as B, getCircularSizeClasses as C, getSizeClasses as v } from "./helpers.es.js";
import { BadgeEmphasis as g, BadgeStyle as z } from "./types.es.js";
import $ from "../../foundation/Icon/Generated/IconCross.es.js";
const j = ({
  children: e,
  status: a,
  icon: n,
  style: d = z.Primary,
  size: w = "medium",
  emphasis: x = g.None,
  disabled: l = !1,
  onClick: i,
  onDismiss: m,
  "data-test-id": o = "badge",
  withTitleAttribute: b = !0
}) => {
  if (!e && !n && !a)
    return null;
  const S = i ? "button" : "span", c = (t) => ["string", "number"].includes(typeof t) ? t : t instanceof Array ? t.map(c).join("") : t && typeof t == "object" && "props" in t ? c(t.props.children) : "", f = c(e), E = !e && !m && Boolean(a || n);
  return /* @__PURE__ */ r.createElement(
    "div",
    {
      className: s([
        "tw-relative tw-rounded-full tw-inline-flex tw-items-center tw-align-top tw-min-w-0 tw-flex-initial tw-transition-color",
        l ? "tw-bg-box-disabled tw-text-box-disabled-inverse" : B(d, !!i, x === g.Strong)
      ]),
      "data-test-id": o
    },
    /* @__PURE__ */ r.createElement(
      S,
      {
        onClick: () => i && i(),
        className: s([
          "tw-inline-flex tw-items-center tw-justify-center tw-rounded-full tw-select-none tw-flex-initial tw-min-w-0",
          i && !l ? "hover:tw-cursor-pointer" : "tw-cursor-default",
          E ? C(w) : v(e, a, n, w, !!m),
          u
        ]),
        "data-test-id": `${o}-button`,
        title: b ? f : void 0
      },
      a && /* @__PURE__ */ r.createElement(N, { status: a, disabled: l, "data-test-id": o }),
      n && /* @__PURE__ */ r.createElement(
        "span",
        {
          "data-test-id": `${o}-icon`,
          className: s([
            "tw-flex-none tw-leading-none",
            l && "tw-opacity-30",
            !!e && "tw-pr-1"
          ])
        },
        y(n, { size: p.Size16 })
      ),
      e && /* @__PURE__ */ r.createElement("span", { className: "tw-text-center tw-text-xxs tw-font-sans tw-font-normal tw-truncate tw-px-0.5" }, e)
    ),
    m && /* @__PURE__ */ r.createElement(
      "button",
      {
        type: "button",
        "data-test-id": `${o}-dismiss`,
        className: s([
          "tw-absolute tw-rounded tw-leading-4",
          u,
          w === "medium" ? "tw-right-1.5" : "tw-right-1"
        ]),
        onClick: () => m(),
        "aria-label": `Dismiss ${f}`
      },
      /* @__PURE__ */ r.createElement($, { size: p.Size16 })
    )
  );
};
j.displayName = "FondueBadge";
export {
  j as Badge
};
