// @frontify/guideline-blocks-settings/dist/components/DownloadButton/DownloadButton.es.js
import { jsx as t } from "react/jsx-runtime";
import { useFocusRing as i } from "@react-aria/focus";
import { Tooltip as s, TooltipPosition as l, FOCUS_STYLE as w, IconArrowCircleDown16 as a } from "@frontify/fondue";
import { joinClassNames as u } from "../../utilities/react/joinClassNames.es.js";
const b = ({ onDownload: o }) => {
  const { isFocused: e, focusProps: n } = i();
  return /* @__PURE__ */ t(
    s,
    {
      withArrow: !0,
      position: l.Top,
      content: "Download",
      enterDelay: 500,
      triggerElement: /* @__PURE__ */ t(
        "button",
        {
          tabIndex: 0,
          "aria-label": "Download",
          ...n,
          className: u(["tw-outline-none tw-rounded", e && w]),
          onClick: o,
          onPointerDown: (r) => r.preventDefault(),
          children: /* @__PURE__ */ t(
            "span",
            {
              "data-test-id": "download-button",
              className: "tw-flex tw-text-xs tw-font-body tw-items-center tw-gap-1 tw-rounded-full tw-bg-box-neutral-strong-inverse hover:tw-bg-box-neutral-strong-inverse-hover active:tw-bg-box-neutral-strong-inverse-pressed tw-text-box-neutral-strong tw-outline tw-outline-1 tw-outline-offset-1 tw-p-1.5 tw-outline-line",
              children: /* @__PURE__ */ t(a, {})
            }
          )
        }
      )
    }
  );
};
export {
  b as DownloadButton
};
