// @frontify/guideline-blocks-settings/dist/settings/gutter.es.js
import "../styles.css.es.js";
import { presetCustomValue as C, numericalOrPixelRule as m, appendUnit as f } from "@frontify/sidebar-settings";
import { GutterSpacing as l, gutterSpacingStyleMap as h } from "./types.es.js";
const I = (e) => {
  const t = e != null && e.id ? e.id : "hasCustomSpacing", g = e != null && e.dependentSettingId ? e.dependentSettingId : "columns", c = e != null && e.spacingChoiceId ? e.spacingChoiceId : "spacingChoice", u = e != null && e.spacingCustomId ? e.spacingCustomId : "spacingCustom", r = e != null && e.defaultValueChoices ? e.defaultValueChoices : l.M;
  return {
    id: t,
    type: "switch",
    defaultValue: !1,
    switchLabel: "Custom",
    label: "Gutter",
    info: "An official nerds term for ‘gap’",
    onChange: (a) => C(a, c, u, h),
    show: (a) => {
      var d;
      return ((d = a.getBlock(g)) == null ? void 0 : d.value) !== "1";
    },
    on: [
      {
        id: u,
        type: "input",
        rules: [m],
        onChange: (a) => f(a, u)
      }
    ],
    off: [
      {
        id: c,
        type: "slider",
        defaultValue: r,
        choices: [
          {
            value: l.Auto,
            label: "Auto"
          },
          {
            value: l.S,
            label: "S"
          },
          {
            value: l.M,
            label: "M"
          },
          {
            value: l.L,
            label: "L"
          }
        ]
      }
    ]
  };
};
export {
  I as getGutterSettings
};
