// @frontify/guideline-blocks-settings/dist/settings/types.es.js
var l = /* @__PURE__ */ ((e) => (e.Solid = "Solid", e.Dashed = "Dashed", e.Dotted = "Dotted", e))(l || {});
const p = {
  Solid: "solid",
  Dotted: "dotted",
  Dashed: "dashed"
};
var o = /* @__PURE__ */ ((e) => (e.None = "None", e.Small = "Small", e.Medium = "Medium", e.Large = "Large", e))(o || {});
const x = {
  None: "0px",
  Small: "2px",
  Medium: "4px",
  Large: "12px"
};
var a = /* @__PURE__ */ ((e) => (e.None = "None", e.Small = "Small", e.Medium = "Medium", e.Large = "Large", e))(a || {});
const d = {
  None: "0px",
  Small: "24px",
  Medium: "36px",
  Large: "60px"
};
var t = /* @__PURE__ */ ((e) => (e.None = "None", e.Small = "Small", e.Medium = "Medium", e.Large = "Large", e))(t || {});
const u = {
  None: "0px",
  Small: "24px",
  Medium: "36px",
  Large: "60px"
};
var r = /* @__PURE__ */ ((e) => (e.Global = "Global", e.Custom = "Custom", e))(r || {}), m = /* @__PURE__ */ ((e) => (e.Auto = "Auto", e.S = "S", e.M = "M", e.L = "L", e))(m || {});
const S = {
  Auto: "4px",
  S: "10px",
  M: "30px",
  L: "50px"
};
export {
  l as BorderStyle,
  m as GutterSpacing,
  t as Margin,
  a as Padding,
  o as Radius,
  r as Security,
  p as borderStyleMap,
  S as gutterSpacingStyleMap,
  u as marginStyleMap,
  d as paddingStyleMap,
  x as radiusStyleMap
};
