// @frontify/fondue/dist/components/RichTextEditor/Plugins/helper.es.js
import e from "react";
const r = ({ icon: t }) => /* @__PURE__ */ e.createElement("span", { className: "tw-p-2 tw-h-8 tw-justify-center tw-items-center tw-flex" }, t), o = "tw-ml-0.5 !tw-text-text-weak hover:!tw-bg-box-selected hover:!tw-text-box-selected-inverse hover:tw-rounded", w = (t = !0) => ({
  root: t ? o : "!tw-text-text-weak !tw-cursor-not-allowed !tw-opacity-50",
  active: "!tw-bg-box-selected tw-rounded !tw-text-box-selected-inverse"
}), c = { root: { width: "24px", height: "24px" } }, i = ({ children: t, id: s }) => /* @__PURE__ */ e.createElement("div", { "data-plugin-id": s }, t), a = {
  left: "tw-text-left",
  right: "tw-text-right",
  center: "tw-text-center",
  justify: "tw-text-justify"
}, l = {
  left: "tw-justify-start tw-text-left",
  center: "tw-justify-center tw-text-center",
  justify: "tw-text-justify",
  right: "tw-justify-end tw-text-right"
}, x = "button";
export {
  i as ButtonWrapper,
  x as ELEMENT_BUTTON,
  r as IconStylingWrapper,
  a as alignmentClassnames,
  c as buttonStyles,
  w as getButtonClassNames,
  l as justifyClassNames
};
