// @frontify/fondue/dist/components/RichTextEditor/Plugins/index.es.js
import { BoldPlugin as r } from "./BoldPlugin/index.es.js";
import { CheckboxListPlugin as e } from "./CheckboxListPlugin/index.es.js";
import { createCheckboxListPlugin as _ } from "./CheckboxListPlugin/index.es.js";
import { CodePlugin as i } from "./CodePlugin/index.es.js";
import { EmojiPlugin as n } from "./EmojiPlugin/index.es.js";
import { SoftBreakPlugin as o } from "./SoftBreakPlugin/index.es.js";
import { ItalicPlugin as t } from "./ItalicPlugin/index.es.js";
import { LinkPlugin as m } from "./LinkPlugin/index.es.js";
import { createLinkPlugin as v } from "./LinkPlugin/index.es.js";
import { PluginComposer as g } from "./PluginComposer.es.js";
import { ResetFormattingPlugin as l } from "./ResetFormattingPlugin/index.es.js";
import { createResetFormattingPlugin as D } from "./ResetFormattingPlugin/index.es.js";
import { StrikethroughPlugin as P } from "./StrikethroughPlugin/index.es.js";
import { TextStylePlugin as u } from "./TextStylePlugin/index.es.js";
import { UnderlinePlugin as p } from "./UnderlinePlugin/index.es.js";
import { ParagraphPlugin as f } from "./TextStylePlugin/TextStyles/paragraphPlugin.es.js";
import { PARAGRAPH_CLASSES as Q, ParagraphMarkupElement as V, ParagraphMarkupElementNode as W, createParagraphPlugin as X } from "./TextStylePlugin/TextStyles/paragraphPlugin.es.js";
import { AlignLeftPlugin as a } from "./AlignPlugin/AlignLeftPlugin/index.es.js";
import { AlignCenterPlugin as w } from "./AlignPlugin/AlignCenterPlugin/index.es.js";
import { AlignRightPlugin as s } from "./AlignPlugin/AlignRightPlugin/index.es.js";
import { AlignJustifyPlugin as d } from "./AlignPlugin/AlignJustifyPlugin/index.es.js";
import { UnorderedListPlugin as h } from "./ListPlugin/UnorderedListPlugin/index.es.js";
import { OrderedListPlugin as x } from "./ListPlugin/OrderedListPlugin/index.es.js";
const c = new g();
c.setPlugin(new o()).setPlugin(new u({ textStyles: [new f()] })).setPlugin(
  [
    new r(),
    new t(),
    new p(),
    new P(),
    new m(),
    new i()
  ],
  [
    new a(),
    new w(),
    new s(),
    new d(),
    new h(),
    new e(),
    new x(),
    new l(),
    new n()
  ]
);
export {
  w as AlignCenterPlugin,
  d as AlignJustifyPlugin,
  a as AlignLeftPlugin,
  s as AlignRightPlugin,
  r as BoldPlugin,
  e as CheckboxListPlugin,
  i as CodePlugin,
  n as EmojiPlugin,
  t as ItalicPlugin,
  m as LinkPlugin,
  x as OrderedListPlugin,
  Q as PARAGRAPH_CLASSES,
  V as ParagraphMarkupElement,
  W as ParagraphMarkupElementNode,
  f as ParagraphPlugin,
  g as PluginComposer,
  l as ResetFormattingPlugin,
  o as SoftBreakPlugin,
  P as StrikethroughPlugin,
  u as TextStylePlugin,
  p as UnderlinePlugin,
  h as UnorderedListPlugin,
  _ as createCheckboxListPlugin,
  v as createLinkPlugin,
  X as createParagraphPlugin,
  D as createResetFormattingPlugin,
  c as defaultPlugins
};
