// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/RichTextEditor.es.js
import { jsx as s } from "react/jsx-runtime";
import { useState as p, useEffect as x } from "react";
import { RichTextEditor as E } from "@frontify/fondue";
import { SerializedText as g } from "./SerializedText.es.js";
import { floatingButtonSelectors as v, floatingButtonActions as T } from "./plugins/ButtonPlugin/components/FloatingButton/floatingButtonStore.es.js";
const R = ({
  id: d = "rte",
  isEditing: f,
  value: t,
  columns: l,
  gap: u,
  placeholder: c,
  plugins: r,
  onTextChange: o,
  showSerializedText: m
}) => {
  const [n, i] = p(!1), h = (e) => {
    o && e !== t && o(e), i(!1);
  };
  return x(() => {
    const e = (a) => (a.preventDefault(), a.returnValue = "Unprocessed changes");
    return n && window.addEventListener("beforeunload", e), () => window.removeEventListener("beforeunload", e);
  }, [n]), f ? /* @__PURE__ */ s(
    E,
    {
      id: d,
      value: t,
      border: !1,
      placeholder: c,
      plugins: r,
      onValueChanged: () => i(!0),
      onTextChange: h,
      hideExternalFloatingModals: (e) => {
        v.isOpen(e) && T.reset();
      }
    }
  ) : /* @__PURE__ */ s(g, { value: t, columns: l, gap: u, show: m, plugins: r });
};
export {
  R as RichTextEditor
};
