// @frontify/fondue/dist/components/Grid/Grid.es.js
import d from "react";
import { GAP_Y_MAPPING as t, GAP_X_MAPPING as a, SPACING_VALUES as l, PADDING_VALUES_MAP as n, MARGIN_VALUES_MAP as A } from "../../utilities/dimensions.es.js";
import { merge as E } from "../../utilities/merge.es.js";
import { Box as S } from "../Box/Box.es.js";
const D = "fondue-grid", L = ({
  column: r = 2,
  spacingX: s = 4,
  spacingY: m = 4,
  children: o,
  "data-test-id": c = D,
  width: N = "100%",
  rowHeight: _ = "auto",
  bg: p,
  color: G,
  margin: P = 0,
  padding: e = 0,
  as: u = "div"
}) => {
  const C = t[m] ? t[m] : t[4], I = a[s] ? a[s] : a[4], f = l.includes(e) ? n[e] : n[0], g = l.includes(e) ? A[P] : A[0];
  let i = 0;
  return d.Children.map(o, (M) => (i++, M)), /* @__PURE__ */ d.createElement(
    S,
    {
      "data-test-id": c,
      className: E([
        "tw-grid",
        p,
        G,
        f,
        g,
        I,
        C
      ]),
      style: {
        width: N,
        gridTemplateColumns: `repeat(${r}, 1fr)`,
        gridTemplateRows: `repeat(${Math.ceil(i / r)}, ${_})`
      },
      as: u
    },
    o
  );
};
L.displayName = "FondueGrid";
export {
  D as GRID_TEST_ID,
  L as Grid
};
