// @frontify/app-bridge/dist/react/useTemplateAssets.es.js
import { useState as d, useEffect as K } from "react";
import { compareObjects as F } from "../utilities/object.es.js";
const U = (a, t, r, o) => {
  const [m, n] = d({}), T = (e) => {
    e.template === t && e.documentId === r && e.documentPageId === o && !F(e.templateAssets, e.prevTemplateAssets) && n(e.templateAssets);
  }, A = async () => t === "cover" ? await a.getCoverPageTemplateAssets() : t === "documentPage" && o ? await a.getDocumentPageTemplateAssets(o) : t === "library" && r ? await a.getLibraryPageTemplateAssets(r) : {}, c = async (e, s) => {
    t === "cover" ? await a.addAssetIdsToCoverPageTemplateAssetKey(e, s) : t === "documentPage" && o ? await a.addAssetIdsToDocumentPageTemplateAssetKey(o, e, s) : t === "library" && r && await a.addAssetIdsToLibraryPageTemplateAssetKey(r, e, s);
  }, y = async (e, s) => {
    t === "cover" ? await a.deleteAssetIdsFromCoverPageTemplateAssetKey(e, s) : t === "documentPage" && o ? await a.deleteAssetIdsFromDocumentPageTemplateAssetKey(o, e, s) : t === "library" && r && await a.deleteAssetIdsFromLibraryPageTemplateAssetKey(r, e, s);
  };
  K(() => {
    let e = !0;
    return t && ((async () => {
      const w = await A();
      e && n(w);
    })(), window.emitter.on("AppBridge:TemplateAssetsUpdated", T)), () => {
      e = !1, window.emitter.off("AppBridge:TemplateAssetsUpdated", T);
    };
  }, [a, r, o]);
  const i = async () => {
    window.emitter.emit("AppBridge:TemplateAssetsUpdated", {
      template: t,
      documentId: r,
      documentPageId: o,
      templateAssets: await A(),
      prevTemplateAssets: { ...m }
    });
  };
  return {
    templateAssets: m,
    addAssetIdsToKey: async (e, s) => {
      await c(e, s), i();
    },
    deleteAssetIdsFromKey: async (e, s) => {
      await y(e, s), i();
    },
    updateAssetIdsFromKey: async (e, s) => {
      var p;
      const f = ((p = (await A())[e]) == null ? void 0 : p.map((l) => l.id)) ?? [];
      try {
        await y(e, f), await c(e, s);
      } catch (l) {
        console.error(l);
      }
      i();
    }
  };
};
export {
  U as useTemplateAssets
};
