// @frontify/fondue/dist/components/LoadingCircle/LoadingCircle.es.js
import { merge as a } from "../../utilities/merge.es.js";
import s from "react";
var o = /* @__PURE__ */ ((r) => (r.Progress = "Progress", r.Positive = "Positive", r.Danger = "Danger", r))(o || {}), w = /* @__PURE__ */ ((r) => (r.ExtraSmall = "ExtraSmall", r.Small = "Small", r.Medium = "Medium", r.Large = "Large", r))(w || {});
const d = {
  Progress: "tw-border-violet-60",
  Positive: "tw-border-green-60",
  Danger: "tw-border-red-60"
}, m = {
  ExtraSmall: "tw-w-4 tw-h-4",
  Small: "tw-w-5 tw-h-5",
  Medium: "tw-w-8 tw-h-8",
  Large: "tw-w-16 tw-h-16"
}, l = ({
  style: r = "Progress",
  size: t = "Medium",
  "data-test-id": e = "loading-circle"
}) => /* @__PURE__ */ s.createElement(
  "div",
  {
    "data-test-id": e,
    className: a([
      "tw-border-2 tw-border-solid tw-rounded-full tw-border-t-transparent tw-animate-spin",
      d[r],
      m[t]
    ])
  }
);
l.displayName = "FondueLoadingCircle";
export {
  l as LoadingCircle,
  w as LoadingCircleSize,
  o as LoadingCircleStyle,
  m as sizeClasses,
  d as statusClasses
};
