// @frontify/fondue/dist/components/DatePicker/DatePickerTrigger.es.js
import { merge as s } from "../../utilities/merge.es.js";
import { Validation as e } from "../../utilities/validation.es.js";
import t, { forwardRef as h } from "react";
import S from "../../foundation/Icon/Generated/IconCaretUp.es.js";
import z from "../../foundation/Icon/Generated/IconCaretDown.es.js";
import { TextInput as F } from "../TextInput/TextInput.es.js";
import R from "../../foundation/Icon/Generated/IconCalendar.es.js";
import { IconSize as l } from "../../foundation/Icon/IconSize.es.js";
const I = h(
  ({ value: o, onClick: a, isClearable: i, placeHolder: m, isCalendarOpen: f, validation: r = e.Default, onDateChanged: n }, d) => {
    const w = r === e.Error || r === e.Warning, g = "tw-right-14", p = "tw-right-8", E = "tw-right-4", c = ((u) => {
      switch (u) {
        case 0:
          return E;
        case 1:
          return p;
        case 2:
          return g;
      }
      return "";
    })(
      (o && i ? 1 : 0) + (w ? 1 : 0)
    );
    return /* @__PURE__ */ t.createElement("div", { ref: d, "aria-hidden": "true" }, /* @__PURE__ */ t.createElement(
      "div",
      {
        className: s([
          "tw-absolute tw-top-2 tw-text-black-60 tw-z-[1]",
          c
        ])
      },
      f ? /* @__PURE__ */ t.createElement(S, { size: l.Size20 }) : /* @__PURE__ */ t.createElement(z, { size: l.Size20 })
    ), /* @__PURE__ */ t.createElement(
      "div",
      {
        className: s([
          "tw-absolute tw-top-0 tw-left-0 tw-bottom-0 tw-z-10",
          c
        ]),
        onClick: a,
        "data-test-id": "open-close-click-zone"
      }
    ), /* @__PURE__ */ t.createElement(
      F,
      {
        decorator: /* @__PURE__ */ t.createElement(R, null),
        placeholder: m,
        value: o,
        clearable: i,
        onEnterPressed: a,
        onClear: () => {
          n && n(null);
        },
        validation: r
      }
    ));
  }
);
I.displayName = "FondueDatePickerTrigger";
export {
  I as DatePickerTrigger
};
