// @frontify/app-bridge/dist/tests/ColorApiDummy.es.js
class e {
  static red(l = 767) {
    return {
      id: l,
      creator: 1,
      created: "2022-06-10T12:35:52.000+00:00",
      modifier: null,
      modified: null,
      valid_from: null,
      valid_to: null,
      project: 1,
      reference_color: null,
      group: null,
      palette: null,
      type: null,
      name: "Red",
      name_css: "red",
      sort: 1,
      r: 255,
      g: 0,
      b: 0,
      hex: "ff0000",
      alpha: 255,
      hue: 361,
      saturation: 100,
      lightness: 46,
      gradient_direction: null,
      gradient_color_one: null,
      gradient_color_two: null,
      gradient_type: null,
      usage: "USAGE",
      priority: null,
      c: 0,
      m: 100,
      y: 100,
      k: 0,
      pantone: null,
      ral: null,
      oracal: null,
      pantone_coated: null,
      pantone_uncoated: null,
      cmyk_coated: null,
      cmyk_uncoated: null,
      cmyk_newspaper: null,
      ncs: null,
      pantone_cp: null,
      pantone_plastics: null,
      pantone_textile: null,
      hks: null,
      three_m: null,
      lab: null,
      opacity: null,
      opacity_css: null,
      css_value: null
    };
  }
  static yellow(l = 768) {
    return {
      id: l,
      creator: 1,
      created: "2022-06-10T12:35:52.000+00:00",
      modifier: null,
      modified: null,
      valid_from: null,
      valid_to: null,
      project: 1,
      reference_color: null,
      group: null,
      palette: null,
      type: null,
      name: "Yellow",
      name_css: "yellow",
      sort: 2,
      hex: "ffff00",
      r: 255,
      g: 255,
      b: 0,
      alpha: 255,
      hue: 361,
      saturation: 100,
      lightness: 46,
      gradient_direction: null,
      gradient_color_one: null,
      gradient_color_two: null,
      gradient_type: null,
      usage: "USAGE",
      priority: null,
      c: 0,
      m: 0,
      y: 100,
      k: 0,
      pantone: null,
      ral: null,
      oracal: null,
      pantone_coated: null,
      pantone_uncoated: null,
      cmyk_coated: null,
      cmyk_uncoated: null,
      cmyk_newspaper: null,
      ncs: null,
      pantone_cp: null,
      pantone_plastics: null,
      pantone_textile: null,
      hks: null,
      three_m: null,
      lab: null,
      opacity: null,
      opacity_css: null,
      css_value: null
    };
  }
  static green(l = 769) {
    return {
      id: l,
      creator: 1,
      created: "2022-06-10T12:35:52.000+00:00",
      modifier: null,
      modified: null,
      valid_from: null,
      valid_to: null,
      project: 1,
      reference_color: null,
      group: null,
      palette: null,
      type: null,
      name: "Green",
      name_css: "green",
      sort: 3,
      hex: "00ff00",
      r: 0,
      g: 255,
      b: 0,
      alpha: 255,
      hue: 361,
      saturation: 100,
      lightness: 46,
      gradient_direction: null,
      gradient_color_one: null,
      gradient_color_two: null,
      gradient_type: null,
      usage: "USAGE",
      priority: null,
      c: 100,
      m: 0,
      y: 100,
      k: 0,
      pantone: null,
      ral: null,
      oracal: null,
      pantone_coated: null,
      pantone_uncoated: null,
      cmyk_coated: null,
      cmyk_uncoated: null,
      cmyk_newspaper: null,
      ncs: null,
      pantone_cp: null,
      pantone_plastics: null,
      pantone_textile: null,
      hks: null,
      three_m: null,
      lab: null,
      opacity: null,
      opacity_css: null,
      css_value: null
    };
  }
}
class a {
  static red() {
    return {
      project_color_palette_id: 23,
      red: 255,
      green: 0,
      blue: 0,
      alpha: 255
    };
  }
}
class u {
  static red() {
    return {
      name: "Red",
      sort: 1,
      name_css: "red",
      red: 255,
      green: 0,
      blue: 0,
      alpha: 255,
      c: 0,
      m: 100,
      y: 100,
      k: 0,
      pantone: null,
      ral: null,
      oracal: null,
      pantone_coated: null,
      pantone_uncoated: null,
      cmyk_coated: null,
      cmyk_uncoated: null,
      cmyk_newspaper: null,
      ncs: null,
      pantone_cp: null,
      pantone_plastics: null,
      pantone_textile: null,
      hks: null,
      three_m: null,
      lab: null
    };
  }
  static yellow() {
    return {
      name: "Yellow",
      sort: 2,
      name_css: "yellow",
      red: 255,
      green: 255,
      blue: 0,
      alpha: 255,
      c: 0,
      m: 0,
      y: 100,
      k: 0,
      pantone: null,
      ral: null,
      oracal: null,
      pantone_coated: null,
      pantone_uncoated: null,
      cmyk_coated: null,
      cmyk_uncoated: null,
      cmyk_newspaper: null,
      ncs: null,
      pantone_cp: null,
      pantone_plastics: null,
      pantone_textile: null,
      hks: null,
      three_m: null,
      lab: null
    };
  }
  static green() {
    return {
      name: "Green",
      sort: 3,
      name_css: "green",
      red: 0,
      green: 255,
      blue: 0,
      alpha: 255,
      c: 100,
      m: 0,
      y: 100,
      k: 0,
      pantone: null,
      ral: null,
      oracal: null,
      pantone_coated: null,
      pantone_uncoated: null,
      cmyk_coated: null,
      cmyk_uncoated: null,
      cmyk_newspaper: null,
      ncs: null,
      pantone_cp: null,
      pantone_plastics: null,
      pantone_textile: null,
      hks: null,
      three_m: null,
      lab: null
    };
  }
}
export {
  a as ColorApiCreateDummy,
  e as ColorApiDummy,
  u as ColorApiPatchDummy
};
