// @frontify/app-bridge/dist/tests/TargetsApiDummy.es.js
class e {
  static target(t) {
    return {
      checked: !1,
      disabled: !1,
      indeterminate: !1,
      label: String(t),
      target: {
        account_id: 1,
        asset_ids: [],
        created: "2022-06-10T12:35:52.000+00:00",
        creator: 1,
        description: "Target description",
        group_ids: [],
        id: t,
        modified: null,
        modifier: null,
        name: `Target ${t}`,
        sort: 1,
        total_groups: null,
        total_links: null,
        total_users: null,
        user_ids: []
      },
      value: t
    };
  }
}
class a {
  static with(t) {
    return {
      appearance: {},
      background_file_id: null,
      background_url: null,
      heading: null,
      id: t,
      link_type: null,
      link_url: null,
      logo_file_id: null,
      logo_url: null,
      mode: null,
      portal_title: "Portal title",
      subheading: null,
      success: !0,
      targets: {
        disabled: [],
        has_selected_targets: !0,
        targets: [
          e.target(1),
          e.target(2),
          e.target(3)
        ]
      },
      title: "Test document"
    };
  }
}
class r {
  static with(t) {
    return {
      success: !0,
      data: !0,
      targets: t
    };
  }
}
class s {
  static with(t) {
    return {
      default: !1,
      disabled: [],
      has_selected_targets: !0,
      id: t,
      success: !0,
      targets: [e.target(1), e.target(2), e.target(3)]
    };
  }
}
export {
  s as DocumentPageTargetsApiDummy,
  a as DocumentTargetsApiDummy,
  e as SingleTargetApiDummy,
  r as UpdateTargetsApiDummy
};
