// @frontify/fondue/dist/components/LinkChooser/LinkChooser.es.js
import { mapToAriaProps as de } from "../ActionMenu/Aria/helper.es.js";
import { Checkbox as pe, CheckboxState as A } from "../Checkbox/Checkbox.es.js";
import { useDropdownAutoHeight as fe } from "../../hooks/useDropdownAutoHeight.es.js";
import { useComboBox as he } from "@react-aria/combobox";
import { DismissButton as F } from "@react-aria/overlays";
import { scrollIntoView as Ee } from "@react-aria/utils";
import { useComboBoxState as we } from "@react-stately/combobox";
import { Validation as Ce } from "../../utilities/validation.es.js";
import { useMachine as Se } from "@xstate/react";
import { AnimatePresence as ye, motion as Re } from "framer-motion";
import t, { useState as y, useMemo as ge, useRef as h, useCallback as q, useEffect as R } from "react";
import { NavigationMenu as Ie } from "./NavigationMenu.es.js";
import { Popover as ke } from "./Popover.es.js";
import { SearchInput as De } from "./SearchInput.es.js";
import { SearchResultsList as be } from "./SearchResultsList.es.js";
import { defaultSection as Be } from "./sections.es.js";
import { linkChooserMachine as Oe } from "./state/machine.es.js";
import { LinkChooserState as E } from "./state/types.es.js";
import { IconName as i } from "./types.es.js";
import { findSection as W, decoratedResults as xe, doesContainSubstring as ve, getDefaultData as Le } from "./utils/helpers.es.js";
import { shouldGoBack as Ne, isLoaded as _e, closeBoxState as K, queryMatchesSelection as Me, openBoxState as Te } from "./utils/state.es.js";
import { createCustomLink as Pe } from "./utils/transformers.es.js";
import { useManualComboBoxEventHandlers as Ae } from "./utils/useManualComboBoxHandlers.es.js";
import Fe from "../../foundation/Icon/Generated/IconDocument.es.js";
import qe from "../../foundation/Icon/Generated/IconDocumentStack.es.js";
import We from "../../foundation/Icon/Generated/IconLink.es.js";
import Ke from "../../foundation/Icon/Generated/IconArrowOutExternal.es.js";
import Ue from "../../foundation/Icon/Generated/IconLayersSingle.es.js";
import $e from "../../foundation/Icon/Generated/IconBuildingBlock.es.js";
const Ge = {
  [i.Document]: /* @__PURE__ */ t.createElement(Fe, null),
  [i.Library]: /* @__PURE__ */ t.createElement(qe, null),
  [i.Link]: /* @__PURE__ */ t.createElement(We, null),
  [i.External]: /* @__PURE__ */ t.createElement(Ke, null),
  [i.Template]: /* @__PURE__ */ t.createElement(Ue, null),
  [i.Block]: /* @__PURE__ */ t.createElement($e, null)
}, Ve = i.Link, Rt = "custom-link", gt = 5, It = "queries", kt = ({
  getGlobalByQuery: U = Le,
  openPreview: $ = window.open,
  clipboardOptions: G = navigator.clipboard,
  selectedResult: d = null,
  openInNewTab: V = !1,
  ariaLabel: H = "Menu",
  extraSections: c = [],
  label: g,
  placeholder: Q,
  onOpenInNewTabChange: z,
  onLinkChange: Y,
  disabled: I,
  clearable: X,
  required: k,
  validation: j = Ce.Default,
  "data-test-id": u = "link-chooser"
}) => {
  var T, P;
  const [{ context: o, matches: a, value: J }, n, w] = Se(
    () => Oe.withContext({
      searchResults: [],
      selectedResult: d,
      query: "",
      interruptedFetch: !1,
      getExtraResultsByQuery: null,
      currentSectionId: Be.id,
      extraSections: c,
      clipboardOptions: G,
      openPreview: $,
      getGlobalByQuery: U,
      onLinkChange: Y
    })
  ), [Z, D] = y(d == null ? void 0 : d.title), p = !Ne(a), b = ge(
    () => [
      !p && { id: "menu-top", menuItems: [W(c, o.currentSectionId)] },
      {
        id: "search",
        menuItems: xe(o.searchResults)
      },
      p && { id: "menu-bottom", menuItems: c.map(({ id: e, title: l }) => ({ id: e, title: l })) }
    ].filter(Boolean),
    [p, c, o.currentSectionId, o.searchResults]
  ), B = de(H, b), O = h(null), s = h(null), x = h(null), m = h(null), v = (e) => {
    const l = o.searchResults.find((me) => me.id === e);
    l && (D(l.title), n({ type: "SET_SELECTED_SEARCH_RESULT", data: { selectedResult: l } })), K(r), C(e);
  }, ee = q(
    (e) => {
      D(e), n({ type: "TYPING", data: { query: e } });
    },
    [n]
  ), [L, C] = y((T = o.selectedResult) == null ? void 0 : T.id), r = we({
    ...B,
    defaultFilter: (e, l) => ve(e, l, c),
    inputValue: Z,
    onInputChange: ee,
    onSelectionChange: v,
    menuTrigger: "manual",
    shouldCloseOnBlur: !1,
    allowsEmptyCollection: !0,
    selectedKey: L
  }), { inputProps: te, listBoxProps: oe, labelProps: ne } = he(
    {
      ...B,
      inputRef: s,
      listBoxRef: x,
      popoverRef: m,
      isRequired: k,
      placeholder: Q
    },
    r
  ), re = q(() => {
    r.setInputValue(""), C(""), n({ type: "CLEARING", data: { query: "" } });
  }, [n, r]), N = () => {
    n("OPEN_DROPDOWN"), Te(r);
  }, le = () => {
    var e;
    document.activeElement !== s.current && ((e = s.current) == null || e.focus()), N();
  }, ae = (e) => {
    a(E.Focused) && e.target !== s.current && e.preventDefault();
  }, f = () => {
    let e = null;
    o.query && (Me(o.selectedResult, o.query) ? e = o.selectedResult : e = Pe(r.inputValue)), n({ type: "CLOSE_DROPDOWN", data: { selectedResult: e } }), e && L !== e.id && C(e.id), K(r);
  }, se = Ae(
    { inputProps: te, inputRef: s, popoverRef: m, state: r },
    {
      onOpen: N,
      onClose: f,
      onNavigate: (e) => {
        var l;
        n(p ? {
          type: "SELECT_EXTRA_SECTION",
          data: {
            getExtraResultsByQuery: ((l = W(c, e)) == null ? void 0 : l.getResults) || null,
            currentSectionId: e.toString()
          }
        } : {
          type: "BACK_TO_DEFAULT",
          data: { getExtraResultsByQuery: null }
        });
      },
      onSelect: v
    }
  );
  R(() => {
    _e(a) && o.interruptedFetch && n({ type: "TYPING", data: { query: o.query } });
  }, [o.interruptedFetch, o.query, a, n, J]);
  const { maxHeight: ie } = fe(s, {
    isOpen: a(E.Focused),
    autoResize: !0
  }), {
    isOpen: _,
    selectionManager: { focusedKey: S }
  } = r;
  R(() => {
    if (S && m.current && _) {
      const e = m.current.querySelector(`[data-key="${S}"]`);
      e && Ee(m.current, e);
    }
  }, [S, _]);
  const [ce, ue] = y(0), M = () => {
    var e;
    ue(((e = O.current) == null ? void 0 : e.getBoundingClientRect().width) || 0);
  };
  return R(() => {
    M(), window.addEventListener("resize", M, !1);
  }, []), /* @__PURE__ */ t.createElement("div", { "data-test-id": u, ref: O, className: "tw-w-full tw-font-sans tw-text-s" }, !!g && /* @__PURE__ */ t.createElement(
    "label",
    {
      ...ne,
      "data-test-id": `${u}-label`,
      className: "tw-text-black-80 tw-mb-1 tw-flex tw-align-items-center"
    },
    g,
    k && /* @__PURE__ */ t.createElement(
      "span",
      {
        "data-test-id": `${u}-label-required`,
        className: "tw-h-4 tw-text-m tw-text-red-60 dark:tw-text-red-50 tw-ml-1"
      },
      "*"
    )
  ), /* @__PURE__ */ t.createElement(
    De,
    {
      ariaProps: se,
      selectedResult: o.selectedResult,
      ref: s,
      disabled: I,
      decorator: Ge[((P = o.selectedResult) == null ? void 0 : P.icon) || Ve],
      clearable: X,
      onClear: re,
      machineService: w,
      validation: j,
      onClick: le,
      onMouseDown: ae
    }
  ), /* @__PURE__ */ t.createElement(ye, null, a(E.Focused) && /* @__PURE__ */ t.createElement(
    Re.div,
    {
      style: {
        width: ce
      },
      className: "tw-absolute tw-left-auto tw-w-full tw-overflow-hidden tw-p-0 tw-shadow-mid tw-list-none tw-m-0 tw-mt-2 tw-z-20",
      key: "content",
      initial: { height: 0 },
      animate: { height: "auto" },
      exit: { height: 0 },
      transition: { ease: [0.04, 0.62, 0.23, 0.98], duration: 0.5 },
      "data-test-id": `${u}-dropdown`
    },
    /* @__PURE__ */ t.createElement(F, { onDismiss: f }),
    /* @__PURE__ */ t.createElement(
      ke,
      {
        popoverRef: m,
        isOpen: a(E.Focused),
        onClose: f,
        maxHeight: ie
      },
      /* @__PURE__ */ t.createElement(
        be,
        {
          ...oe,
          listBoxRef: x,
          state: r,
          menuBlocks: b,
          border: !1,
          machineService: w
        }
      ),
      /* @__PURE__ */ t.createElement("div", { "data-test-id": `${u}-action-menu`, className: "tw-border-t tw-border-black-10" }, /* @__PURE__ */ t.createElement(Ie, { machineService: w, state: r }))
    ),
    /* @__PURE__ */ t.createElement(F, { onDismiss: f })
  )), /* @__PURE__ */ t.createElement("div", { className: "tw-my-2", "data-test-id": `${u}-new-tab` }, /* @__PURE__ */ t.createElement(
    pe,
    {
      value: "new-tab",
      disabled: I,
      state: V ? A.Checked : A.Unchecked,
      onChange: z,
      label: "Open in New Tab"
    }
  )));
};
export {
  Rt as CUSTOM_LINK_ID,
  Ve as DEFAULT_ICON,
  Ge as IconOptions,
  kt as LinkChooser,
  gt as MAX_STORED_ITEMS,
  It as QUERIES_STORAGE_KEY
};
