// @frontify/fondue/dist/components/LinkChooser/NavigationMenu.es.js
import { getInteractionModality as l } from "@react-aria/interactions";
import { useOption as u } from "@react-aria/listbox";
import { merge as p } from "../../utilities/merge.es.js";
import { useActor as f } from "@xstate/react";
import o, { useRef as d, useMemo as g } from "react";
import { defaultSection as E } from "./sections.es.js";
import { LinkChooserState as I, DropdownState as S } from "./state/types.es.js";
import { SelectionIndicatorIcon as h, MenuItemContentSize as k } from "../MenuItem/types.es.js";
import v from "../../foundation/Icon/Generated/IconArrowLeft.es.js";
import { MenuItem as b } from "../MenuItem/MenuItem.es.js";
const L = ({ machineService: e, state: r }) => {
  const [{ matches: n, context: i }, a] = f(e);
  return n(`${I.Focused}.${S.Default}`) ? /* @__PURE__ */ o.createElement("ul", { className: "tw-py-2" }, i.extraSections.map((t) => /* @__PURE__ */ o.createElement(
    w,
    {
      key: t.id,
      section: t,
      state: r,
      onPress: () => {
        a({
          type: "SELECT_EXTRA_SECTION",
          data: { getExtraResultsByQuery: t.getResults, currentSectionId: t.id }
        });
      }
    }
  ))) : null;
}, w = ({
  section: { id: e, title: r },
  onPress: n,
  state: i,
  direction: a = "right"
}) => {
  const t = d(null), { isFocused: m } = u(
    {
      key: e,
      shouldFocusOnHover: !1
    },
    i,
    t
  ), s = g(
    () => e === E.id ? /* @__PURE__ */ o.createElement("p", { className: "tw-ml-1 tw-text-black-80 tw-capitalize" }, r) : r,
    [e, r]
  ), c = l() !== "pointer";
  return /* @__PURE__ */ o.createElement(
    "div",
    {
      onClick: n,
      tabIndex: 0,
      role: "menuitem",
      "aria-label": `Navigate to ${e} section of search results.`,
      "data-key": e,
      "data-test-id": "link-chooser-navigation-menu-item",
      className: p(["hover:tw-bg-black-10", m && c && "tw-bg-black-10"])
    },
    /* @__PURE__ */ o.createElement(
      b,
      {
        title: s,
        selectionIndicator: a === "right" ? h.CaretRight : void 0,
        decorator: a === "left" ? /* @__PURE__ */ o.createElement(v, null) : void 0,
        size: k.XSmall
      }
    )
  );
};
export {
  L as NavigationMenu,
  w as NavigationMenuItem
};
