// @frontify/fondue/dist/components/LinkChooser/SearchResultsList.es.js
import { getMenuItems as g, getKeyItemRecord as y } from "../ActionMenu/Aria/helper.es.js";
import { useListBox as k } from "@react-aria/listbox";
import { useActor as N } from "@xstate/react";
import e, { useRef as R, useMemo as v } from "react";
import I from "./assets/background.svg.es.js";
import u from "./assets/no-results.svg.es.js";
import S from "./assets/nook-animated.png.es.js";
import { NavigationMenuItem as b } from "./NavigationMenu.es.js";
import { SearchResultSection as B } from "./SearchResultSection.es.js";
import { defaultSection as F } from "./sections.es.js";
import { findSection as A } from "./utils/helpers.es.js";
import { isFetching as L, isUnsuccessful as U, shouldGoBack as j } from "./utils/state.es.js";
const J = (t) => {
  const s = R(null), { listBoxRef: l = s, state: r, menuBlocks: w, machineService: i } = t, { listBoxProps: f } = k(t, r, l), d = g(w), p = y(d), [
    {
      context: { extraSections: a, currentSectionId: m, searchResults: h, query: o },
      matches: n
    },
    x
  ] = N(i), E = v(
    () => A(a, m) || F,
    [m, a]
  );
  return L(n) ? /* @__PURE__ */ e.createElement(P, null) : U(n) ? /* @__PURE__ */ e.createElement(K, null) : /* @__PURE__ */ e.createElement("div", null, j(n) && /* @__PURE__ */ e.createElement("div", { className: "tw-mt-2" }, /* @__PURE__ */ e.createElement(
    b,
    {
      state: r,
      section: E,
      onPress: () => {
        x({
          type: "BACK_TO_DEFAULT",
          data: { getExtraResultsByQuery: null }
        });
      },
      direction: "left"
    }
  )), /* @__PURE__ */ e.createElement(
    "ul",
    {
      ...f,
      "data-test-id": "link-chooser-results-list",
      ref: l,
      className: "tw-list-none tw-p-0 tw-m-0 tw-bg-white tw-z-20 focus-visible:tw-outline-none"
    },
    h.length > 0 ? [...r.collection].filter((c) => c.key === "search").map((c) => /* @__PURE__ */ e.createElement(
      B,
      {
        key: c.key,
        heading: c,
        state: r,
        keyItemRecord: p,
        machineService: i
      }
    )) : /* @__PURE__ */ e.createElement(
      M,
      {
        prompt: o ? `We could not find any results for "${o}".` : "Use the search above to discover your brand assets, or enter a URL",
        icon: o ? u : I
      }
    )
  ));
}, M = ({ prompt: t, icon: s }) => /* @__PURE__ */ e.createElement(
  "div",
  {
    "data-test-id": "link-chooser-empty-results",
    className: "tw-flex tw-flex-col tw-justify-center tw-items-center tw-h-[278px] tw-p-3"
  },
  /* @__PURE__ */ e.createElement("img", { className: "tw-w-[150px] tw-mb-5", src: s, alt: "Icon" }),
  /* @__PURE__ */ e.createElement("p", { className: "tw-text-black-60 tw-text-center" }, t)
), K = ({
  error: t = "An error occurred while fetching the results"
}) => /* @__PURE__ */ e.createElement(
  "div",
  {
    "data-test-id": "link-chooser-error",
    className: "tw-flex tw-flex-col tw-justify-center tw-items-center tw-h-[278px] tw-p-3"
  },
  /* @__PURE__ */ e.createElement("img", { className: "tw-w-[75px] tw-mb-5", src: u, alt: "Error" }),
  /* @__PURE__ */ e.createElement("p", { className: "tw-text-black-60 tw-text-center" }, t)
), P = () => /* @__PURE__ */ e.createElement(
  "div",
  {
    "data-test-id": "link-chooser-loader",
    className: "tw-flex tw-flex-col tw-justify-center tw-items-center tw-h-[278px] tw-p-3"
  },
  /* @__PURE__ */ e.createElement("img", { className: "tw-w-[50px]", src: S, alt: "Fetching" })
);
export {
  J as SearchResultsList
};
