// @frontify/fondue/dist/components/LinkChooser/SearchResultOption.es.js
import { MenuItem as h } from "../MenuItem/MenuItem.es.js";
import { getInteractionModality as S } from "@react-aria/interactions";
import { useOption as I } from "@react-aria/listbox";
import { merge as N } from "../../utilities/merge.es.js";
import { useActor as y } from "@xstate/react";
import e, { useRef as F } from "react";
import { IconOptions as L } from "./LinkChooser.es.js";
import { DropdownState as m, LinkChooserState as M, SectionState as O } from "./state/types.es.js";
import { findSection as D } from "./utils/helpers.es.js";
import { MenuItemContentSize as R } from "../MenuItem/types.es.js";
const J = ({
  item: t,
  state: n,
  keyItemRecord: r,
  machineService: w
}) => {
  const a = F(null), { optionProps: p, isDisabled: d, isSelected: f, isFocused: x } = I(
    {
      key: t.key,
      shouldFocusOnHover: !1
    },
    n,
    a
  ), [
    {
      context: { extraSections: u, currentSectionId: b },
      matches: k
    }
  ] = y(w), c = r[t.key], E = c.icon ? L[c.icon] : void 0, o = D(u, b), i = (s) => k(`${M.Focused}.${s}.${O.Loaded}`), g = S() !== "pointer", v = (s) => {
    var l;
    return ((l = o == null ? void 0 : o.renderPreview) == null ? void 0 : l.call(o, s)) || /* @__PURE__ */ e.createElement($, { ...s });
  };
  return /* @__PURE__ */ e.createElement(
    "li",
    {
      ...p,
      ref: a,
      className: N([
        "tw-relative hover:tw-bg-black-10 tw-list-none tw-outline-none",
        d && "tw-pointer-events-none tw-top-px",
        x && g && "tw-bg-black-10"
      ])
    },
    i(m.Default) && /* @__PURE__ */ e.createElement(h, { ...c, active: f, decorator: E, size: R.Large }),
    i(m.ExtraSection) && v(c)
  );
}, $ = ({ title: t, subtitle: n, preview: r }) => /* @__PURE__ */ e.createElement("div", { className: "tw-flex tw-px-5 tw-py-1.5 tw-cursor-pointer" }, /* @__PURE__ */ e.createElement("div", { className: "tw-flex tw-flex-shrink-0 tw-w-[75px] tw-h-[75px] tw-max-w-xs tw-bg-black-0 tw-rounded" }, r && /* @__PURE__ */ e.createElement("img", { className: "tw-w-full tw-object-contain", src: r, alt: t })), /* @__PURE__ */ e.createElement("div", { className: "tw-flex tw-flex-col tw-justify-center tw-ml-4" }, /* @__PURE__ */ e.createElement("p", { className: "tw-block tw-text-md tw-leading-tight tw-hover:underline tw-text-black-80" }, t), n && /* @__PURE__ */ e.createElement("p", { className: "tw-text-[12px] tw-text-black-80" }, n)));
export {
  J as SearchResultOption
};
