// @frontify/guideline-blocks-settings/dist/components/BlockInjectButton/BlockInjectButton.es.js
import { jsxs as c, jsx as e, Fragment as z } from "react/jsx-runtime";
import { LoadingCircle as O, IconExclamationMarkTriangle as X, Flyout as A, ActionMenu as F, MenuItemContentSize as I, IconArrowCircleUp20 as R, IconImageStack20 as T } from "@frontify/fondue";
import { useState as g, useRef as V } from "react";
import { joinClassNames as P } from "../../utilities/react/joinClassNames.es.js";
import { FileExtensionSets as Y } from "@frontify/app-bridge";
const H = ({
  onDrop: s,
  label: u,
  icon: p,
  secondaryLabel: f,
  isLoading: l,
  fillParentContainer: N,
  onAssetChooseClick: x,
  onUploadClick: k,
  withMenu: j = !0,
  onClick: m,
  validFileType: v,
  verticalLayout: M
}) => {
  const [d, b] = g(!1), [i, w] = g(), h = V(null), [o, a] = g(void 0), S = (t) => {
    if (t.preventDefault(), b(!1), !B(t.dataTransfer.files)) {
      a("Invalid"), setTimeout(() => {
        a(void 0);
      }, 1e3);
      return;
    }
    s == null || s(t.dataTransfer.files);
  }, B = (t) => {
    if (!v)
      return !0;
    for (let r = 0; r < t.length; r++) {
      const n = t[r].name.split(".").pop() ?? "";
      if (!Y[v].includes(n))
        return !1;
    }
    return !0;
  }, C = (t) => {
    if (!h.current || l)
      return;
    const { left: r, top: n } = h.current.getBoundingClientRect(), E = t.clientX - r, y = t.clientY - n;
    w([E, y]);
  };
  return /* @__PURE__ */ c(
    "button",
    {
      ref: h,
      "data-test-id": "block-inject-button",
      className: P([
        "tw-font-body tw-relative tw-text-sm tw-leading-4 tw-border tw-flex tw-items-center tw-justify-center tw-cursor-pointer tw-gap-3 tw-w-full first:tw-rounded-tl last:tw-rounded-br",
        M ? "[&:not(:first-child)]:tw-border-t-0 first:tw-rounded-tr last:tw-rounded-bl" : "[&:not(:first-child)]:tw-border-l-0  first:tw-rounded-bl last:tw-rounded-tr",
        N ? "tw-h-full" : "tw-h-[72px]",
        d && !l ? "tw-border-dashed" : "tw-border-solid",
        i && "tw-bg-blank-state-pressed-inverse",
        d && "tw-bg-blank-state-weak-inverse",
        o ? "!tw-border-red-50 !tw-cursor-not-allowed" : " tw-border-blank-state-line",
        l || i || d || o ? "" : "tw-text-text-weak hover:tw-text-blank-state-hover hover:tw-bg-blank-state-hover-inverse hover:tw-border-blank-state-line-hover active:tw-text-blank-state-pressed active:tw-bg-blank-state-pressed-inverse active:tw-border-blank-state-line-hover",
        (d || i) && !o ? "[&>*]:tw-pointer-events-none tw-border-blank-state-line-hover" : "tw-bg-blank-state-shaded-inverse tw-text-blank-state-shaded"
      ]),
      onDragEnter: s ? (t) => {
        var r;
        if (b(!0), v === "Images")
          for (const n of Array.from(t.dataTransfer.items))
            (r = n == null ? void 0 : n.type) != null && r.startsWith("image/") ? a(void 0) : a("Invalid");
      } : void 0,
      onDragLeave: s ? () => {
        b(!1), a(void 0);
      } : void 0,
      onDrop: s ? S : void 0,
      onClick: (t) => {
        j && C(t), m == null || m();
      },
      children: [
        l ? /* @__PURE__ */ e(O, {}) : (
          // eslint-disable-next-line unicorn/no-nested-ternary
          o ? /* @__PURE__ */ c("div", { className: " tw-flex tw-items-center tw-justify-center tw-text-red-60 tw-font-medium", children: [
            /* @__PURE__ */ e(X, {}),
            o
          ] }) : /* @__PURE__ */ c(z, { children: [
            p && /* @__PURE__ */ e("div", { children: p }),
            (u || f) && /* @__PURE__ */ c("div", { className: "tw-flex tw-flex-col tw-items-start", children: [
              u && /* @__PURE__ */ e("div", { className: "tw-font-medium", children: u }),
              f && /* @__PURE__ */ e("div", { className: "tw-font-normal", children: f })
            ] })
          ] })
        ),
        i && /* @__PURE__ */ e(
          "div",
          {
            className: "tw-absolute tw-left-0 tw-top-full tw-z-20",
            style: {
              left: i[0],
              top: i[1]
            },
            children: /* @__PURE__ */ e(
              A,
              {
                onOpenChange: (t) => !t && w(void 0),
                isOpen: !0,
                fitContent: !0,
                hug: !1,
                legacyFooter: !1,
                trigger: /* @__PURE__ */ e("div", {}),
                children: /* @__PURE__ */ e(
                  F,
                  {
                    menuBlocks: [
                      {
                        id: "menu",
                        menuItems: [
                          ...k ? [
                            {
                              id: "upload",
                              size: I.XSmall,
                              title: "Upload asset",
                              onClick: () => {
                                k(), w(void 0);
                              },
                              initialValue: !0,
                              decorator: /* @__PURE__ */ e("div", { className: "tw-mr-2", children: /* @__PURE__ */ e(R, {}) })
                            }
                          ] : [],
                          ...x ? [
                            {
                              id: "asset",
                              size: I.XSmall,
                              title: "Browse asset",
                              onClick: () => {
                                x(), w(void 0);
                              },
                              initialValue: !0,
                              decorator: /* @__PURE__ */ e("div", { className: "tw-mr-2", children: /* @__PURE__ */ e(T, {}) })
                            }
                          ] : []
                        ]
                      }
                    ]
                  }
                )
              }
            )
          }
        )
      ]
    }
  );
};
export {
  H as BlockInjectButton
};
