// @frontify/fondue/dist/components/Card/Card.es.js
import m, { useRef as d, useState as p } from "react";
import { useButton as c } from "@react-aria/button";
import { merge as g } from "../../utilities/merge.es.js";
import { mergeProps as v } from "@react-aria/utils";
import { useFocusRing as x } from "@react-aria/focus";
import { FOCUS_STYLE as P } from "../../utilities/focusStyle.es.js";
const b = ({
  hoverable: r = !1,
  children: s,
  onClick: t,
  "data-test-id": n = "card"
}) => {
  const o = d(null), [u, i] = p(!1), { buttonProps: a } = c(
    {
      elementType: "div",
      onPress: (e) => t == null ? void 0 : t(e),
      onPressChange: (e) => {
        i(e);
      }
    },
    o
  ), { isFocusVisible: l, focusProps: w } = x(), f = t ? v(a, w) : {};
  return /* @__PURE__ */ m.createElement(
    "div",
    {
      ...f,
      "data-test-id": n,
      ref: o,
      className: g([
        "tw-w-full tw-bg-white tw-outline tw-outline-1 tw-outline-line tw-outline-offset--1 tw-rounded tw-overflow-hidden",
        (r || t) && "hover:tw-outline-line-xx-strong",
        t ? "active:tw-outline-line-xx-strong active:tw-outline-2 active:tw-outline-offset--2 tw-cursor-pointer" : "tw-cursor-default",
        l && P,
        u && t ? "tw-outline-line-xx-strong tw-outline-2 tw-outline-offset--2" : ""
      ])
    },
    s
  );
};
b.displayName = "FondueCard";
export {
  b as Card
};
