// @frontify/fondue/dist/components/Table/TableColumnHeader.es.js
import { CheckboxState as w, Checkbox as A } from "../Checkbox/Checkbox.es.js";
import { IconSize as N } from "../../foundation/Icon/IconSize.es.js";
import { FOCUS_VISIBLE_STYLE as y } from "../../utilities/focusStyle.es.js";
import { merge as p } from "../../utilities/merge.es.js";
import e, { useState as u, useRef as D, useEffect as z, cloneElement as B } from "react";
import f from "../../foundation/Icon/Generated/IconArrowBidirectional.es.js";
import { SortDirection as U, SelectionMode as F } from "./Table.es.js";
import L from "../../foundation/Icon/Generated/IconArrowDown.es.js";
import O from "../../foundation/Icon/Generated/IconArrowUp.es.js";
var _ = /* @__PURE__ */ ((t) => (t.Default = "Default", t.SelectAll = "SelectAll", t))(_ || {});
const H = ({
  column: t,
  type: d = "Default",
  rowIds: b,
  sortDirection: r,
  selectionMode: h,
  isColumnSorted: a = !1,
  handleSortChange: k,
  setSelectedRows: o
}) => {
  const {
    key: n,
    rendered: x,
    props: { allowsSorting: l }
  } = t, [E, c] = u(/* @__PURE__ */ e.createElement(f, null)), [s, S] = u(!1), i = D(null), g = l ? "button" : "span";
  if (z(() => {
    c(a ? r === U.Descending ? /* @__PURE__ */ e.createElement(L, null) : /* @__PURE__ */ e.createElement(O, null) : /* @__PURE__ */ e.createElement(f, null));
  }, [a, r]), d === "SelectAll") {
    const m = {
      "aria-label": "Select all",
      scope: "col"
    }, C = s ? w.Checked : w.Unchecked, I = () => {
      o && (o(s ? [] : b), S((v) => !v));
    };
    return /* @__PURE__ */ e.createElement(
      "th",
      {
        ...m,
        ref: i,
        className: "tw-pl-8 tw-py-3 tw-pr-4 tw-w-16 tw-border-l-4 tw-border-transparent tw-group tw-outline-none",
        "data-test-id": "table-select-cell"
      },
      h === F.SingleSelect ? /* @__PURE__ */ e.createElement("span", { className: "tw-sr-only" }, m["aria-label"]) : /* @__PURE__ */ e.createElement(A, { value: n, "aria-label": n, state: C, onChange: I })
    );
  }
  return /* @__PURE__ */ e.createElement(
    "th",
    {
      ref: i,
      className: p([
        "tw-text-xs tw-font-medium tw-text-black-100 dark:tw-text-white tw-px-4 tw-py-3 tw-outline-none tw-group focus-visible:bg-violet-90",
        l ? "tw-cursor-pointer" : ""
      ]),
      "data-test-id": "table-column",
      scope: "col",
      onClick: l ? () => k(t.key, r) : () => null
    },
    /* @__PURE__ */ e.createElement(
      g,
      {
        className: p([
          "tw-flex tw-gap-x-1 tw-items-center",
          y,
          l ? "tw-cursor-pointer" : "tw-cursor-default"
        ])
      },
      x,
      l && /* @__PURE__ */ e.createElement(
        "span",
        {
          "aria-hidden": "true",
          className: a ? "tw-text-violet-50" : "tw-text-black-40 dark:tw-text-black-60 group-hover:tw-text-black-100 dark:group-hover:tw-text-white"
        },
        B(E, { size: N.Size12 })
      )
    )
  );
};
H.displayName = "FondueTableColumnHeader";
export {
  H as TableColumnHeader,
  _ as TableColumnHeaderType
};
