// @frontify/fondue/dist/components/RichTextEditor/Plugins/LinkPlugin/FloatingLink/InsertLinkModal/InsertModal.es.js
import { Button as l } from "../../../../../Button/Button.es.js";
import e from "react";
import { FormControl as r } from "../../../../../FormControl/FormControl.es.js";
import { Checkbox as h } from "../../../../../Checkbox/Checkbox.es.js";
import E from "../../../../../../foundation/Icon/Generated/IconCheckMark20.es.js";
import { TextInput as n } from "../../../../../TextInput/TextInput.es.js";
import { ButtonSize as o, ButtonStyle as f, ButtonEmphasis as x } from "../../../../../Button/ButtonTypes.es.js";
const M = ({
  state: t,
  onTextChange: m,
  onUrlChange: i,
  onToggleTab: c,
  onCancel: d,
  onSave: s,
  isValidUrlOrEmpty: a,
  hasValues: u,
  testId: w,
  children: p
}) => /* @__PURE__ */ e.createElement("div", { "data-test-id": w, className: "tw-bg-white tw-rounded tw-shadow tw-p-7 tw-min-w-[400px] tw-overflow-y-auto" }, /* @__PURE__ */ e.createElement(
  r,
  {
    label: {
      children: "Text",
      htmlFor: "linkText",
      required: !0
    }
  },
  /* @__PURE__ */ e.createElement(n, { id: "linkText", value: t.text, placeholder: "Link Text", onChange: m })
), p, /* @__PURE__ */ e.createElement("div", { className: "tw-pt-5" }, /* @__PURE__ */ e.createElement(
  r,
  {
    label: {
      children: "URL",
      htmlFor: "url",
      required: !0
    }
  },
  /* @__PURE__ */ e.createElement(
    n,
    {
      id: "url",
      value: t.url,
      placeholder: "https://example.com",
      focusOnMount: !0,
      onChange: i
    }
  )
), !a() && /* @__PURE__ */ e.createElement("div", { className: "tw-text-red-65 tw-mt-3" }, "Please enter a valid URL.")), /* @__PURE__ */ e.createElement("div", { className: "tw-pt-5" }, /* @__PURE__ */ e.createElement(h, { value: "new-tab", label: "Open in new tab", state: t.newTab, onChange: c })), /* @__PURE__ */ e.createElement("div", { className: "tw-mt-3" }, /* @__PURE__ */ e.createElement("div", { className: "tw-pt-5 tw-flex tw-gap-x-3 tw-justify-end tw-border-t tw-border-t-black-10" }, /* @__PURE__ */ e.createElement(
  l,
  {
    onClick: d,
    size: o.Medium,
    style: f.Default,
    emphasis: x.Default
  },
  "Cancel"
), /* @__PURE__ */ e.createElement(
  l,
  {
    onClick: s,
    size: o.Medium,
    icon: /* @__PURE__ */ e.createElement(E, null),
    disabled: !a() || !u
  },
  "Save"
))));
export {
  M as InsertModal
};
