// @frontify/fondue/dist/components/RichTextEditor/Plugins/LinkPlugin/FloatingLink/InsertLinkModal/useInsertModal.es.js
import { useEffect as h, useReducer as y } from "react";
import { useEditorRef as f, floatingLinkSelectors as u, useHotkeys as E, floatingLinkActions as a, submitFloatingLink as U, getPluginOptions as b, ELEMENT_LINK as k } from "@udecode/plate";
import { CheckboxState as o } from "../../../../../Checkbox/Checkbox.es.js";
import { getLegacyUrl as m, getUrl as x } from "../../utils/getUrl.es.js";
const w = {
  url: "",
  text: "",
  newTab: o.Unchecked
}, I = () => {
  const [r, n] = y((t, c) => {
    const { type: s, payload: l } = c;
    switch (s) {
      case "NEW_TAB":
        return {
          ...t,
          newTab: o.Checked
        };
      case "SAME_TAB":
        return {
          ...t,
          newTab: o.Unchecked
        };
      case "URL":
      case "TEXT":
      case "INIT":
        return {
          ...t,
          ...l
        };
      default:
        return t;
    }
  }, w);
  return [r, n];
}, S = () => {
  const r = f(), [n, t] = I();
  h(() => {
    const e = m(r), g = x(r);
    t({
      type: "INIT",
      payload: {
        text: u.text(),
        newTab: u.newTab() ? o.Checked : o.Unchecked,
        url: e && g === "" ? e : u.url()
      }
    });
  }, [t, r]);
  const c = (e) => {
    t({
      type: "TEXT",
      payload: { text: e }
    });
  }, s = (e) => {
    t({
      type: "URL",
      payload: { url: e }
    });
  }, l = (e) => {
    t(e ? { type: "NEW_TAB" } : { type: "SAME_TAB" });
  }, p = () => {
    a.hide();
  }, i = (e) => {
    !d() || !T || (a.text(n.text), a.url(n.url), a.newTab(n.newTab === o.Checked), U(r) && (e == null || e.preventDefault()));
  }, T = n.url !== "" && n.text !== "", d = () => {
    const { isUrl: e } = b(r, k);
    return !n.url || e && e(n.url);
  };
  return E(
    "enter",
    i,
    {
      enableOnFormTags: ["INPUT"]
    },
    []
  ), { state: n, onTextChange: c, onUrlChange: s, onToggleTab: l, onCancel: p, onSave: i, hasValues: T, isValidUrlOrEmpty: d };
};
export {
  I as InsertModalState,
  S as useInsertModal
};
