// @frontify/fondue/dist/components/RichTextEditor/Plugins/TextStylePlugin/TextStyleDropdown/DropdownItem.es.js
import { getMark as s, someNode as r, setMarks as w, toggleNodeType as l, getAboveNode as c, ELEMENT_LI as i } from "@udecode/plate";
import p from "react";
import { MARK_TEXT_STYLE as n } from "../../ListPlugin/ListPlugin.es.js";
import { merge as u } from "../../../../../utilities/merge.es.js";
import { ELEMENT_CHECK_ITEM as x } from "../../CheckboxListPlugin/id.es.js";
const E = (t) => c(t, {
  match: { type: i },
  mode: "lowest"
}) || r(t, {
  match: { type: x },
  mode: "lowest"
}), g = ({ editor: t, type: e, children: m }) => {
  const a = t != null && t.selection ? s(t, n) === e.id : r(t, { match: { type: e } });
  return /* @__PURE__ */ p.createElement(
    "button",
    {
      "data-test-id": "textstyle-option",
      type: "button",
      className: u([
        "tw-block tw-w-full tw-text-left tw-px-3 tw-py-2 tw-outline-none tw-cursor-pointer tw-truncate hover:tw-bg-box-neutral-hover hover:w-text-box-neutral-inverse-hover tw-max-h-20 tw-max-w-[15rem] tw-min-h-[2.25rem]",
        a ? "tw-text-box-neutral-inverse tw-bg-box-neutral" : "tw-text-text"
      ]),
      onMouseDown: (o) => {
        o.preventDefault(), o.stopPropagation(), !(!t || !t.selection) && (w(t, { [n]: e.id }), E(t) || l(t, {
          activeType: e.id,
          inactiveType: e.id
        }));
      }
    },
    m
  );
};
export {
  g as DropdownItem
};
