// @frontify/fondue/dist/components/RichTextEditor/Plugins/TextStylePlugin/TextStyleDropdown/DropdownTrigger.es.js
import t, { forwardRef as s } from "react";
import { merge as o } from "../../../../../utilities/merge.es.js";
import l from "../../../../../foundation/Icon/Generated/IconCaretDown12.es.js";
const i = ({ label: w, open: e, onClick: n }, a) => /* @__PURE__ */ t.createElement(
  "button",
  {
    ref: a,
    "data-test-id": "textstyle-dropdown-trigger",
    type: "button",
    className: "tw-cursor-pointer tw-h-8 tw-w-full",
    onMouseDown: (r) => {
      r.preventDefault(), r.stopPropagation(), n();
    }
  },
  /* @__PURE__ */ t.createElement(
    "div",
    {
      className: o([
        "tw-relative tw-inline-flex tw-flex-row tw-items-center tw-justify-between tw-overflow-hidden tw-text-text tw-rounded tw-gap-1 tw-px-2 hover:tw-text-box-selected-inverse hover:tw-bg-box-selected tw-h-6 tw-w-28",
        e && "tw-text-box-selected-inverse tw-bg-box-selected"
      ])
    },
    /* @__PURE__ */ t.createElement("span", { className: "tw-text-xs tw-truncate" }, w),
    /* @__PURE__ */ t.createElement("div", { className: o(["tw-transition-transform", e && "tw-rotate-180"]) }, /* @__PURE__ */ t.createElement(l, null))
  )
), p = s(i);
export {
  p as DropdownTrigger,
  i as DropdownTriggerComponent
};
