// @frontify/fondue/dist/components/RichTextEditor/Plugins/TextStylePlugin/TextStyleDropdown/index.es.js
import t from "react";
import { DropdownItem as E } from "./DropdownItem.es.js";
import { DropdownTrigger as T } from "./DropdownTrigger.es.js";
import { useTextStyleDropdown as g } from "./useTextStyleDropdown.es.js";
import { DEFAULT_TEXT_STYLE_VALUE as m } from "../types.es.js";
const k = ({ editorId: i, textStyles: o = [] }) => {
  var d;
  const {
    state: { editor: l, toggle: w, isOpen: p },
    dropdownProps: s,
    triggerRef: c,
    dropdownRef: f,
    key: b
  } = g(i);
  let n = m;
  const a = o.find((e) => {
    var r;
    return e.id === b && ((r = e.props) == null ? void 0 : r.label);
  });
  return a && (n = ((d = a.props) == null ? void 0 : d.label) || m), /* @__PURE__ */ t.createElement(t.Fragment, null, /* @__PURE__ */ t.createElement(T, { label: n, open: p, onClick: w, ref: c }), p && /* @__PURE__ */ t.createElement(
    "div",
    {
      className: "tw-divide-y tw-divide-line tw-bg-base tw-shadow-md tw-border tw-border-line tw-z-[1000] tw-overflow-auto tw-min-h-[40px]",
      ref: f,
      ...s
    },
    o.map((e) => {
      var r;
      return /* @__PURE__ */ t.createElement(E, { editor: l, type: e, key: e.id }, /* @__PURE__ */ t.createElement("span", { style: e.styles ?? {} }, (r = e.props) == null ? void 0 : r.label));
    })
  ));
};
export {
  k as TextStyleDropdown
};
