// @frontify/fondue/dist/layout/LegacyStack/LegacyStack.es.js
import { merge as r } from "../../utilities/merge.es.js";
import l from "react";
const c = {
  none: "tw-p-0",
  xxs: "tw-p-xxs",
  xs: "tw-p-xs",
  s: "tw-p-s",
  m: "tw-p-m",
  l: "tw-p-l",
  xl: "tw-p-xl",
  xxl: "tw-p-xxl"
}, o = {
  none: "tw-gap-0",
  xxs: "tw-gap-xxs",
  xs: "tw-gap-xs",
  s: "tw-gap-s",
  m: "tw-gap-m",
  l: "tw-gap-l",
  xl: "tw-gap-xl",
  xxl: "tw-gap-xxl"
}, i = {
  start: "tw-justify-start",
  end: "tw-justify-end",
  center: "tw-justify-center",
  between: "tw-justify-between",
  around: "tw-justify-around",
  evenly: "tw-justify-evenly"
}, d = {
  start: "tw-items-start",
  end: "tw-items-end",
  center: "tw-items-center",
  stretch: "tw-items-stretch"
}, f = "fondue-legacy-stack", m = ({
  children: t,
  padding: e,
  spacing: s,
  direction: a = "row",
  wrap: w = !1,
  justify: n = "start",
  align: x = "stretch",
  "data-test-id": p = f
}) => /* @__PURE__ */ l.createElement(
  "div",
  {
    "data-test-id": p,
    className: r([
      "tw-flex",
      c[e],
      o[s],
      i[n],
      d[x],
      a === "row" ? "tw-flex-row" : "tw-flex-col",
      w ? "tw-flex-wrap" : "tw-flex-nowrap"
    ])
  },
  t
);
m.displayName = "FondueLegacyStack";
export {
  m as LegacyStack
};
