// @frontify/fondue/dist/components/RichTextEditor/Plugins/MentionPlugin/MentionMarkupElement/MarkupElementNode.es.js
import t from "react";
import { MentionableCategory as r } from "../types.es.js";
import c from "../../../../../foundation/Icon/Generated/IconPeople12.es.js";
const l = ({ attributes: a, element: o, nodeProps: n, children: e }) => /* @__PURE__ */ t.createElement(
  "span",
  {
    ...a,
    "data-slate-value": o == null ? void 0 : o.value,
    "data-slate-key": o == null ? void 0 : o.key,
    "data-slate-category": o == null ? void 0 : o.category,
    contentEditable: !1,
    style: {
      lineHeight: "10px",
      padding: "1px",
      margin: "0 1px",
      fontWeight: "bold",
      verticalAlign: "baseline",
      display: "inline-block",
      borderRadius: "2px",
      backgroundColor: "rgb(227, 232, 246)",
      color: "rgb(130, 95, 255)"
    },
    ...n
  },
  e
), i = ({ children: a, ...o }) => /* @__PURE__ */ t.createElement(l, { ...o }, /* @__PURE__ */ t.createElement(
  "span",
  {
    style: {
      paddingRight: "3px",
      display: "inline-block",
      margin: "-1px 0"
    }
  },
  /* @__PURE__ */ t.createElement(c, null)
), a), s = ({ children: a, ...o }) => /* @__PURE__ */ t.createElement(l, { ...o }, "@", a), d = ({ children: a, ...o }) => /* @__PURE__ */ t.createElement(l, { ...o }, a), b = ({ category: a, children: o, ...n }) => {
  const e = {
    [r.USER]: d,
    [r.GROUP]: i,
    [r.ALL]: s
  }, p = e[a] ?? e[r.USER];
  return /* @__PURE__ */ t.createElement(p, { ...n }, o);
};
export {
  b as MarkupElementNodeComponent
};
