// @frontify/guideline-blocks-settings/dist/components/Attachments/AttachmentItem.es.js
import { jsxs as h, jsx as t } from "react/jsx-runtime";
import { forwardRef as E, useState as L, useEffect as y } from "react";
import { useFileInput as M, useAssetUpload as P } from "@frontify/app-bridge";
import { useSortable as V } from "@dnd-kit/sortable";
import { useFocusRing as B } from "@react-aria/focus";
import { LoadingCircle as O, LoadingCircleSize as $, FOCUS_STYLE as T, IconGrabHandle20 as X, Flyout as G, FlyoutPlacement as H, Button as _, IconPen20 as Y, ButtonEmphasis as q, ActionMenu as J, MenuItemContentSize as g, IconArrowCircleUp20 as K, IconImageStack20 as Q, MenuItemStyle as W, IconTrashBin20 as Z, IconImage24 as tt, IconPlayFrame24 as et, IconMusicNote24 as ot, IconDocument24 as at } from "@frontify/fondue";
import { joinClassNames as v } from "../../utilities/react/joinClassNames.es.js";
const rt = (e) => e === "IMAGE" ? /* @__PURE__ */ t(tt, {}) : e === "VIDEO" ? /* @__PURE__ */ t(et, {}) : e === "AUDIO" ? /* @__PURE__ */ t(ot, {}) : /* @__PURE__ */ t(at, {}), k = E(
  ({
    item: e,
    isEditing: c,
    draggableProps: d,
    transformStyle: u,
    isDragging: o,
    isOverlay: r,
    isLoading: i,
    onDelete: w,
    onReplaceWithBrowse: m,
    onReplaceWithUpload: S
  }, N) => {
    const [a, n] = L(), [C, { selectedFiles: l }] = M({ multiple: !0, accept: "image/*" }), [F, { results: x, doneAll: p }] = P(), { focusProps: R, isFocusVisible: I } = B();
    y(() => {
      l && F(l[0]);
    }, [l]), y(() => {
      p && S(x[0]);
    }, [p, x]);
    const z = (s, b) => {
      fetch(s).then((A) => {
        A.blob().then((U) => {
          const j = URL.createObjectURL(U), f = document.createElement("a");
          f.href = j, f.download = b, f.click();
        });
      });
    }, D = i || l && !p;
    return /* @__PURE__ */ h(
      "button",
      {
        "aria-label": "Download attachment",
        "data-test-id": "attachments-item",
        onClick: () => z(e.genericUrl, e.fileName),
        ref: N,
        style: {
          ...u,
          opacity: o && !r ? 0.3 : 1,
          fontFamily: "var(-f-theme-settings-body-font-family)"
        },
        className: v([
          "tw-cursor-pointer tw-text-left tw-w-full tw-relative tw-flex tw-gap-3 tw-px-5 tw-py-3 tw-items-center tw-group hover:tw-bg-box-neutral-hover",
          o ? "tw-bg-box-neutral-hover" : ""
        ]),
        children: [
          /* @__PURE__ */ t("div", { className: "tw-text-text-weak group-hover:tw-text-box-neutral-inverse-hover", children: D ? /* @__PURE__ */ t(O, { size: $.Small }) : rt(e.objectType) }),
          /* @__PURE__ */ h("div", { className: "tw-text-s tw-flex-1 tw-min-w-0", children: [
            /* @__PURE__ */ t("div", { className: "tw-whitespace-nowrap tw-overflow-hidden tw-text-ellipsis tw-font-bold tw-text-text-weak group-hover:tw-text-box-neutral-inverse-hover", children: e.title }),
            /* @__PURE__ */ t("div", { className: "tw-text-text-weak", children: `${e.fileSizeHumanReadable} - ${e.extension}` })
          ] }),
          c && /* @__PURE__ */ h(
            "div",
            {
              "data-test-id": "attachments-actionbar",
              className: v([
                "tw-flex tw-gap-0.5 group-focus:tw-opacity-100 focus-visible:tw-opacity-100 focus-within:tw-opacity-100 group-hover:tw-opacity-100",
                r || (a == null ? void 0 : a.id) === e.id ? "tw-opacity-100" : "tw-opacity-0"
              ]),
              children: [
                /* @__PURE__ */ t(
                  "button",
                  {
                    ...R,
                    ...d,
                    "aria-label": "Drag attachment",
                    className: v([
                      " tw-border-button-border tw-bg-button-background active:tw-bg-button-background-pressed tw-group tw-border tw-box-box tw-relative tw-flex tw-items-center tw-justify-center tw-outline-none tw-font-medium tw-rounded tw-h-9 tw-w-9 ",
                      o || r ? "tw-cursor-grabbing tw-bg-button-background-pressed hover:tw-bg-button-background-pressed" : "tw-cursor-grab hover:tw-bg-button-background-hover",
                      I && T,
                      I && "tw-z-[2]"
                    ]),
                    children: /* @__PURE__ */ t(X, {})
                  }
                ),
                /* @__PURE__ */ t("div", { "data-test-id": "attachments-actionbar-flyout", children: /* @__PURE__ */ t(
                  G,
                  {
                    placement: H.Right,
                    isOpen: (a == null ? void 0 : a.id) === e.id,
                    fitContent: !0,
                    legacyFooter: !1,
                    onOpenChange: (s) => n(s ? e : void 0),
                    trigger: (s, b) => /* @__PURE__ */ t(
                      _,
                      {
                        ref: b,
                        icon: /* @__PURE__ */ t(Y, {}),
                        emphasis: q.Default,
                        onClick: () => n(e)
                      }
                    ),
                    children: /* @__PURE__ */ t(
                      J,
                      {
                        menuBlocks: [
                          {
                            id: "menu",
                            menuItems: [
                              {
                                id: "upload",
                                size: g.XSmall,
                                title: "Replace with upload",
                                onClick: () => {
                                  C(), n(void 0);
                                },
                                initialValue: !0,
                                decorator: /* @__PURE__ */ t("div", { className: "tw-mr-2", children: /* @__PURE__ */ t(K, {}) })
                              },
                              {
                                id: "asset",
                                size: g.XSmall,
                                title: "Replace with asset",
                                onClick: () => {
                                  m(), n(void 0);
                                },
                                initialValue: !0,
                                decorator: /* @__PURE__ */ t("div", { className: "tw-mr-2", children: /* @__PURE__ */ t(Q, {}) })
                              }
                            ]
                          },
                          {
                            id: "menu-delete",
                            menuItems: [
                              {
                                id: "delete",
                                size: g.XSmall,
                                title: "Delete",
                                style: W.Danger,
                                onClick: () => {
                                  w(), n(void 0);
                                },
                                initialValue: !0,
                                decorator: /* @__PURE__ */ t("div", { className: "tw-mr-2", children: /* @__PURE__ */ t(Z, {}) })
                              }
                            ]
                          }
                        ]
                      }
                    )
                  }
                ) })
              ]
            }
          )
        ]
      }
    );
  }
);
k.displayName = "AttachmentItem";
const wt = (e) => {
  const { attributes: c, listeners: d, setNodeRef: u, transform: o, transition: r, isDragging: i } = V({
    id: e.item.id
  }), w = {
    transform: o ? `translate(${o.x}px, ${o.y}px)` : "",
    transition: r,
    zIndex: i ? 2 : 1
  }, m = { ...c, ...d };
  return /* @__PURE__ */ t(
    k,
    {
      ref: u,
      isDragging: i,
      transformStyle: w,
      draggableProps: m,
      ...e
    }
  );
};
export {
  k as AttachmentItem,
  wt as SortableAttachmentItem
};
