// @frontify/fondue/dist/components/Accordion/AccordionHeader.es.js
import t, { useState as s, isValidElement as m, cloneElement as w } from "react";
import { IconSize as x } from "../../foundation/Icon/IconSize.es.js";
import { merge as d } from "../../utilities/merge.es.js";
import { AccordionHeaderIcon as f } from "./AccordionHeaderIcon.es.js";
const p = "fondue-accordion-header", E = ({
  isOpen: a = !1,
  decorator: r,
  disabled: n = !1,
  children: l,
  "data-test-id": e = p
}) => {
  const [o, i] = s(null), c = /* @__PURE__ */ t.createElement(f, { isOpen: a });
  return /* @__PURE__ */ t.createElement("span", { "data-test-id": e, className: "tw-block tw-px-8 tw-py-6" }, /* @__PURE__ */ t.createElement(
    "span",
    {
      "data-test-id": `${e}-container`,
      role: "navigation",
      "aria-label": `${(o == null ? void 0 : o.innerText) || ""}`,
      className: d([
        "tw-flex tw-items-center tw-gap-x-1.5 tw-w-full tw-flex-row",
        !n && a && "tw-text-text",
        !n && !a && "tw-text-text-weak hover:tw-text-text",
        n && "tw-text-text-disabled"
      ])
    },
    m(r) && /* @__PURE__ */ t.createElement("span", { "data-test-id": `${e}-decorator`, className: "tw-shrink-0 tw-leading-4" }, w(r, {
      size: x.Size16
    })),
    /* @__PURE__ */ t.createElement(
      "span",
      {
        ref: i,
        "data-test-id": `${e}-text`,
        className: "tw-text-left tw-text-m tw-font-normal"
      },
      l
    ),
    c && /* @__PURE__ */ t.createElement("span", { "data-test-id": `${e}-icon-container`, className: "tw-ml-auto tw-shrink-0" }, c)
  ));
};
E.displayName = "FondueAccordionHeader";
export {
  p as ACCORDION_HEADER_TEST_ID,
  E as AccordionHeader
};
