// @frontify/fondue/dist/components/ColorInputFlyout/ColorPickerTrigger.es.js
import { MenuItemContent as f } from "../MenuItem/MenuItemContent.es.js";
import { Trigger as g } from "../Trigger/Trigger.es.js";
import { IconSize as u } from "../../foundation/Icon/IconSize.es.js";
import { useMemoizedId as x } from "../../hooks/useMemoizedId.es.js";
import { useFocusRing as b } from "@react-aria/focus";
import { toShortRgb as d } from "../../utilities/colors.es.js";
import { merge as r } from "../../utilities/merge.es.js";
import e from "react";
import y from "tinycolor2";
import { ColorInputTitle as E } from "./ColorInputTitle.es.js";
import I from "../../foundation/Icon/Generated/IconDroplet.es.js";
const h = ({
  id: m,
  currentColor: t,
  format: i,
  isOpen: n = !1,
  disabled: o = !1,
  clearable: l = !1,
  onClear: a,
  onDelete: p
}) => {
  const { isFocusVisible: s, focusProps: w } = b(), c = t ? y(d(t)).toRgbString() : "";
  return /* @__PURE__ */ e.createElement(
    g,
    {
      isOpen: n,
      disabled: o,
      isFocusVisible: s,
      showClear: !!t && l,
      onClear: a,
      onDelete: p
    },
    /* @__PURE__ */ e.createElement(
      "button",
      {
        ...w,
        type: "button",
        id: x(m),
        disabled: o,
        className: r([
          "tw-overflow-hidden tw-flex-auto tw-h-full tw-rounded tw-text-left tw-outline-none tw-py-2 tw-pl-3 tw-min-h-[34px] tw-pr-7",
          !t && "tw-text-black-60",
          o && "tw-text-black-40"
        ])
      },
      /* @__PURE__ */ e.createElement(
        f,
        {
          title: t ? /* @__PURE__ */ e.createElement(E, { format: i, currentColor: t }) : "Select color",
          decorator: t ? /* @__PURE__ */ e.createElement(
            "span",
            {
              className: r([
                "tw-h-4 tw-w-4 tw-rounded tw-flex tw-items-center tw-justify-center tw-ring-1 tw-ring-black-10 tw-ring-offset-1",
                o && "tw-opacity-50"
              ]),
              style: { backgroundColor: c }
            }
          ) : /* @__PURE__ */ e.createElement("span", { className: "tw-text-black-70" }, /* @__PURE__ */ e.createElement(I, { size: u.Size12 }))
        }
      )
    )
  );
};
h.displayName = "FondueColorInputTrigger";
export {
  h as ColorInputTrigger
};
