// @frontify/fondue/dist/components/ColorInputFlyout/ColorPickerFlyout.es.js
import { ColorPicker as y } from "../ColorPicker/ColorPicker.es.js";
import { Flyout as E } from "../Flyout/Flyout.es.js";
import r, { useState as c } from "react";
import { ColorFormat as O } from "../../types/colors.es.js";
import { ColorInputTrigger as v } from "./ColorPickerTrigger.es.js";
import { ColorPreview as w } from "../ColorPicker/ColorPreview.es.js";
const x = ({
  id: g,
  onClick: t,
  onClose: o,
  onSelect: u,
  currentColor: a,
  palettes: d,
  disabled: n = !1,
  clearable: F = !1,
  onClear: i,
  onDelete: m
}) => {
  const [l, e] = c(!1), [f, h] = c(O.Hex), s = (p) => {
    p || o == null || o(), e(p);
  }, P = () => {
    e(!1), t == null || t();
  };
  return /* @__PURE__ */ r.createElement(
    E,
    {
      hug: !1,
      onConfirm: P,
      isOpen: l,
      onCancel: () => s(!1),
      contentMinHeight: 150,
      fixedHeader: /* @__PURE__ */ r.createElement(w, { color: a || { red: 255, green: 255, blue: 255 } }),
      onOpenChange: s,
      isTriggerDisabled: n,
      trigger: /* @__PURE__ */ r.createElement(
        v,
        {
          isOpen: l,
          currentColor: a,
          format: f,
          disabled: n,
          id: g,
          clearable: F,
          onClear: () => {
            e(!1), i && i();
          },
          onDelete: m ? () => {
            e(!1), m && m();
          } : void 0
        }
      )
    },
    /* @__PURE__ */ r.createElement(
      y,
      {
        currentFormat: f,
        setFormat: h,
        palettes: d,
        showPreview: !1,
        currentColor: a || { red: 255, green: 255, blue: 255 },
        onSelect: u
      }
    )
  );
};
x.displayName = "FondueColorPickerFlyout";
export {
  x as ColorPickerFlyout
};
