// @frontify/fondue/dist/components/Slider/Slider.es.js
import s, { useState as d, useCallback as W, useMemo as nt, useEffect as A } from "react";
import { useMemoizedId as at } from "../../hooks/useMemoizedId.es.js";
import { merge as p } from "../../utilities/merge.es.js";
import { Validation as X } from "../../utilities/validation.es.js";
import { clamp as x } from "../../utilities/number.es.js";
import { debounce as st } from "../../utilities/debounce.es.js";
import { TextInput as lt, TextInputType as ct } from "../TextInput/TextInput.es.js";
var it = /* @__PURE__ */ ((E) => (E.MinMax = "MinMax", E.ValueOutOfRange = "ValueOutOfRange", E.ValueNaN = "ValueNaN", E))(it || {});
const wt = 0, ut = 100, ft = 1, dt = 5, Et = "Slider text input", mt = 3, K = ["ArrowUp", "ArrowRight"], B = ["ArrowDown", "ArrowLeft"], S = ["Home", "End"], pt = ({
  id: E,
  label: _,
  value: k,
  min: o = wt,
  max: l = ut,
  showMinMax: U = !1,
  step: u = ft,
  stepMultiplier: z = dt,
  valueSuffix: r = "",
  onError: y,
  onChange: $,
  "aria-label": I = Et,
  disabled: e = !1,
  "data-test-id": N = "fondue-slider"
}) => {
  const [c, b] = d(), [i, h] = d(""), [m, v] = d(), [D, P] = d(), [n, Y] = d(null), [H, R] = d(!1), [O, L] = d(!1), T = at(E), j = (t) => {
    h(`${t.replace(r, "")}${r}`);
  }, f = W(
    ({ clientX: t, rawValue: a }) => {
      if (!n)
        return;
      const w = l - o, tt = n.getBoundingClientRect().x, et = n.clientWidth;
      let V;
      if (t !== void 0)
        V = (t - tt) / et;
      else {
        V = ((a || o) - o) / w, P(x(V, 0, 1) * 100);
        return;
      }
      const ot = x(V, 0, 1), rt = w / u, C = x(Math.round(ot * rt) * u + o, o, l);
      P((C - o) / w * 100), h(`${C}${r}`);
    },
    [n, l, o, u, r]
  ), g = nt(
    () => n ? st((t) => {
      e || f({
        clientX: t.clientX ?? t.touches[0].clientX
      });
    }, mt) : () => {
    },
    [f, n, e]
  ), M = W(() => {
    n && (L(!1), n.removeEventListener("mousemove", g), n.addEventListener("touchmove", g), window.removeEventListener("mouseup", M));
  }, [n, g]), F = (t) => {
    !n || e || (L(!0), f({
      clientX: t.clientX ?? t.touches[0].clientX
    }), n.addEventListener("touchmove", g), n.addEventListener("mousemove", g), window.addEventListener("mouseup", M));
  }, q = (t) => {
    if (![...K, ...B, ...S].includes(t.key) || c === void 0 || e)
      return;
    t.preventDefault();
    let a = 0, w = 0;
    B.includes(t.key) ? a = -u : K.includes(t.key) && (a = u), S.includes(t.key) ? w = t.key === "Home" ? o : l : (a *= t.shiftKey ? z : 1, w = x(Math.floor(c / u) * u + a, o, l)), b(w), h(`${w}${r}`);
  }, G = () => {
    e || R(!0);
  }, J = () => {
    e || R(!1);
  }, Q = () => {
    e || L(!0);
  }, Z = () => {
    e || L(!1);
  };
  return A(() => {
    if (!n)
      return;
    if (o > l) {
      v(
        "MinMax"
        /* MinMax */
      );
      return;
    }
    const t = k || o;
    b(t), h(`${t}${r}`), f({ rawValue: t });
  }, [n, f, o, l, k, r]), A(() => {
    if (!i)
      return;
    const t = i.replace(r, ""), a = +t;
    if (Number.isNaN(a) || !t) {
      b(void 0), v(
        "ValueNaN"
        /* ValueNaN */
      );
      return;
    }
    if (a < o || a > l) {
      b(void 0), v(
        "ValueOutOfRange"
        /* ValueOutOfRange */
      );
      return;
    }
    v(void 0), b(a);
  }, [i, r, o, l]), A(() => {
    if (m || c === void 0 || !i)
      return;
    const a = +i.replace(r, "");
    Number.isNaN(a) || c !== a || (f({ rawValue: c }), e || $({
      raw: c,
      withSuffix: i.replace(r, "") + r
    }));
  }, [c, i, r, m, $, f, e]), A(() => {
    m && y && y(m);
  }, [m, y]), /* @__PURE__ */ s.createElement("div", { className: "tw-flex tw-flex-col", "data-test-id": N, id: T }, /* @__PURE__ */ s.createElement(
    "label",
    {
      htmlFor: T,
      className: p([!_ && "tw-hidden", e && "tw-text-text-disabled"]),
      "data-test-id": `${N}-label`
    },
    _
  ), /* @__PURE__ */ s.createElement("div", { className: "tw-flex" }, /* @__PURE__ */ s.createElement("div", { className: p(["tw-flex-1 tw-flex tw-items-center"]) }, U && /* @__PURE__ */ s.createElement("div", { className: p(["tw-mr-3", e && "tw-text-text-disabled"]) }, o, r), /* @__PURE__ */ s.createElement(
    "button",
    {
      ref: Y,
      "data-test-id": `${N}-interactive`,
      className: "tw-flex-1 tw-relative tw-h-full tw-cursor-pointer disabled:tw-cursor-default tw-outline-none",
      onMouseOver: G,
      onMouseOut: J,
      onFocus: Q,
      onBlur: Z,
      onMouseDown: F,
      onMouseUp: M,
      onTouchStart: F,
      onTouchEnd: M,
      onKeyDown: q,
      disabled: e
    },
    /* @__PURE__ */ s.createElement("span", { className: "tw-absolute tw-block tw-top-1/2 tw--translate-y-1/2 tw-w-full tw-h-1 tw-rounded-sm tw-bg-box-neutral tw-flex-1" }),
    D !== void 0 && /* @__PURE__ */ s.createElement(
      "span",
      {
        role: "slider",
        "aria-valuenow": c,
        "aria-valuemin": o,
        "aria-valuemax": l,
        "aria-label": I,
        "data-test-id": `${N}-track`,
        className: p([
          "tw-absolute tw-block tw-top-1/2 tw--translate-y-1/2 tw-origin-left tw-w-full tw-h-1  tw-rounded-sm tw-bg-box-neutral-strong tw-flex-1",
          e && "tw-bg-box-neutral"
        ]),
        style: { width: `${D}%` }
      }
    ),
    /* @__PURE__ */ s.createElement(
      "span",
      {
        className: p([
          "tw-absolute tw-block tw-top-1/2 tw--translate-y-1/2 tw--translate-x-1/2 tw-w-5 tw-h-5",
          "after:tw-absolute after:tw-block after:tw-top-1/2 after:tw-left-1/2 after:tw--translate-y-1/2 after:tw--translate-x-1/2 after:tw-w-5 after:tw-h-5 after:tw-bg-base after:tw-rounded-full after:tw-border after:tw-border-line-strong",
          H && !O && "after:tw-shadow",
          O && "after:tw-border-line-xx-strong before:tw-content-[''] before:tw-opacity-25 before:tw-top-1/2 before:tw--translate-y-1/2 before:tw--translate-x-1/2 before:tw-left-1/2 before:tw-block before:tw-rounded-full before:tw-absolute before:tw-z-0 before:tw-w-9 before:tw-h-9 before:tw-bg-box-neutral-strong",
          e && "after:tw-bg-box-neutral after:tw-border-line-weak"
        ]),
        style: { left: `${D}%` }
      }
    )
  ), U && /* @__PURE__ */ s.createElement("div", { className: p(["tw-ml-3", e && "tw-text-text-disabled"]) }, l, r), /* @__PURE__ */ s.createElement("div", { className: "tw-w-16 tw-ml-3" }, /* @__PURE__ */ s.createElement(
    lt,
    {
      id: `${T}-text-input`,
      value: i,
      placeholder: I,
      type: ct.Text,
      validation: m ? X.Error : X.Default,
      onChange: j,
      disabled: e
    }
  )))));
};
pt.displayName = "FondueSlider";
export {
  Et as ARIA_LABEL_DEFAULT_VALUE,
  ut as MAX_DEFAULT_VALUE,
  wt as MIN_DEFAULT_VALUE,
  ft as STEP_DEFAULT_VALUE,
  dt as STEP_MULTIPLIER_DEFAULT_VALUE,
  pt as Slider,
  it as SliderError
};
