// @frontify/fondue/dist/components/InlineDialog/InlineDialog.es.js
import M from "react";
import { Modality as e } from "../../types/dialog.es.js";
import { Trigger as c } from "../../utilities/dialogs/Trigger.es.js";
import { Content as N } from "../../utilities/dialogs/Content.es.js";
import { Overlay as x } from "../../utilities/dialogs/Overlay.es.js";
import { Z_INDEX_MODAL as y } from "../../utilities/dialogs/constants.es.js";
const o = ({
  children: r,
  open: a,
  placement: i = "bottom-start",
  offset: n = [0, 8],
  flip: l = !1,
  enablePortal: m = !0,
  maxHeight: d = "auto",
  maxWidth: f = 400,
  minHeight: g = 0,
  minWidth: p = 0,
  modality: t = e.Modal,
  handleClose: s,
  "data-test-id": u = "fondue-inlineDialog",
  darkUnderlay: D = !1,
  autoHeight: I = !1
}) => /* @__PURE__ */ M.createElement(
  x,
  {
    "data-test-id": u,
    open: a,
    placement: i,
    offset: n,
    flip: l,
    enablePortal: m,
    minWidth: p,
    minHeight: g,
    maxWidth: f,
    maxHeight: d,
    modality: t,
    handleClose: s,
    role: t === e.NonModal ? "region" : "dialog",
    darkUnderlay: D,
    autoHeight: I,
    zIndex: y
  },
  r
);
o.displayName = "FondueInlineDialog";
o.Trigger = c;
o.Content = N;
export {
  o as InlineDialog
};
