// @frontify/fondue/dist/components/Tree/TreeItem/TreeItem.es.js
import D, { memo as Ut, useRef as Xt, useEffect as qt, useCallback as T, Children as ot, useMemo as pt } from "react";
import { useSortable as zt } from "@dnd-kit/sortable";
import { useDndContext as Gt, useDndMonitor as Jt } from "@dnd-kit/core";
import { CSS as Ot } from "@dnd-kit/utilities";
import Qt from "../../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/noop.es.js";
import { merge as lt } from "../../../utilities/merge.es.js";
import { FOCUS_VISIBLE_STYLE as Zt } from "../../../utilities/focusStyle.es.js";
import { DragHandle as kt } from "./DragHandle.es.js";
import { ExpandButton as Ct } from "./ExpandButton.es.js";
import { useDebounce as ut } from "../../../hooks/useDebounce.es.js";
import { useTreeItem as Pt } from "./useTreeItem.es.js";
import { EXPAND_ONHOVER_DELAY as jt, INDENTATION_WIDTH as mt } from "../helpers/constants.es.js";
import { removeFragmentsAndEnrichChildren as te } from "../utils/removeFragmentsAndEnrichChildren.es.js";
import { useDeepCompareEffect as ee } from "../utils/useDeepCompareEffect.es.js";
const se = ({ isSorting: s, wasDragging: v }) => !(s || v), ae = Ut(
  ({
    id: s,
    type: v,
    label: i,
    onDrop: L,
    accepts: wt,
    showCaret: M,
    children: c,
    parentId: R,
    level: l = 0,
    contentComponent: b,
    treeDraggable: ft = !1,
    onClick: W,
    onSelect: _,
    onExpand: u,
    onShrink: I,
    registerOverlay: x,
    registerNodeChildren: $,
    unregisterNodeChildren: o,
    draggable: it = !0,
    ignoreItemDoubleClick: S = !1,
    expandOnSelect: ct = !1,
    "data-test-id": bt = "fondue-tree-item"
  }) => {
    var G, J, O, Q, Z, k, C, P, j, tt, et, st, at, dt, rt;
    const { active: r, over: y } = Gt(), { isSelected: w, isExpanded: m, projection: e } = Pt(s), V = Xt(), Y = ft && it, a = (r == null ? void 0 : r.id) === s, t = a && e !== null && e !== void 0 ? e : null, ht = typeof ((J = (G = y == null ? void 0 : y.data) == null ? void 0 : G.current) == null ? void 0 : J.accepts) == "string" ? (O = y.data.current.accepts) == null ? void 0 : O.split(", ") : [], gt = typeof (t == null ? void 0 : t.accepts) == "string" ? (Q = t.accepts) == null ? void 0 : Q.split(", ") : [], N = ((Z = r == null ? void 0 : r.data.current) == null ? void 0 : Z.type) || "", A = (N == null ? void 0 : N.replace(/-\d+$/, "")) || "", h = ((k = e == null ? void 0 : e.previousNode) == null ? void 0 : k.depth) !== void 0 && (e == null ? void 0 : e.depth) > ((C = e == null ? void 0 : e.previousNode) == null ? void 0 : C.depth), Dt = h && e.depth - 1 === ((P = e == null ? void 0 : e.previousNode) == null ? void 0 : P.depth), B = h && ((j = e == null ? void 0 : e.previousNode) == null ? void 0 : j.accepts) !== void 0 && (((tt = e == null ? void 0 : e.previousNode) == null ? void 0 : tt.accepts.includes(`${A}-deeper`)) || ((et = e == null ? void 0 : e.previousNode) == null ? void 0 : et.accepts.includes(`${A}-within`))), It = a && Dt && ((st = t == null ? void 0 : t.previousNode) == null ? void 0 : st.accepts) !== void 0 && ((at = t == null ? void 0 : t.previousNode) == null ? void 0 : at.accepts.includes(`${A}-within`)) || (t == null ? void 0 : t.isWithinParent) && gt.includes(`${A}-within`), g = a && (r == null ? void 0 : r.data.current) && (ht.includes(N) && !h || It), K = ut((d) => {
      V.current === d && (u == null || u(d));
    }, jt);
    qt(() => {
      var d, p, f;
      a && (V.current = h ? (d = t == null ? void 0 : t.previousNode) == null ? void 0 : d.id : null), a && B && (t != null && t.parentId) && t.previousNode && t.parentId === t.previousNode.id && t.parentId !== ((f = (p = r == null ? void 0 : r.data) == null ? void 0 : p.current) == null ? void 0 : f.parentId) && K(t == null ? void 0 : t.parentId);
    }, [
      r == null ? void 0 : r.data,
      t == null ? void 0 : t.parentId,
      t == null ? void 0 : t.previousNode,
      K,
      a,
      B,
      h
    ]);
    const xt = T(
      (d) => {
        var nt;
        const { over: p, active: f } = d;
        !a || !t || f.id === (p == null ? void 0 : p.id) && (t == null ? void 0 : t.depth) === ((nt = f.data.current) == null ? void 0 : nt.level) || a && p && g && L && L({
          id: f.id,
          parentId: t.parentId,
          sort: t.position,
          contentComponent: b,
          parentType: t.type
        });
      },
      [a, t, g, L, b]
    ), yt = T(
      (d) => {
        d.active.id === s && (x == null || x({ contentComponent: b, children: c, id: s, label: i, level: l }));
      },
      [c, b, s, i, l, x]
    ), Nt = T(
      (d) => {
        d.active.id === s && document.body.style.setProperty("cursor", g ? "grabbing" : "no-drop");
      },
      [g, s]
    );
    Jt({
      onDragEnd: xt,
      onDragStart: yt,
      onDragMove: Nt
    });
    const U = T(
      (d) => {
        d == null || d.stopPropagation(), m ? I == null || I(s) : u == null || u(s);
      },
      [s, m, u, I]
    ), At = ut(
      (d) => {
        d.stopPropagation(), !(S && d.detail >= 2) && (ct && U(), _ == null || _(s), W == null || W(s));
      },
      S ? 300 : 0
    ), X = R && (r == null ? void 0 : r.id) === R, q = ot.count(c) > 0, { enrichedChildren: F, childrenIds: Et } = pt(() => {
      const d = te(c, { parentId: s, level: l + 1 });
      return {
        enrichedChildren: d,
        childrenIds: d.map((p) => p.props.id)
      };
    }, [c, s, l]), {
      attributes: Tt,
      listeners: vt,
      transform: n,
      transition: Lt,
      setDraggableNodeRef: Rt,
      setDroppableNodeRef: Wt,
      setActivatorNodeRef: _t
    } = zt({
      id: s,
      disabled: !Y,
      data: { type: v, accepts: wt, parentId: R, level: l },
      animateLayoutChanges: se,
      transition: null
    });
    ee(() => {
      if (ot.count(F) === 0) {
        o == null || o(s);
        return;
      }
      if (a || X) {
        o == null || o(s);
        return;
      }
      m ? $ == null || $({ id: s, children: F }) : o == null || o(s);
    }, [a, m, X, F, s]);
    const $t = pt(
      () => lt([
        Zt,
        "tw-cursor-default tw-transition-colors tw-outline-none tw-ring-inset tw-group tw-px-2.5 tw-no-underline tw-leading-5 tw-h-10",
        !a && !w && "active:tw-bg-box-neutral-pressed",
        !a && w ? "tw-font-medium tw-bg-box-neutral-strong tw-text-box-neutral-strong-inverse hover:tw-bg-box-neutral-strong-hover" : "hover:tw-bg-box-neutral tw-text-text",
        n != null && n.y ? "tw-bg-box-neutral-strong-inverse tw-text-text tw-font-normal" : ""
      ]),
      [a, w, n == null ? void 0 : n.y]
    ), Bt = !a, Ft = m && !a, H = Y && !a, Ht = i !== void 0 && !a, E = !a && (M === void 0 ? q : M);
    let z = "";
    !a && !m && E && B && ((dt = e == null ? void 0 : e.previousNode) == null ? void 0 : dt.id) === s && (e == null ? void 0 : e.depth) > ((rt = e == null ? void 0 : e.previousNode) == null ? void 0 : rt.depth) && (z = "tw-border-solid tw-rounded tw-border-2 tw-border-box-selected-strong");
    const Mt = lt([
      "tw-transition-colors tw-flex tw-items-center tw-leading-5 tw-width-full",
      a ? "tw-border-dashed tw-rounded-sm tw-border-2 tw-pr-0 tw-h-12" : "tw-h-10",
      a && (g ? "tw-border-box-selected-strong tw-bg-box-selected-hover" : "tw-bg-box-negative-hover tw-border-box-negative-strong-hover"),
      z
    ]), St = t != null && t.depth ? t.depth * mt : void 0, Vt = a ? 0 : l * mt, Yt = {
      paddingLeft: St ?? Vt
    }, Kt = {
      transform: Ot.Transform.toString(n),
      transition: Lt
    };
    return /* @__PURE__ */ D.createElement(
      "li",
      {
        id: s,
        key: s,
        tabIndex: 0,
        role: "treeitem",
        style: Yt,
        onKeyDown: Qt,
        "aria-label": i,
        "aria-level": l + 1,
        onClick: At,
        className: $t,
        ref: Wt,
        "data-test-id": bt,
        "aria-selected": w,
        "aria-expanded": m,
        "data-has-children": q,
        "aria-owns": Et.join(" ")
      },
      /* @__PURE__ */ D.createElement("div", { ref: Rt, className: Mt, style: Kt }, /* @__PURE__ */ D.createElement(
        kt,
        {
          ...vt,
          ...Tt,
          active: w,
          ref: _t,
          disabled: !H,
          "aria-hidden": !H,
          className: H ? "tw-visible" : "tw-invisible tw-pointer-events-none"
        }
      ), /* @__PURE__ */ D.createElement(
        Ct,
        {
          active: n != null && n.y ? !1 : w,
          onClick: U,
          expanded: Ft,
          disabled: !E,
          "aria-hidden": !E,
          className: E ? "tw-visible" : "tw-invisible tw-pointer-events-none"
        }
      ), Ht && /* @__PURE__ */ D.createElement("span", { className: "first:tw-ml-3.5 tw-w-full tw-h-full tw-flex tw-items-center" }, i), Bt && b)
    );
  }
);
ae.displayName = "FondueTreeItem";
export {
  ae as TreeItem
};
