// @frontify/fondue/dist/components/ScrollWrapper/ScrollWrapper.es.js
import { useFocusRing as f } from "@react-aria/focus";
import e, { useRef as h } from "react";
import "../../styles.css.es.js";
import { useScrollWrapper as g } from "./hooks/useScrollWrapper.es.js";
import { ScrollWrapperDirection as o, scrollWrapperDirections as E } from "./types.es.js";
import { merge as x } from "../../utilities/merge.es.js";
import { FOCUS_STYLE as S } from "../../utilities/focusStyle.es.js";
const n = {
  bottom: "linear-gradient(180deg, rgba(232, 233, 233, 0) 0%, #E8E9E9 100%)",
  top: "linear-gradient(0deg, rgba(232, 233, 233, 0) 0%, #E8E9E9 100%)"
}, W = ({
  tabindex: s = 0,
  direction: r = o.Vertical,
  children: c,
  scrollShadows: l = !0,
  "data-test-id": d = "scroll-wrapper"
}) => {
  const t = h(null), [{ showTopShadow: w, showBottomShadow: p }, m] = g(t), { isFocusVisible: u, focusProps: b } = f(), a = r === o.Vertical || r === o.Both, i = t.current ? t.current.clientWidth + 8 : "100%";
  return /* @__PURE__ */ e.createElement("div", { "data-test-id": d, className: "tw-h-full tw-relative tw-flex-auto tw-flex tw-flex-col tw-min-h-0" }, a && w && l && /* @__PURE__ */ e.createElement(
    "div",
    {
      "data-test-id": "fondue-scrollwrapper-top-shadow",
      className: "tw-h-3 tw-w-full tw-absolute tw-z-10 tw-top-0 tw-left-0 tw-mix-blend-darken tw-border-t tw-border-line",
      style: {
        background: n.top,
        width: i
      }
    }
  ), /* @__PURE__ */ e.createElement(
    "div",
    {
      "data-test-id": "fondue-scrollwrapper-content",
      ref: t,
      tabIndex: s,
      role: "region",
      "aria-label": "Scrollable dialogue content",
      className: x([
        E[r],
        "tw-flex-auto tw-min-h-0 tw-outline-none tw-pt-px tw-pb-2 tw-px-2 tw--mx-2",
        u && S
      ]),
      ...m,
      ...b
    },
    c
  ), a && p && l && /* @__PURE__ */ e.createElement(
    "div",
    {
      "data-test-id": "fondue-scrollwrapper-bottom-shadow",
      className: "tw-h-3 tw-absolute tw-z-10 tw-bottom-0 tw-left-0 tw-mix-blend-darken tw-border-b tw-border-line",
      style: {
        background: n.bottom,
        width: i
      }
    }
  ));
};
W.displayName = "FondueScrollWrapper";
export {
  W as ScrollWrapper
};
