// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/plugins/LinkPlugin/index.es.js
import { Plugin as n } from "@frontify/fondue";
import { createPluginFactory as o, createLinkPlugin as m } from "@udecode/plate";
import { CustomFloatingLink as a } from "./FloatingLink/CustomFloatingLink.es.js";
import { LINK_PLUGIN as r } from "./id.es.js";
import { LinkButton as l } from "./LinkButton.es.js";
import { LinkMarkupElement as s } from "./LinkMarkupElement/index.es.js";
import { isValidUrl as u } from "./utils/url.es.js";
import { BlockStyles as g } from "../styles.es.js";
const k = (i) => o({
  ...m(),
  renderAfterEditable: a,
  options: {
    isUrl: u,
    rangeBeforeOptions: {
      matchString: " ",
      skipInvalid: !0,
      afterMatch: !0
    },
    triggerFloatingLinkHotkeys: "command+k, ctrl+k",
    appBridge: i
  }
})();
class y extends n {
  constructor(t, e = g[r]) {
    super(r, {
      button: l,
      markupElement: new s(),
      ...t
    }), this.styles = {}, this.styles = e, this.appBridge = t == null ? void 0 : t.appBridge;
  }
  plugins() {
    return [k(this.appBridge)];
  }
}
export {
  y as LinkPlugin,
  k as createLinkPlugin
};
