// @frontify/fondue/dist/components/Toast/Toast.es.js
import { INSET_BORDER as c } from "../../utilities/borderStyle.es.js";
import { merge as i } from "../../utilities/merge.es.js";
import { AnimatePresence as d, motion as w } from "framer-motion";
import t from "react";
import { toastStylesBackgroundColorsMap as p, ToastAnimationDirection as m } from "./types.es.js";
const E = "tw-min-h-[4.25rem] tw-bg-base tw-rounded-lg tw-shadow-mid tw-z-[120000]", T = i([E, c]), a = (e) => e === m.BottomToTop ? "150%" : "-150%", S = ({
  isOpen: e,
  style: r,
  icon: s,
  animationDirection: o = m.BottomToTop,
  children: n,
  "data-test-id": l = "toast"
}) => /* @__PURE__ */ t.createElement(d, null, e && /* @__PURE__ */ t.createElement(
  w.div,
  {
    className: T,
    initial: {
      opacity: 0,
      y: a(o)
    },
    animate: { opacity: 1, y: 0 },
    exit: { opacity: 0, y: a(o) },
    "data-test-id": l,
    "aria-live": "polite",
    role: "alert"
  },
  /* @__PURE__ */ t.createElement("div", { className: "tw-flex" }, /* @__PURE__ */ t.createElement(
    "div",
    {
      className: i([
        "tw-flex tw-items-center tw-justify-center tw-min-h-[4.25rem] tw-min-w-[3rem] tw-rounded-l-lg tw-text-white",
        p[r]
      ])
    },
    s
  ), n)
));
S.displayName = "FondueToast";
export {
  S as Toast
};
