// @frontify/fondue/dist/components/RichTextEditor/Plugins/UnderlinePlugin/index.es.js
import { createUnderlinePlugin as r } from "@udecode/plate";
import { UNDERLINE_PLUGIN as n } from "./id.es.js";
import { UnderlineMarkupElement as t, UNDERLINE_CLASSES as o } from "./UnderlineMarkupElement.es.js";
import { UnderlineMarkupElementNode as L } from "./UnderlineMarkupElement.es.js";
import { UnderlineButton as l } from "./UnderlineButton.es.js";
import { Plugin as i } from "../Plugin.es.js";
const m = [
  {
    validNodeName: ["U"]
  },
  {
    validStyle: {
      textDecoration: ["underline"]
    }
  }
];
class E extends i {
  constructor(e) {
    super(n, {
      button: l,
      markupElement: new t(),
      ...e
    });
  }
  plugins() {
    return [
      r({
        deserializeHtml: {
          rules: [...m, { validNodeName: ["SPAN"], validClassName: o }]
        },
        options: {
          hotkey: ["mod+u", "ctrl+u"]
        }
      })
    ];
  }
}
export {
  o as UNDERLINE_CLASSES,
  t as UnderlineMarkupElement,
  L as UnderlineMarkupElementNode,
  E as UnderlinePlugin
};
