// @frontify/fondue/dist/components/FormControl/FormControl.es.js
import { InputLabel as l } from "../InputLabel/InputLabel.es.js";
import { merge as u } from "../../utilities/merge.es.js";
import { Validation as m } from "../../utilities/validation.es.js";
import r, { isValidElement as v, cloneElement as E } from "react";
var g = /* @__PURE__ */ ((t) => (t.Primary = "Primary", t.Positive = "Positive", t.Danger = "Danger", t))(g || {});
const V = {
  Primary: m.Default,
  Positive: m.Success,
  Danger: m.Error
}, f = ({
  text: t,
  disabled: i,
  style: o,
  fullWidth: w = !1,
  "data-test-id": e = "form-control"
}) => {
  let a;
  switch (!0) {
    case i:
      a = "tw-text-black-40";
      break;
    case o === "Danger":
      a = "tw-text-red-60";
      break;
    case o === "Positive":
      a = "tw-text-green-60";
      break;
    default:
      a = "tw-text-black-80";
      break;
  }
  return /* @__PURE__ */ r.createElement(
    "span",
    {
      "data-test-id": `${e}-helper-text`,
      className: `tw-text-s tw-font-sans ${w ? "tw-w-full" : ""} ${a}`
    },
    t
  );
};
f.displayName = "FondueHelperText";
var k = /* @__PURE__ */ ((t) => (t.Before = "Before", t.After = "After", t))(k || {}), N = /* @__PURE__ */ ((t) => (t.Horizontal = "Horizontal", t.Vertical = "Vertical", t))(N || {});
const P = ({
  label: t,
  children: i,
  extra: o,
  name: w,
  helper: e,
  disabled: a,
  clickable: d,
  direction: s = "Vertical",
  style: c = "Primary",
  "data-test-id": n = "form-control"
}) => {
  const x = (e == null ? void 0 : e.position) === "Before";
  return /* @__PURE__ */ r.createElement(
    "div",
    {
      "data-test-id": n,
      "data-name": w,
      className: u([
        "tw-flex tw-items-center tw-gap-2",
        s === "Horizontal" ? "tw-flex-row" : "tw-w-full tw-flex-col"
      ])
    },
    ((t == null ? void 0 : t.children) || o) && /* @__PURE__ */ r.createElement(
      "div",
      {
        className: u([
          "tw-flex tw-flew-row tw-items-center tw-justify-between",
          s === "Vertical" && "tw-w-full"
        ])
      },
      (t == null ? void 0 : t.children) && /* @__PURE__ */ r.createElement(l, { ...t, disabled: a, clickable: d }),
      o && /* @__PURE__ */ r.createElement(
        "span",
        {
          "data-test-id": `${n}-extra`,
          className: "tw-pl-2 tw-text-black-80 tw-font-sans tw-text-s tw-whitespace-nowrap"
        },
        o
      )
    ),
    (e == null ? void 0 : e.text) && x && /* @__PURE__ */ r.createElement(
      f,
      {
        style: c,
        fullWidth: s === "Vertical",
        text: e.text,
        disabled: a,
        "data-test-id": n
      }
    ),
    i && /* @__PURE__ */ r.createElement(
      "div",
      {
        className: s === "Vertical" ? "tw-w-full tw-grid tw-grid-cols-1 tw-gap-5" : ""
      },
      v(i) ? E(i, {
        id: (t == null ? void 0 : t.htmlFor) ?? i.props.id,
        disabled: a,
        validation: V[c]
      }) : i
    ),
    (e == null ? void 0 : e.text) && !x && /* @__PURE__ */ r.createElement(
      f,
      {
        fullWidth: s === "Vertical",
        text: e.text,
        disabled: a,
        style: c,
        "data-test-id": n
      }
    )
  );
};
P.displayName = "FondueFormControl";
export {
  P as FormControl,
  N as FormControlDirection,
  g as FormControlStyle,
  k as HelperPosition
};
