// @frontify/fondue/dist/components/OrderableList/OrderableList.es.js
import t, { useState as b, useEffect as I } from "react";
import { CollectionItem as k } from "./CollectionItem.es.js";
import { HTML5Backend as E } from "react-dnd-html5-backend";
import { DndProvider as y } from "react-dnd";
import { useId as D } from "@react-aria/utils";
import { DropZone as c } from "../DropZone/DropZone.es.js";
import { moveItems as g } from "../../utilities/dnd/moveItems.es.js";
const v = (o, a) => o.sort === null && a.sort === null || o.sort === null ? 1 : a.sort === null ? -1 : o.sort - a.sort, C = ({
  onMove: o,
  dragDisabled: a,
  items: n,
  renderContent: d,
  "data-test-id": i = "orderable-list"
}) => {
  const [l, p] = b(n), s = D();
  I(() => {
    const e = [...n];
    e.sort(v), p(e);
  }, [n]);
  const m = (e, r, f) => {
    const u = g(e, r, f, l);
    o(u);
  };
  return /* @__PURE__ */ t.createElement(y, { backend: E }, /* @__PURE__ */ t.createElement("div", { className: "tw-outline-none", "data-test-id": i }, l.map((e, r) => /* @__PURE__ */ t.createElement(t.Fragment, { key: `dropzone-orderable-list-key-${r}` }, /* @__PURE__ */ t.createElement(
    c,
    {
      key: `orderable-list-item-${r}-before`,
      data: {
        targetItem: e,
        position: "before"
      },
      onDrop: m,
      accept: s
    }
  ), /* @__PURE__ */ t.createElement(
    k,
    {
      key: e.id,
      dragDisabled: a,
      item: e,
      renderContent: d,
      listId: s
    }
  ), r === n.length - 1 && /* @__PURE__ */ t.createElement(
    c,
    {
      key: `orderable-list-item-${r}-after`,
      data: {
        targetItem: e,
        position: "after"
      },
      onDrop: m,
      accept: s
    }
  )))));
};
C.displayName = "FondueOrderableList";
export {
  C as OrderableList
};
