// @frontify/fondue/dist/components/Divider/Divider.es.js
import t from "react";
var o = /* @__PURE__ */ ((e) => (e.NoLine = "noline", e.Dashed = "dashed", e.Solid = "solid", e.Dotted = "dotted", e))(o || {}), i = /* @__PURE__ */ ((e) => (e.Small = "36px", e.Medium = "60px", e.Large = "96px", e))(i || {});
const l = {
  noline: "tw-border-none",
  dashed: "tw-border-dashed",
  solid: "tw-border-solid",
  dotted: "tw-border-dotted"
}, m = "fondue-divider", w = ({
  vertical: e = !1,
  style: r = "solid",
  height: d = "36px",
  "data-test-id": a = m,
  color: s = "#CCC"
}) => {
  const n = `tw-w-px tw-h-full tw-border-r tw-m-0 ${l[r]}`;
  return /* @__PURE__ */ t.createElement(t.Fragment, null, e && /* @__PURE__ */ t.createElement(
    "div",
    {
      "aria-hidden": "true",
      className: "tw-flex tw-self-stretch tw-mt-0 tw-mb-0 tw-items-center tw-justify-center",
      "data-test-id": a,
      style: {
        marginLeft: parseInt(d) / 2,
        marginRight: parseInt(d) / 2
      }
    },
    /* @__PURE__ */ t.createElement(
      "div",
      {
        className: n,
        style: { borderRightColor: s },
        "data-test-id": "fondue-divider-line"
      }
    )
  ), !e && /* @__PURE__ */ t.createElement(
    "div",
    {
      "aria-hidden": "true",
      className: "tw-flex tw-items-center tw-w-full",
      style: { height: d },
      "data-test-id": a
    },
    /* @__PURE__ */ t.createElement(
      "hr",
      {
        className: `tw-border-t tw-m-0 tw-w-full ${l[r]}`,
        style: { borderTopColor: s },
        "data-test-id": "fondue-divider-line"
      }
    )
  ));
};
w.displayName = "FondueDivider";
export {
  w as Divider,
  i as DividerHeight,
  o as DividerStyle
};
