// @frontify/fondue/dist/components/RadioPill/RadioPill.es.js
import { IconSize as m } from "../../foundation/Icon/IconSize.es.js";
import { useFocusRing as c } from "@react-aria/focus";
import { FOCUS_STYLE as n } from "../../utilities/focusStyle.es.js";
import { merge as w } from "../../utilities/merge.es.js";
import t, { cloneElement as p } from "react";
const d = ({
  label: a,
  active: o,
  icon: e,
  onClick: r,
  "data-test-id": l = "radio-pill"
}) => {
  const { isFocusVisible: i, focusProps: s } = c();
  return /* @__PURE__ */ t.createElement(
    "button",
    {
      "data-test-id": l,
      type: "button",
      className: w([
        "tw-inline-flex tw-items-center tw-rounded-full tw-text-xs tw-px-2 tw-py-1",
        o ? "dark:tw-bg-black-5 dark:tw-text-black-100 tw-bg-black-100 tw-text-white" : "tw-text-black-80 dark:tw-text-black-40",
        i && n,
        r ? "tw-cursor-pointer" : "tw-cursor-auto"
      ]),
      onClick: r,
      ...s
    },
    e && /* @__PURE__ */ t.createElement("span", { className: "tw-mr-1" }, p(e, { size: m.Size16 })),
    /* @__PURE__ */ t.createElement("span", null, a)
  );
};
d.displayName = "FondueRadioPill";
export {
  d as RadioPill
};
