// @frontify/guideline-blocks-settings/dist/components/BlockItemWrapper/BlockItemWrapper.es.js
import { jsxs as g, jsx as r } from "react/jsx-runtime";
import { useState as a, useRef as F, useEffect as b } from "react";
import { Toolbar as h } from "./Toolbar.es.js";
import { joinClassNames as c } from "../../utilities/react/joinClassNames.es.js";
const E = ({
  children: l,
  toolbarFlyoutItems: p,
  toolbarItems: e,
  shouldHideWrapper: w,
  shouldHideComponent: n = !1,
  isDragging: u,
  shouldFillContainer: f,
  outlineOffset: o = 2,
  shouldBeShown: i = !1
}) => {
  const [t, m] = a(i), [y, s] = a(!1), v = F(null);
  if (b(() => {
    t || s(!0);
  }, [t]), w)
    return l;
  const x = e == null ? void 0 : e.filter((d) => d !== void 0);
  return /* @__PURE__ */ g(
    "div",
    {
      ref: v,
      onFocus: () => s(!1),
      onPointerEnter: () => s(!1),
      "data-test-id": "block-item-wrapper",
      style: {
        outlineOffset: o
      },
      className: c([
        "tw-relative tw-group tw-outline-1 tw-outline-box-selected-inverse",
        f && "tw-flex-1 tw-h-full tw-w-full",
        "hover:tw-outline focus-within:tw-outline",
        (t || i) && "tw-outline",
        n && "tw-opacity-0"
      ]),
      children: [
        /* @__PURE__ */ r(
          "div",
          {
            style: {
              right: -1 - o,
              bottom: `calc(100% - ${2 + o}px)`
            },
            className: c([
              "tw-pointer-events-none tw-absolute tw-bottom-[calc(100%-4px)] tw-right-[-3px] tw-w-full tw-opacity-0 tw-z-10",
              "group-hover:tw-opacity-100 group-focus:tw-opacity-100 focus-within:tw-opacity-100",
              (t || i) && "tw-opacity-100"
            ]),
            children: /* @__PURE__ */ r(
              h,
              {
                isFlyoutOpen: t,
                isFlyoutDisabled: y,
                setIsFlyoutOpen: m,
                flyoutItems: p,
                items: x,
                isDragging: u
              }
            )
          }
        ),
        l
      ]
    }
  );
};
export {
  E as BlockItemWrapper
};
