// @frontify/fondue/dist/components/DialogHeader/DialogHeader.es.js
import e from "react";
import { merge as l } from "../../utilities/merge.es.js";
import { dialogPaddingMap as c } from "../../types/dialog.es.js";
import { FOCUS_VISIBLE_STYLE as d } from "../../utilities/focusStyle.es.js";
import { Flex as s } from "../Flex/Flex.es.js";
import { Text as p } from "../../typography/Text/Text.es.js";
import w from "../../foundation/Icon/Generated/IconCross.es.js";
import { Box as f } from "../Box/Box.es.js";
const g = ({
  title: i,
  collapseBottom: n = !1,
  size: r = "medium",
  onClose: a,
  icon: o,
  badge: m,
  "data-test-id": t = "fondue-dialog-header"
}) => /* @__PURE__ */ e.createElement(
  f,
  {
    "data-test-id": t,
    className: l([
      "tw-px-6 tw-flex tw-items-center tw-border-b tw-border-b-line",
      c[r],
      n && "tw-pb-0 tw-border-none"
    ])
  },
  /* @__PURE__ */ e.createElement(s, { alignItems: "center", "data-test-id": `${t}-title` }, o && /* @__PURE__ */ e.createElement("span", { "data-test-id": `${t}-icon`, className: "tw-mr-1.5" }, o), /* @__PURE__ */ e.createElement(p, { weight: "x-strong", size: r }, i), m && /* @__PURE__ */ e.createElement("div", { "data-test-id": `${t}-badge`, className: "tw-ml-1.5" }, m)),
  a && /* @__PURE__ */ e.createElement(
    "button",
    {
      "data-test-id": `${t}-close`,
      "aria-label": `${t}-close`,
      onClick: a,
      className: l([d, "tw-rounded tw-absolute tw-right-5"])
    },
    /* @__PURE__ */ e.createElement(w, null)
  )
);
g.displayName = "FondueDialogHeader";
export {
  g as DialogHeader
};
