// @frontify/fondue/dist/components/ActionMenu/ActionMenu/ActionMenu.es.js
import { AriaList as I } from "../Aria/AriaList.es.js";
import { AriaMenuItem as M } from "../Aria/AriaMenuItem.es.js";
import { AriaSection as b } from "../Aria/AriaSection.es.js";
import { getMenuItems as k, getKeyItemRecord as R, mapToAriaProps as g, getDisabledItemIds as E } from "../Aria/helper.es.js";
import { useMenu as K } from "@react-aria/menu";
import { useTreeState as L } from "@react-stately/tree";
import o, { useRef as P } from "react";
const F = ({
  menuBlocks: r,
  ariaLabel: i = "Action Menu",
  focus: c,
  border: u = !0,
  scrollable: p = !1,
  onClick: l
}) => {
  const n = k(r), d = R(n), a = g(i, r), e = L({
    ...a,
    selectionMode: "none",
    disabledKeys: E(n)
  }), m = P(null), { menuProps: f } = K({ ...a, autoFocus: c }, e, m);
  return /* @__PURE__ */ o.createElement(I, { ariaProps: { ...f }, ref: m, border: u, scrollable: p }, [...e.collection].map((s) => {
    const { key: y, "aria-label": A } = s;
    return /* @__PURE__ */ o.createElement(b, { key: y, ariaLabel: A }, [...s.childNodes].map((t) => /* @__PURE__ */ o.createElement(
      M,
      {
        key: t.key,
        menuItem: d[t.key],
        state: e,
        node: t,
        onClick: l
      }
    )));
  }));
};
F.displayName = "FondueActionMenu";
export {
  F as ActionMenu
};
