// @frontify/fondue/dist/components/MenuItem/MenuItemContent.es.js
import e, { cloneElement as c } from "react";
import { IconSize as s } from "../../foundation/Icon/IconSize.es.js";
import { merge as l } from "../../utilities/merge.es.js";
import { MenuItemContentSize as t } from "./types.es.js";
const o = {
  [t.XSmall]: "tw-pr-1.5",
  [t.Small]: "tw-pr-1.5",
  [t.Large]: "tw-pr-2"
}, p = ({
  title: n,
  decorator: i,
  switchComponent: r,
  subtitle: m,
  ariaProps: w,
  size: a = t.Small,
  children: d
}) => /* @__PURE__ */ e.createElement(
  "div",
  {
    ...w,
    "data-test-id": "menu-item-content",
    className: "tw-flex tw-box-border tw-items-center tw-font-sans tw-text-s tw-flex-auto tw-min-w-0"
  },
  i && /* @__PURE__ */ e.createElement("span", { className: l(["tw-shrink-0", o[a]]), "data-test-id": "menu-item-decorator" }, c(i, {
    size: a === t.Large ? s.Size24 : s.Size16
  })),
  /* @__PURE__ */ e.createElement("div", { className: "tw-flex-1 tw-overflow-hidden tw-text-ellipsis tw-whitespace-nowrap" }, /* @__PURE__ */ e.createElement(
    "div",
    {
      "data-test-id": "menu-item-title",
      className: "tw-select-none tw-text-ellipsis tw-overflow-hidden",
      title: typeof n == "string" ? n : ""
    },
    d || n
  ), m && a === t.Large && /* @__PURE__ */ e.createElement("div", { "data-test-id": "menu-item-subtitle", className: "tw-select-none tw-text-xxs", title: m }, m)),
  r && /* @__PURE__ */ e.createElement("span", { className: l(["tw-shrink-0", o[a]]), "data-test-id": "menu-item-decorator" }, r)
);
p.displayName = "FondueMenuItemContent";
export {
  p as MenuItemContent
};
