// @frontify/fondue/dist/components/Modal/ModalVisual.es.js
import e from "react";
import { merge as m } from "../../utilities/merge.es.js";
import { FrontifyPattern as i } from "../FrontifyPattern/FrontifyPattern.es.js";
import { patternThemes as s } from "../../foundation/Pattern/PatternTheme.es.js";
import { PatternScale as l } from "../../foundation/Pattern/PatternScale.es.js";
import { PatternDesign as o } from "../../foundation/Pattern/PatternDesigns.es.js";
const r = ({
  pattern: a = o.DigitalAssets,
  foregroundColor: t
}) => /* @__PURE__ */ e.createElement(
  "div",
  {
    "data-test-id": "modal-visual",
    className: m([
      "tw-h-[1802px] tw-absolute tw-bottom-0",
      t && s[t]
    ]),
    style: {
      clipPath: "path('M0 0L252 145.492C256.95 148.35 260 153.632 260 159.349V1641.98C260 1647.7 256.95 1652.98 252 1655.84L0 1801.33V1501.11V0Z')"
    }
  },
  /* @__PURE__ */ e.createElement("div", { className: "tw-absolute tw-bottom-0" }, /* @__PURE__ */ e.createElement(
    i,
    {
      pattern: a,
      scaleOrigin: ["left", "bottom"],
      scale: l.XXL,
      foregroundColor: t
    }
  ))
);
r.displayName = "FondueModalVisual";
export {
  r as ModalVisual
};
