// @frontify/fondue/dist/components/Modal/Modal.es.js
import t, { memo as h, useRef as E } from "react";
import { merge as M } from "../../utilities/merge.es.js";
import { ModalWidth as a } from "./types.es.js";
import { ModalVisual as N } from "./ModalVisual.es.js";
import { useOverlay as D, usePreventScroll as P, useModal as g, OverlayContainer as A } from "@react-aria/overlays";
import { FocusScope as b } from "@react-aria/focus";
import { useDialog as F } from "@react-aria/dialog";
import { motion as n, AnimatePresence as L } from "framer-motion";
import { ModalTitle as I } from "./context/ModalTitle.es.js";
import { ModalHeader as O } from "./ModalHeader.es.js";
import { ModalBody as R } from "./ModalBody.es.js";
import { ModalFooter as S } from "./ModalFooter.es.js";
import { ModalLayout as T, MODAL_PADDING as m } from "./context/ModalLayout.es.js";
const _ = {
  initial: {
    opacity: 0
  },
  show: {
    opacity: 1,
    transition: {
      duration: 0.15
    }
  },
  exit: {
    opacity: 0,
    transition: {
      duration: 0.15
    }
  }
}, V = {
  initial: { y: -200 },
  show: { y: 0, transition: { duration: 0.3 } },
  exit: { y: 0 }
}, k = {
  [a.Default]: "tw-max-w-[800px]",
  [a.Small]: "tw-max-w-[600px]",
  [a.Large]: "tw-max-w-[1200px]"
}, B = 50, d = h((o) => {
  const { visual: e, children: s, width: w = a.Default, zIndex: c = B, compact: i = !1 } = o, l = E(null), {
    overlayProps: f,
    underlayProps: { onPointerDown: u }
  } = D(o, l);
  P();
  const { modalProps: p } = g(), { dialogProps: x, titleProps: y } = F(o, l), v = i ? m.compact : m.default;
  return /* @__PURE__ */ t.createElement(
    n.div,
    {
      variants: _,
      initial: "initial",
      animate: "show",
      exit: "exit",
      style: {
        background: "rgba(0, 0, 0, .5)",
        zIndex: c
      },
      onPointerDown: u,
      "data-is-underlay": !0,
      className: "tw-fixed tw-top-0 tw-left-0 tw-bottom-0 tw-right-0 tw-flex tw-justify-center tw-items-center tw-p-4"
    },
    /* @__PURE__ */ t.createElement(b, { contain: !0, restoreFocus: !0, autoFocus: !0 }, /* @__PURE__ */ t.createElement(
      n.div,
      {
        variants: V,
        className: M([k[w], "tw-w-full tw-max-h-full tw-h-contents tw-flex tw-flex-col"])
      },
      /* @__PURE__ */ t.createElement(
        "div",
        {
          ...f,
          ...x,
          ...p,
          ref: l,
          "data-test-id": "modal-container",
          className: "tw-flex-initial tw-min-h-0 tw-w-full tw-flex tw-bg-white tw-overflow-hidden tw-border tw-border-solid tw-border-line-strong tw-rounded tw-shadow-2xl"
        },
        (e == null ? void 0 : e.pattern) && /* @__PURE__ */ t.createElement("div", { className: "tw-w-[260px] tw-relative tw-flex-shrink-0 tw-overflow-hidden" }, /* @__PURE__ */ t.createElement(N, { ...e })),
        /* @__PURE__ */ t.createElement("div", { className: "tw-flex tw-flex-col tw-flex-1 tw-space-y-6 tw-overflow-hidden" }, /* @__PURE__ */ t.createElement(T.Provider, { value: { compact: i, padding: v } }, /* @__PURE__ */ t.createElement(I.Provider, { value: y }, s)))
      )
    ))
  );
});
d.displayName = "Modal";
const r = ({ isOpen: o, ...e }) => /* @__PURE__ */ t.createElement(A, null, /* @__PURE__ */ t.createElement(L, null, o && /* @__PURE__ */ t.createElement(d, { ...e, isOpen: !0 })));
r.displayName = "FondueModal";
r.Header = O;
r.Body = R;
r.Footer = S;
export {
  r as Modal
};
