// @frontify/fondue/dist/components/AssetInput/SingleAsset/SelectedAsset.es.js
import { ActionMenu as I } from "../../ActionMenu/ActionMenu/ActionMenu.es.js";
import { IconSize as P } from "../../../foundation/Icon/IconSize.es.js";
import { useMemoizedId as b } from "../../../hooks/useMemoizedId.es.js";
import { useButton as B } from "@react-aria/button";
import { useFocusRing as D, FocusScope as z } from "@react-aria/focus";
import { useMenuTrigger as M } from "@react-aria/menu";
import { useOverlay as W, DismissButton as h } from "@react-aria/overlays";
import { mergeProps as j } from "@react-aria/utils";
import { useMenuTriggerState as L } from "@react-stately/menu";
import { FOCUS_STYLE as Y } from "../../../utilities/focusStyle.es.js";
import { merge as s } from "../../../utilities/merge.es.js";
import { AnimatePresence as U, motion as V } from "framer-motion";
import t, { useRef as g, useState as _, useEffect as $ } from "react";
import { AssetInputSize as c } from "../AssetInput.es.js";
import { AssetThumbnail as q } from "../AssetThumbnail.es.js";
import { AssetSubline as G } from "./AssetSubline.es.js";
import { SpinningCircle as H } from "./SpinningCircle.es.js";
import J from "../../../foundation/Icon/Generated/IconCaretDown.es.js";
const K = ({
  asset: e,
  size: n,
  actions: x,
  isLoading: m,
  hideSize: E = !1,
  hideExtension: v = !1
}) => {
  const k = b(), w = b(), r = g(null), o = L({ closeOnSelect: !0 }), { isOpen: l, focusStrategy: y } = o, { menuTriggerProps: S } = M({}, o, r), { buttonProps: N } = B(S, r), { isFocusVisible: i, focusProps: A } = D(), u = g(null), { overlayProps: F } = W(
    { onClose: () => o.close(), shouldCloseOnBlur: !0, isOpen: l, isDismissable: !0 },
    u
  ), f = (e == null ? void 0 : e.name) || "Your Asset", [C, O] = _(0);
  return $(() => {
    let a = null;
    const R = () => {
      var p;
      const T = ((p = r.current) == null ? void 0 : p.getBoundingClientRect().width) ?? 0;
      a = window.setTimeout(() => O(T), 0);
    }, d = new ResizeObserver(R);
    return r.current && d.observe(r.current), () => {
      a && clearTimeout(a), d.disconnect();
    };
  }, []), /* @__PURE__ */ t.createElement(
    "div",
    {
      className: "tw-font-sans tw-w-full tw-text-s tw-bg-transparent tw-font-normal tw-min-w-0",
      "aria-labelledby": w,
      title: f,
      "data-test-id": "asset-single-input"
    },
    /* @__PURE__ */ t.createElement(
      "button",
      {
        ...j(N, A),
        ref: r,
        className: s([
          "tw-w-full tw-flex tw-border tw-rounded tw-overflow-hidden hover:tw-border-black-90 dark:hover:tw-border-black-40 focus-visible:tw-outline-none",
          i && Y,
          n === c.Large ? "tw-h-[11.5rem] tw-flex-col" : "tw-h-14",
          l || i ? "tw-border-black-90 dark:tw-border-black-10" : "tw-border-black-20 dark:tw-border-black-80"
        ])
      },
      m && !e && /* @__PURE__ */ t.createElement(
        "div",
        {
          className: s([
            "tw-flex tw-justify-center tw-items-center",
            n === c.Large ? "tw-w-full tw-h-32" : "tw-w-14 tw-h-full"
          ])
        },
        /* @__PURE__ */ t.createElement(H, { size: n })
      ),
      e && /* @__PURE__ */ t.createElement(q, { asset: e, size: n, isActive: l || i }),
      /* @__PURE__ */ t.createElement(
        "div",
        {
          className: s([
            "tw-min-w-0 tw-max-w-full tw-flex tw-flex-auto tw-self-stretch tw-border-black-100 tw-border-opacity-25",
            n === c.Large ? "tw-h-14 tw-border-t" : "tw-h-full tw-border-l"
          ])
        },
        /* @__PURE__ */ t.createElement("div", { className: "tw-min-w-0 tw-pr-3 tw-pl-4 tw-flex tw-flex-auto tw-flex-col tw-items-start tw-justify-center tw-h-full" }, /* @__PURE__ */ t.createElement(
          "span",
          {
            id: w,
            className: s([
              "tw-max-w-full tw-text-black-100 tw-text-s tw-truncate dark:tw-text-white",
              (l || i) && "tw-font-medium"
            ])
          },
          f
        ), /* @__PURE__ */ t.createElement(
          G,
          {
            asset: e,
            isLoading: m,
            hideSize: E,
            hideExtension: v
          }
        )),
        /* @__PURE__ */ t.createElement("div", { className: "tw-p-4 tw-flex tw-flex-none tw-items-center tw-justify-center" }, /* @__PURE__ */ t.createElement(
          "span",
          {
            className: s([
              "tw-transition-transform",
              l ? "tw-rotate-180 tw-text-black-90" : "tw-text-black-60"
            ])
          },
          /* @__PURE__ */ t.createElement(J, { size: P.Size16 })
        ))
      )
    ),
    /* @__PURE__ */ t.createElement(U, null, l && /* @__PURE__ */ t.createElement(
      V.div,
      {
        style: {
          width: C
        },
        className: "tw-absolute tw-left-auto tw-w-full tw-overflow-hidden tw-box-border tw-p-0 tw-shadow-mid tw-list-none tw-m-0 tw-mt-2 tw-z-20",
        "data-test-id": "asset-single-input-flyout",
        key: `asset-input-menu-${k}`,
        initial: { height: 0 },
        animate: { height: "auto" },
        exit: { height: 0 },
        transition: { ease: [0.04, 0.62, 0.23, 0.98], duration: 0.5 }
      },
      /* @__PURE__ */ t.createElement(z, { restoreFocus: !0 }, /* @__PURE__ */ t.createElement("div", { ...F, ref: u }, /* @__PURE__ */ t.createElement(h, { onDismiss: () => o.close() }), /* @__PURE__ */ t.createElement(
        I,
        {
          menuBlocks: x,
          focus: y,
          onClick: () => o.close()
        }
      ), /* @__PURE__ */ t.createElement(h, { onDismiss: () => o.close() })))
    ))
  );
};
K.displayName = "FondueSelectedAsset";
export {
  K as SelectedAsset
};
