// @frontify/app-bridge/dist/index.es.js
import { enableMapSet as e } from "immer";
import { AppBridgeCreateAsset as m } from "./AppBridgeCreateAsset.es.js";
import { HttpClientError as u } from "./errors/HttpClientError.es.js";
import { FetchError as a } from "./errors/NotifyError.es.js";
import { TimeoutReachedError as f } from "./errors/TimeoutReachedError.es.js";
import { closeAssetChooser as n, openAssetChooser as D } from "./registries/commands/AssetChooser.es.js";
import { openAssetViewer as y } from "./registries/commands/AssetViewer.es.js";
import { openNavigationManager as g } from "./registries/commands/NavigationManager.es.js";
import { closeTemplateChooser as C, openTemplateChooser as T } from "./registries/commands/TemplateChooser.es.js";
import { useAssetChooser as d } from "./react/useAssetChooser.es.js";
import { useAssetUpload as h } from "./react/useAssetUpload.es.js";
import { useAssetViewer as b } from "./react/useAssetViewer.es.js";
import { useBlockAssets as U } from "./react/useBlockAssets.es.js";
import { useBlockTemplates as R } from "./react/useBlockTemplates.es.js";
import { useBlockSettings as L } from "./react/useBlockSettings.es.js";
import { useBrandportalLink as j } from "./react/useBrandportalLink.es.js";
import { BulkDownloadState as F, useBulkDownload as G } from "./react/useBulkDownload.es.js";
import { useCategorizedDocumentPages as I } from "./react/useCategorizedDocumentPages.es.js";
import { useColorPalettes as N } from "./react/useColorPalettes.es.js";
import { useColors as z } from "./react/useColors.es.js";
import { useCoverPage as q } from "./react/useCoverPage.es.js";
import { useDocumentCategories as K } from "./react/useDocumentCategories.es.js";
import { useDocumentGroups as W } from "./react/useDocumentGroups.es.js";
import { useDocumentPageTargets as Y } from "./react/useDocumentPageTargets.es.js";
import { useDocumentSection as $ } from "./react/useDocumentSection.es.js";
import { useDocumentTargets as oe } from "./react/useDocumentTargets.es.js";
import { useEditorState as te } from "./react/useEditorState.es.js";
import { useFileInput as pe } from "./react/useFileInput.es.js";
import { useFileUpload as se } from "./react/useFileUpload.es.js";
import { useGroupedDocuments as ie } from "./react/useGroupedDocuments.es.js";
import { useGuidelineActions as xe } from "./react/useGuidelineActions.es.js";
import { useNavigationManager as De } from "./react/useNavigationManager.es.js";
import { usePageTemplateSettings as ye } from "./react/usePageTemplateSettings.es.js";
import { usePrivacySettings as ge } from "./react/usePrivacySettings.es.js";
import { useReadyForPrint as Ce } from "./react/useReadyForPrint.es.js";
import { useTemplateAssets as Pe } from "./react/useTemplateAssets.es.js";
import { useThemeAssets as Se } from "./react/useThemeAssets.es.js";
import { useThemeSettings as Be } from "./react/useThemeSettings.es.js";
import { useTemplateChooser as ke } from "./react/useTemplateChooser.es.js";
import { useUncategorizedDocumentPages as Ee } from "./react/useUncategorizedDocumentPages.es.js";
import { useUngroupedDocuments as ve } from "./react/useUngroupedDocuments.es.js";
import { getAppBridgeBlockStub as Oe } from "./tests/AppBridgeBlockStub.es.js";
import { getAppBridgePlatformAppStub as we } from "./tests/AppBridgePlatformAppStub.es.js";
import { getAppBridgeThemeStub as Ge } from "./tests/AppBridgeThemeStub.es.js";
import { AssetApiDummy as Ie } from "./tests/AssetApiDummy.es.js";
import { AssetChooserOptionsDummy as Ne } from "./tests/AssetChooserOptionsDummy.es.js";
import { AssetChooserResultDummy as ze } from "./tests/AssetChooserResultDummy.es.js";
import { AssetDummy as qe } from "./tests/AssetDummy.es.js";
import { BrandportalLinkApiDummy as Ke } from "./tests/BrandportalLinkApiDummy.es.js";
import { BrandportalLinkDummy as We } from "./tests/BrandportalLinkDummy.es.js";
import { BulkDownloadDummy as Ye } from "./tests/BulkDownloadDummy.es.js";
import { ColorApiCreateDummy as $e, ColorApiDummy as eo, ColorApiPatchDummy as oo } from "./tests/ColorApiDummy.es.js";
import { ColorCreateDummy as to, ColorDummy as mo, ColorPatchDummy as po } from "./tests/ColorDummy.es.js";
import { ColorPaletteApiCreateDummy as so, ColorPaletteApiDummy as ao, ColorPaletteApiPatchDummy as io } from "./tests/ColorPaletteApiDummy.es.js";
import { ColorPaletteCreateDummy as xo, ColorPaletteDummy as no, ColorPalettePatchDummy as Do } from "./tests/ColorPaletteDummy.es.js";
import { CoverPageApiDummy as yo } from "./tests/CoverPageApiDummy.es.js";
import { CoverPageDummy as go } from "./tests/CoverPageDummy.es.js";
import { DocumentApiDummy as Co } from "./tests/DocumentApiDummy.es.js";
import { DocumentCategoryApiDummy as Po } from "./tests/DocumentCategoryApiDummy.es.js";
import { DocumentCategoryDummy as ho } from "./tests/DocumentCategoryDummy.es.js";
import { DocumentDummy as bo } from "./tests/DocumentDummy.es.js";
import { DocumentGroupApiDummy as Uo } from "./tests/DocumentGroupApiDummy.es.js";
import { DocumentGroupDummy as Ro } from "./tests/DocumentGroupDummy.es.js";
import { DocumentPageApiDummy as Lo, DocumentPageDuplicateApiDummy as Oo } from "./tests/DocumentPageApiDummy.es.js";
import { DocumentPageDummy as wo, DocumentPageDuplicateDummy as Fo } from "./tests/DocumentPageDummy.es.js";
import { DocumentSectionApiDummy as Mo } from "./tests/DocumentSectionApiDummy.es.js";
import { DocumentSectionDummy as Ho } from "./tests/DocumentSectionDummy.es.js";
import { GuidelineSearchResultApiDummy as Vo } from "./tests/GuidelineSearchResultApiDummy.es.js";
import { GuidelineSearchResultDummy as _o } from "./tests/GuidelineSearchResultDummy.es.js";
import { HttpUtilResponseDummy as Jo } from "./tests/HttpUtilResponseDummy.es.js";
import { DocumentPageTargetsApiDummy as Qo, DocumentTargetsApiDummy as Wo, SingleTargetApiDummy as Xo, UpdateTargetsApiDummy as Yo } from "./tests/TargetsApiDummy.es.js";
import { DocumentPageTargetsDummy as $o, DocumentTargetsDummy as er, UpdateTargetsDummy as or } from "./tests/TargetsDummy.es.js";
import { TemplateApiLegacyDummy as tr } from "./tests/TemplateApiLegacyDummy.es.js";
import { TemplateLegacyDummy as pr } from "./tests/TemplateLegacyDummy.es.js";
import { TemplateDummy as sr } from "./tests/TemplateDummy.es.js";
import { UserApiDummy as ir } from "./tests/UserApiDummy.es.js";
import { UserDummy as xr } from "./tests/UserDummy.es.js";
import { withAppBridgeBlockStubs as Dr } from "./tests/withAppBridgeBlockStubs.es.js";
import { withAppBridgeThemeStubs as yr } from "./tests/withAppBridgeThemeStubs.es.js";
import { LinkSettingsDisplay as gr, LinkSettingsIconPosition as Ar, LinkType as Cr } from "./types/Document.es.js";
import { DocumentPageVisibility as Pr } from "./types/DocumentPage.es.js";
import { FileExtension as Sr } from "./types/FileExtension.es.js";
import { FileExtensionSets as Br } from "./types/FileExtensionSets.es.js";
import { FileType as kr } from "./types/FileType.es.js";
import { AssetChooserObjectType as Er, AssetChooserProjectType as Rr, TerrificEvent as vr } from "./types/Terrific.es.js";
import { Topic as Or } from "./types/Topic.es.js";
import { rgbObjectToRgbString as wr, rgbStringToRgbObject as Fr } from "./utilities/color.es.js";
import { getDatasetByClassName as Mr, getDatasetByElement as Ir } from "./utilities/dom.es.js";
import { getMimeType as Nr, mimeTypes as Vr } from "./utilities/file.es.js";
import { generateRandomString as _r } from "./utilities/hash.es.js";
import { HttpClient as Jr } from "./utilities/httpClient.es.js";
import { notify as Qr } from "./utilities/notify.es.js";
import { compareObjects as Xr, convertObjectCase as Yr, isObject as Zr, mergeDeep as $r } from "./utilities/object.es.js";
import { SUBSCRIBE_TIMEOUT as ot, subscribe as rt } from "./utilities/subscribe.es.js";
e();
export {
  m as AppBridgeCreateAsset,
  Ie as AssetApiDummy,
  Er as AssetChooserObjectType,
  Ne as AssetChooserOptionsDummy,
  Rr as AssetChooserProjectType,
  ze as AssetChooserResultDummy,
  qe as AssetDummy,
  Ke as BrandportalLinkApiDummy,
  We as BrandportalLinkDummy,
  Ye as BulkDownloadDummy,
  F as BulkDownloadState,
  $e as ColorApiCreateDummy,
  eo as ColorApiDummy,
  oo as ColorApiPatchDummy,
  to as ColorCreateDummy,
  mo as ColorDummy,
  so as ColorPaletteApiCreateDummy,
  ao as ColorPaletteApiDummy,
  io as ColorPaletteApiPatchDummy,
  xo as ColorPaletteCreateDummy,
  no as ColorPaletteDummy,
  Do as ColorPalettePatchDummy,
  po as ColorPatchDummy,
  yo as CoverPageApiDummy,
  go as CoverPageDummy,
  Co as DocumentApiDummy,
  Po as DocumentCategoryApiDummy,
  ho as DocumentCategoryDummy,
  bo as DocumentDummy,
  Uo as DocumentGroupApiDummy,
  Ro as DocumentGroupDummy,
  Lo as DocumentPageApiDummy,
  wo as DocumentPageDummy,
  Oo as DocumentPageDuplicateApiDummy,
  Fo as DocumentPageDuplicateDummy,
  Qo as DocumentPageTargetsApiDummy,
  $o as DocumentPageTargetsDummy,
  Pr as DocumentPageVisibility,
  Mo as DocumentSectionApiDummy,
  Ho as DocumentSectionDummy,
  Wo as DocumentTargetsApiDummy,
  er as DocumentTargetsDummy,
  a as FetchError,
  Sr as FileExtension,
  Br as FileExtensionSets,
  kr as FileType,
  Vo as GuidelineSearchResultApiDummy,
  _o as GuidelineSearchResultDummy,
  Jr as HttpClient,
  u as HttpClientError,
  Jo as HttpUtilResponseDummy,
  gr as LinkSettingsDisplay,
  Ar as LinkSettingsIconPosition,
  Cr as LinkType,
  ot as SUBSCRIBE_TIMEOUT,
  Xo as SingleTargetApiDummy,
  tr as TemplateApiLegacyDummy,
  sr as TemplateDummy,
  pr as TemplateLegacyDummy,
  vr as TerrificEvent,
  f as TimeoutReachedError,
  Or as Topic,
  Yo as UpdateTargetsApiDummy,
  or as UpdateTargetsDummy,
  ir as UserApiDummy,
  xr as UserDummy,
  n as closeAssetChooser,
  C as closeTemplateChooser,
  Xr as compareObjects,
  Yr as convertObjectCase,
  _r as generateRandomString,
  Oe as getAppBridgeBlockStub,
  we as getAppBridgePlatformAppStub,
  Ge as getAppBridgeThemeStub,
  Mr as getDatasetByClassName,
  Ir as getDatasetByElement,
  Nr as getMimeType,
  Zr as isObject,
  $r as mergeDeep,
  Vr as mimeTypes,
  Qr as notify,
  D as openAssetChooser,
  y as openAssetViewer,
  g as openNavigationManager,
  T as openTemplateChooser,
  wr as rgbObjectToRgbString,
  Fr as rgbStringToRgbObject,
  rt as subscribe,
  d as useAssetChooser,
  h as useAssetUpload,
  b as useAssetViewer,
  U as useBlockAssets,
  L as useBlockSettings,
  R as useBlockTemplates,
  j as useBrandportalLink,
  G as useBulkDownload,
  I as useCategorizedDocumentPages,
  N as useColorPalettes,
  z as useColors,
  q as useCoverPage,
  K as useDocumentCategories,
  W as useDocumentGroups,
  Y as useDocumentPageTargets,
  $ as useDocumentSection,
  oe as useDocumentTargets,
  te as useEditorState,
  pe as useFileInput,
  se as useFileUpload,
  ie as useGroupedDocuments,
  xe as useGuidelineActions,
  De as useNavigationManager,
  ye as usePageTemplateSettings,
  ge as usePrivacySettings,
  Ce as useReadyForPrint,
  Pe as useTemplateAssets,
  ke as useTemplateChooser,
  Se as useThemeAssets,
  Be as useThemeSettings,
  Ee as useUncategorizedDocumentPages,
  ve as useUngroupedDocuments,
  Dr as withAppBridgeBlockStubs,
  yr as withAppBridgeThemeStubs
};
