// @frontify/fondue/dist/components/Flyout/Flyout.es.js
import { watchModals as V } from "@react-aria/aria-modal-polyfill";
import { useButton as Y } from "@react-aria/button";
import { useFocusRing as H, FocusScope as N } from "@react-aria/focus";
import { useOverlayTrigger as G, OverlayContainer as W } from "@react-aria/overlays";
import { mergeProps as j } from "@react-aria/utils";
import { useOverlayTriggerState as k } from "@react-stately/overlays";
import { FOCUS_STYLE as q } from "../../utilities/focusStyle.es.js";
import { merge as z } from "../../utilities/merge.es.js";
import t, { useRef as i, useEffect as A } from "react";
import { useContainScroll as J } from "./hooks/useContainScroll.es.js";
import { useOverlayPositionWithBottomMargin as K } from "./hooks/useOverlayPositionWithBottomMargin.es.js";
import { Overlay as Q } from "./Overlay.es.js";
import { LegacyFlyoutFooter as X } from "./FlyoutFooter.es.js";
const uo = "#eaebeb", go = "10px";
var Z = /* @__PURE__ */ ((o) => (o.Top = "top", o.Bottom = "bottom", o.TopLeft = "top left", o.BottomLeft = "bottom left", o.TopRight = "top right", o.BottomRight = "bottom right", o.Right = "right", o.Left = "left", o))(Z || {});
const $ = ({
  trigger: o,
  isTriggerDisabled: a = !1,
  decorator: p,
  onConfirm: c,
  onCancel: u,
  children: g,
  onOpenChange: d,
  isOpen: e = !1,
  title: T = "",
  badges: b = [],
  hug: v = !0,
  fitContent: R = !1,
  fixedHeader: y,
  fixedFooter: E,
  contentMinHeight: h,
  legacyFooter: F = !0,
  placement: L = "bottom left",
  offset: w,
  updatePositionOnContentChange: C = !1
}) => {
  const n = k({ isOpen: e, onOpenChange: d }), { toggle: D, close: O } = n, r = i(null), s = i(null), l = i(null), { isFocusVisible: m, focusProps: B } = H(), { triggerProps: I, overlayProps: S } = G(
    { type: "dialog" },
    n,
    r
  ), { positionProps: _ } = K({
    triggerRef: r,
    overlayRef: s,
    scrollRef: l,
    isOpen: e,
    placement: L,
    offset: w,
    updatePositionOnContentChange: C
  }), { buttonProps: x, isPressed: P } = Y(
    { onPress: () => D(), elementType: "div", isDisabled: a },
    r
  );
  A(() => {
    const U = V();
    return () => U();
  }, []), J(s, { isDisabled: !e });
  const f = j(x, I, B, {
    "aria-label": "Toggle Flyout Menu"
  }), M = typeof o == "function" ? o(f, r, { isFocusVisible: m, isPressed: P }) : /* @__PURE__ */ t.createElement(
    "div",
    {
      ...f,
      ref: r,
      className: z([
        "tw-outline-none tw-rounded",
        v ? "tw-mx-3" : "tw-w-full",
        m && q
      ]),
      "data-test-id": "flyout-trigger"
    },
    o
  );
  return /* @__PURE__ */ t.createElement(t.Fragment, null, M, e && /* @__PURE__ */ t.createElement(W, null, /* @__PURE__ */ t.createElement(N, { restoreFocus: !0, contain: !0 }, /* @__PURE__ */ t.createElement(
    Q,
    {
      title: T,
      badges: b,
      decorator: p,
      isOpen: e,
      onClose: O,
      overlayTriggerProps: S,
      positionProps: _,
      fixedHeader: y,
      fixedFooter: F ? /* @__PURE__ */ t.createElement(X, { onConfirm: c, onCancel: u }) : E,
      ref: s,
      scrollRef: l,
      fitContent: R,
      contentMinHeight: h
    },
    g
  ))));
};
$.displayName = "FondueFlyout";
export {
  uo as FLYOUT_DIVIDER_COLOR,
  go as FLYOUT_DIVIDER_HEIGHT,
  $ as Flyout,
  Z as FlyoutPlacement
};
