// @frontify/fondue/dist/components/Flyout/FlyoutFooter.es.js
import { merge as i } from "../../utilities/merge.es.js";
import o from "react";
import u from "../../foundation/Icon/Generated/IconCheckMark.es.js";
import { Button as s } from "../Button/Button.es.js";
import { ButtonSize as c, ButtonStyle as r, ButtonEmphasis as l } from "../Button/ButtonTypes.es.js";
const n = ({ buttons: e, border: t = !0 }) => /* @__PURE__ */ o.createElement(
  "div",
  {
    className: i([
      "tw-flex tw-gap-x-3 tw-rounded-b tw-justify-end tw-py-5 tw-px-8 tw-bg-base",
      t && "tw-border-t tw-border-line",
      // The footer min-height should be 76px, since we Tailwind doesn't have a tw-h-19 class
      // we force TW to create it
      // The 19 comes from, 10 from the tw-py-5 + 9 from button height tw-h-9
      "tw-min-h-[4.75rem]"
    ])
  },
  e.map((a, m) => /* @__PURE__ */ o.createElement(s, { key: `flyout-footer-button-${m}`, ...a, size: c.Medium }))
);
n.displayName = "FondueFlyoutFooter";
const p = ({
  onConfirm: e,
  onCancel: t
}) => /* @__PURE__ */ o.createElement(
  n,
  {
    buttons: e ? [
      {
        style: r.Default,
        emphasis: l.Default,
        children: "Cancel",
        onClick: t
      },
      {
        style: r.Default,
        emphasis: l.Strong,
        children: "Confirm",
        onClick: e,
        icon: /* @__PURE__ */ o.createElement(u, null)
      }
    ] : [
      {
        style: r.Default,
        emphasis: l.Default,
        children: "Close",
        onClick: t
      }
    ]
  }
);
p.displayName = "FondueLegacyFlyoutFooter";
export {
  n as FlyoutFooter,
  p as LegacyFlyoutFooter
};
