// @frontify/fondue/dist/components/Dropdown/SelectMenu/SelectMenu.es.js
import { getMenuItems as f, getKeyItemRecord as p } from "../../ActionMenu/Aria/helper.es.js";
import { useListBox as y } from "@react-aria/listbox";
import { merge as w } from "../../../utilities/merge.es.js";
import r, { useRef as d } from "react";
import { SelectMenuItem as b } from "./SelectMenuItem.es.js";
import { SelectMenuSection as k } from "./SelectMenuSection.es.js";
const M = ({
  menuBlocks: m,
  ariaProps: n,
  state: e,
  ariaLabel: a = "Select Menu",
  scrollable: c = !1
}) => {
  const s = f(m), i = p(s), l = d(null), { listBoxProps: u } = y(
    {
      ...n,
      "aria-label": a
    },
    e,
    l
  );
  return /* @__PURE__ */ r.createElement(
    "ul",
    {
      ...u,
      ref: l,
      className: w([
        "tw-list-none tw-p-0 tw-m-0 tw-bg-base tw-z-20 focus-visible:tw-outline-none",
        c && "tw-overflow-y-auto"
      ])
    },
    [...e.collection].map((t) => /* @__PURE__ */ r.createElement(k, { key: t.key, ariaLabel: t["aria-label"] }, [...t.childNodes].map((o) => /* @__PURE__ */ r.createElement(
      b,
      {
        key: o.key,
        state: e,
        node: o,
        menuItem: i[o.key]
      }
    ))))
  );
};
M.displayName = "FondueSelectMenu";
export {
  M as SelectMenu
};
