// @frontify/fondue/dist/components/Dropdown/SelectMenu/SelectMenuItem.es.js
import { useFocusRing as w } from "@react-aria/focus";
import { useOption as S } from "@react-aria/listbox";
import { mergeProps as b } from "@react-aria/utils";
import { FOCUS_STYLE_INSET as F } from "../../../utilities/focusStyle.es.js";
import { merge as I } from "../../../utilities/merge.es.js";
import o, { useRef as g } from "react";
import { MenuItem as y } from "../../MenuItem/MenuItem.es.js";
const E = ({ state: r, menuItem: s, node: n }) => {
  const e = g(null), { optionProps: i, isSelected: m } = S(
    {
      key: n.key,
      shouldFocusOnHover: !1
    },
    r,
    e
  ), { title: l, decorator: c, subtitle: u, size: p, style: a, disabled: t, selectionIndicator: f } = s, { isFocusVisible: d, focusProps: v } = w();
  return /* @__PURE__ */ o.createElement(
    "li",
    {
      ...b(i, v),
      "data-test-id": "menu-item",
      className: I([
        "tw-relative hover:tw-bg-box-neutral-hover tw-list-none tw-outline-none",
        t && "tw-pointer-events-none tw-top-px",
        d && F
      ]),
      ref: e
    },
    /* @__PURE__ */ o.createElement(
      y,
      {
        title: l,
        decorator: c,
        subtitle: u,
        size: p,
        style: a,
        disabled: t,
        active: m,
        selectionIndicator: f
      }
    )
  );
};
E.displayName = "FondueSelectMenuItem";
export {
  E as SelectMenuItem
};
