// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/plugins/shared/LinkSelector/DocumentLink.es.js
import { jsxs as i, Fragment as l, jsx as e } from "react/jsx-runtime";
import { useState as x, useEffect as b } from "react";
import { PageLinks as m } from "./PageLinks.es.js";
import { merge as a, IconColorFan16 as h } from "@frontify/fondue";
const u = ({
  document: t,
  appBridge: c,
  selectedUrl: w,
  onSelectUrl: n,
  itemsToExpandInitially: r
}) => {
  const [o, s] = x(t.id === r.documentId), d = t.permanentLink === w;
  return b(() => {
    t.id === r.documentId && s(!0);
  }, [r, t.id]), /* @__PURE__ */ i(l, { children: [
    /* @__PURE__ */ i(
      "button",
      {
        "data-test-id": "internal-link-selector-document-link",
        className: a([
          "tw-flex tw-flex-1 tw-space-x-2 tw-items-center tw-py-2 tw-px-2.5 tw-leading-5 tw-cursor-pointer tw-w-full",
          d ? "tw-bg-box-selected-strong tw-text-box-selected-strong-inverse hover:tw-bg-box-selected-strong-hover:hover hover:tw-text-box-selected-strong-inverse-hover:hover" : "hover:tw-bg-box-neutral-hover hover:tw-text-box-neutral-inverse-hover"
        ]),
        onClick: () => n(t.permanentLink),
        onFocus: () => n(t.permanentLink),
        children: [
          /* @__PURE__ */ e(
            "div",
            {
              role: "button",
              tabIndex: 0,
              "data-test-id": "tree-item-toggle",
              className: "tw-flex tw-items-center tw-justify-center tw-p-1.5 tw-cursor-pointer",
              onClick: () => s(!o),
              onFocus: () => s(!o),
              children: /* @__PURE__ */ e(
                "div",
                {
                  className: a([
                    "tw-transition-transform tw-w-0 tw-h-0 tw-font-normal tw-border-t-4 tw-border-t-transparent tw-border-b-4 tw-border-b-transparent tw-border-l-4 tw-border-l-x-strong",
                    o ? "tw-rotate-90" : ""
                  ])
                }
              )
            }
          ),
          /* @__PURE__ */ e(h, {}),
          /* @__PURE__ */ e("span", { className: "tw-text-s", children: t.title }),
          /* @__PURE__ */ e("span", { className: "tw-flex-auto tw-font-sans tw-text-xs tw-text-right", children: "Document" })
        ]
      }
    ),
    o && /* @__PURE__ */ e(
      m,
      {
        appBridge: c,
        documentId: t.id,
        selectedUrl: w,
        onSelectUrl: n,
        itemsToExpandInitially: r
      }
    )
  ] });
};
export {
  u as DocumentLink
};
