// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/plugins/shared/LinkSelector/DocumentLinks.es.js
import { jsx as i, Fragment as v } from "react/jsx-runtime";
import { LoadingCircle as h } from "@frontify/fondue";
import { useState as r, useEffect as d } from "react";
import { DocumentLink as x } from "./DocumentLink.es.js";
const C = ({ appBridge: e, selectedUrl: n, onSelectUrl: m }) => {
  const [u, l] = r(!0), [f, I] = r([]), [g, p] = r({
    documentId: void 0,
    pageId: void 0
  }), o = [...f.values()];
  d(() => {
    n && o.length > 0 && y().then((t) => {
      p(t);
    });
  }, [o.length]), d(() => {
    e.getAllDocuments().then((t) => {
      I(t);
    }).finally(() => {
      l(!1);
    });
  }, []);
  const y = async () => {
    const t = {
      documentId: void 0,
      pageId: void 0
    };
    if (o.find((s) => s.permanentLink === n))
      return t;
    for (const s of o) {
      const D = await e.getDocumentPagesByDocumentId(s.id);
      e.getAllDocuments();
      const a = [...D.values()];
      if (!!a.find((c) => c.permanentLink === n))
        return t.documentId = s.id, t;
      for (const c of a)
        if (!![...(await e.getDocumentSectionsByDocumentPageId(c.id)).values()].find((L) => L.permanentLink === n))
          return t.documentId = s.id, t.pageId = c.id, t;
    }
    return t;
  };
  return u ? /* @__PURE__ */ i("div", { className: "tw-flex tw-justify-center tw-p-4", children: /* @__PURE__ */ i(h, {}) }) : /* @__PURE__ */ i(v, { children: o.map((t) => /* @__PURE__ */ i(
    x,
    {
      document: t,
      appBridge: e,
      selectedUrl: n,
      onSelectUrl: m,
      itemsToExpandInitially: g
    },
    t.id
  )) });
};
export {
  C as DocumentLinks
};
