// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/plugins/shared/LinkSelector/LinkSelector.es.js
import { jsxs as d, jsx as t } from "react/jsx-runtime";
import { Button as y, IconLink as D, ButtonSize as S, ButtonType as B, ButtonStyle as l, ButtonEmphasis as r, Modal as i } from "@frontify/fondue";
import { useOverlayTriggerState as L } from "@react-stately/overlays";
import { useState as C, useEffect as T } from "react";
import { DocumentLinks as x } from "./DocumentLinks.es.js";
const I = ({ appBridge: p, url: o, onUrlChange: m }) => {
  const { open: u, isOpen: f, close: s } = L({}), [n, c] = C(o), k = (e) => {
    c(e);
  }, h = (e) => {
    e.key === "Enter" && a();
  };
  T(() => {
    o && !n && c(o);
  }, [o, n]);
  const a = () => {
    m(n), s();
  };
  return /* @__PURE__ */ d("div", { "data-test-id": "internal-link-selector", onKeyDown: h, children: [
    /* @__PURE__ */ t(
      y,
      {
        icon: /* @__PURE__ */ t(D, {}),
        size: S.Medium,
        type: B.Button,
        style: l.Default,
        emphasis: r.Default,
        onClick: () => u(),
        children: "Internal link"
      }
    ),
    /* @__PURE__ */ d(i, { zIndex: 1001, onClose: () => s(), isOpen: f, isDismissable: !0, children: [
      /* @__PURE__ */ t(i.Header, { title: "Select internal link" }),
      /* @__PURE__ */ t(i.Body, { children: /* @__PURE__ */ t(x, { appBridge: p, selectedUrl: n, onSelectUrl: k }) }),
      /* @__PURE__ */ t(
        i.Footer,
        {
          buttons: [
            {
              children: "Cancel",
              onClick: () => s(),
              style: l.Default,
              emphasis: r.Default
            },
            {
              children: "Choose",
              onClick: (e) => {
                e == null || e.preventDefault(), a();
              },
              style: l.Default,
              emphasis: r.Strong,
              disabled: !n
            }
          ]
        }
      )
    ] })
  ] });
};
export {
  I as LinkSelector
};
