// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/plugins/shared/LinkSelector/PageLinks.es.js
import { jsx as t, Fragment as u } from "react/jsx-runtime";
import { LoadingCircle as p } from "@frontify/fondue";
import { useState as o, useEffect as w } from "react";
import { PageLink as h } from "./PageLink.es.js";
const k = ({
  appBridge: s,
  documentId: r,
  selectedUrl: i,
  onSelectUrl: c,
  itemsToExpandInitially: m
}) => {
  const [l, g] = o([]), [n, d] = o(!0), a = [...l.values()], f = !n && a.length > 0;
  return w(() => {
    s.getDocumentPagesByDocumentId(r).then((e) => {
      g(e);
    }).finally(() => {
      d(!1);
    });
  }, []), n ? /* @__PURE__ */ t("div", { className: "tw-flex tw-justify-center tw-p-4", children: /* @__PURE__ */ t(p, {}) }) : f ? /* @__PURE__ */ t(u, { children: a.map((e) => /* @__PURE__ */ t(
    h,
    {
      page: e,
      appBridge: s,
      selectedUrl: i,
      onSelectUrl: c,
      itemsToExpandInitially: m
    },
    e.id
  )) }) : /* @__PURE__ */ t("div", { className: "tw-py-2 tw-px-2.5 tw-pl-7 tw-leading-5 tw-text-s tw-text-text-weak", children: "This document does not contain any pages." });
};
export {
  k as PageLinks
};
