// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/plugins/shared/LinkSelector/PageLink.es.js
import { jsxs as d, Fragment as b, jsx as e } from "react/jsx-runtime";
import { useDocumentSection as u } from "@frontify/app-bridge";
import { merge as l } from "@frontify/fondue";
import { useState as v, useEffect as f } from "react";
import { SectionLink as p } from "./SectionLink.es.js";
const F = ({ page: t, selectedUrl: c, onSelectUrl: n, itemsToExpandInitially: o, appBridge: m }) => {
  const [r, s] = v(t.id === o.documentId), x = t.permanentLink === c, { documentSections: h } = u(m, t.id), i = [...h.values()], w = i.length > 0;
  return f(() => {
    t.id === o.pageId && s(!0);
  }, [o, t.id]), /* @__PURE__ */ d(b, { children: [
    /* @__PURE__ */ e(
      "div",
      {
        tabIndex: 0,
        "data-test-id": "internal-link-selector-page-link",
        className: l([
          "tw-py-2 tw-px-2.5 tw-leading-5 tw-cursor-pointer",
          w ? "tw-pl-7" : "tw-pl-12",
          x ? "tw-bg-box-selected-strong tw-text-box-selected-strong-inverse hover:tw-bg-box-selected-strong-hover:hover hover:tw-text-box-selected-strong-inverse-hover:hover" : "hover:tw-bg-box-neutral-hover hover:tw-text-box-neutral-inverse-hover"
        ]),
        onClick: () => n(t.permanentLink),
        onFocus: () => n(t.permanentLink),
        children: /* @__PURE__ */ d("div", { className: "tw-flex tw-flex-1 tw-space-x-1 tw-items-center tw-h-6", children: [
          w && /* @__PURE__ */ e(
            "button",
            {
              "data-test-id": "tree-item-toggle",
              className: "tw-flex tw-items-center tw-justify-center tw-p-1.5 tw-cursor-pointer",
              onClick: () => s(!r),
              onFocus: () => s(!r),
              children: /* @__PURE__ */ e(
                "div",
                {
                  className: l([
                    "tw-transition-transform tw-w-0 tw-h-0 tw-font-normal tw-border-t-4 tw-border-t-transparent tw-border-b-4 tw-border-b-transparent tw-border-l-4 tw-border-l-x-strong",
                    r ? "tw-rotate-90" : ""
                  ])
                }
              )
            }
          ),
          /* @__PURE__ */ e("span", { className: "tw-text-s", children: t.title }),
          /* @__PURE__ */ e("span", { className: "tw-flex-auto tw-font-sans tw-text-xs tw-text-right", children: "Page" })
        ] }, t.id)
      }
    ),
    r && i.length > 0 && i.map((a) => /* @__PURE__ */ e(
      p,
      {
        section: a,
        selectedUrl: c,
        onSelectUrl: n
      },
      a.id
    ))
  ] });
};
export {
  F as PageLink
};
