// @frontify/guideline-blocks-settings/dist/index.es.js
import "./styles.css.es.js";
export * from "@frontify/sidebar-settings";
import { BlockInjectButton as i } from "./components/BlockInjectButton/BlockInjectButton.es.js";
import { BlockItemWrapper as l } from "./components/BlockItemWrapper/BlockItemWrapper.es.js";
import { DEFAULT_DRAGGING_TOOLTIP as u, DEFAULT_DRAG_TOOLTIP as f } from "./components/BlockItemWrapper/constants.es.js";
import { Attachments as s } from "./components/Attachments/Attachments.es.js";
import { DownloadButton as P } from "./components/DownloadButton/DownloadButton.es.js";
import { RichTextEditor as c } from "./components/RichTextEditor/RichTextEditor.es.js";
import { LinkPlugin as T, createLinkPlugin as E } from "./components/RichTextEditor/plugins/LinkPlugin/index.es.js";
import { BUTTON_PLUGIN as A, ButtonPlugin as R, ELEMENT_BUTTON as C, createButtonPlugin as I } from "./components/RichTextEditor/plugins/ButtonPlugin/createButtonPlugin.es.js";
import { withButton as k } from "./components/RichTextEditor/plugins/ButtonPlugin/withButton.es.js";
import { insertButton as F } from "./components/RichTextEditor/plugins/ButtonPlugin/transforms/insertButton.es.js";
import { submitFloatingButton as L } from "./components/RichTextEditor/plugins/ButtonPlugin/transforms/submitFloatingButton.es.js";
import { unwrapButton as H } from "./components/RichTextEditor/plugins/ButtonPlugin/transforms/unwrapButton.es.js";
import { upsertButton as _ } from "./components/RichTextEditor/plugins/ButtonPlugin/transforms/upsertButton.es.js";
import { upsertButtonText as w } from "./components/RichTextEditor/plugins/ButtonPlugin/transforms/upsertButtonText.es.js";
import { wrapButton as W } from "./components/RichTextEditor/plugins/ButtonPlugin/transforms/wrapButton.es.js";
import { getUrlFromEditor as v } from "./components/RichTextEditor/plugins/ButtonPlugin/utils/getUrl.es.js";
import { createButtonNode as V } from "./components/RichTextEditor/plugins/ButtonPlugin/utils/createButtonNode.es.js";
import { triggerFloatingButton as q } from "./components/RichTextEditor/plugins/ButtonPlugin/utils/triggerFloatingButton.es.js";
import { triggerFloatingButtonEdit as J } from "./components/RichTextEditor/plugins/ButtonPlugin/utils/triggerFloatingButtonEdit.es.js";
import { triggerFloatingButtonInsert as Y } from "./components/RichTextEditor/plugins/ButtonPlugin/utils/triggerFloatingButtonInsert.es.js";
import { BlockButtonStyles as $ } from "./components/RichTextEditor/plugins/ButtonPlugin/utils/styles.es.js";
import { Custom1Plugin as rt } from "./components/RichTextEditor/plugins/TextStylePlugins/custom1Plugin.es.js";
import { Custom2Plugin as ot } from "./components/RichTextEditor/plugins/TextStylePlugins/custom2Plugin.es.js";
import { Custom3Plugin as pt } from "./components/RichTextEditor/plugins/TextStylePlugins/custom3Plugin.es.js";
import { Heading1Plugin as gt } from "./components/RichTextEditor/plugins/TextStylePlugins/heading1Plugin.es.js";
import { Heading2Plugin as mt } from "./components/RichTextEditor/plugins/TextStylePlugins/heading2Plugin.es.js";
import { Heading3Plugin as xt } from "./components/RichTextEditor/plugins/TextStylePlugins/heading3Plugin.es.js";
import { Heading4Plugin as ft } from "./components/RichTextEditor/plugins/TextStylePlugins/heading4Plugin.es.js";
import { ImageCaptionPlugin as st } from "./components/RichTextEditor/plugins/TextStylePlugins/imageCaptionPlugin.es.js";
import { ImageTitlePlugin as Pt } from "./components/RichTextEditor/plugins/TextStylePlugins/imageTitlePlugin.es.js";
import { PARAGRAPH_CLASSES as ct, ParagraphMarkupElement as yt, ParagraphMarkupElementNode as Tt, ParagraphPlugin as Et, createParagraphPlugin as ht } from "./components/RichTextEditor/plugins/TextStylePlugins/paragraphPlugin.es.js";
import { QuoteMarkupElementNode as Rt, QuotePlugin as Ct, createQuotePlugin as It } from "./components/RichTextEditor/plugins/TextStylePlugins/quotePlugin.es.js";
import { AllTextStylePlugins as kt, AllTextStyles as bt, TextStylePluginsWithoutImage as Ft, TextStylesWithoutImage as Gt } from "./components/RichTextEditor/plugins/TextStylePlugins/helpers.es.js";
import { BlockStyles as Dt, TextStyles as Ht } from "./components/RichTextEditor/plugins/styles.es.js";
import { getDefaultPluginsWithLinkChooser as _t } from "./components/RichTextEditor/pluginPresets/defaultPluginsWithLinkChooser.es.js";
import { THEME_PREFIX as wt } from "./components/RichTextEditor/constants.es.js";
import { convertToRteValue as Wt } from "./helpers/convertToRichTextValue.es.js";
import { customCoordinatesGetterFactory as vt } from "./helpers/customCoordinatesGetterFactory.es.js";
import { hasRichTextValue as Vt } from "./helpers/hasRichTextValue.es.js";
import { isDownloadable as qt } from "./helpers/isDownloadable.es.js";
import { mapAppBridgeColorPaletteToFonduePalette as Jt, mapAppBridgeColorPalettesToFonduePalettes as Kt } from "./helpers/mapColorPalettes.es.js";
import { addHttps as Zt } from "./helpers/addHttps.es.js";
import { useAttachments as tr } from "./hooks/useAttachments.es.js";
import { useDndSensors as er } from "./hooks/useDndSensors.es.js";
import { getBackgroundSettings as nr } from "./settings/background.es.js";
import { getBorderSettings as ar } from "./settings/border.es.js";
import { getBorderRadiusSettings as ir, getBorderRadiusSlider as mr } from "./settings/borderRadius.es.js";
import { getExtendedBorderRadiusSettings as xr } from "./settings/borderRadiusExtended.es.js";
import { getGutterSettings as fr } from "./settings/gutter.es.js";
import { getMarginSettings as sr, getMarginSlider as Sr } from "./settings/margin.es.js";
import { getMarginExtendedSettings as Br } from "./settings/marginExtended.es.js";
import { getPaddingSettings as yr, getPaddingSlider as Tr } from "./settings/padding.es.js";
import { getPaddingExtendedSettings as hr } from "./settings/paddingExtended.es.js";
import { BorderStyle as Rr, GutterSpacing as Cr, Margin as Ir, Padding as Mr, Radius as kr, Security as br, borderStyleMap as Fr, gutterSpacingStyleMap as Gr, marginStyleMap as Lr, paddingStyleMap as Dr, radiusStyleMap as Hr } from "./settings/types.es.js";
import { getSecurityDownloadableSetting as _r, getSecurityGlobalControlId as Or } from "./settings/securityDownloadable.es.js";
import { getSecurityGlobalControlSetting as Ur } from "./settings/securityGlobalControl.es.js";
import { isDark as jr } from "./utilities/color/isDark.es.js";
import { toHex8String as Qr } from "./utilities/color/toHex8String.es.js";
import { toHexString as Xr } from "./utilities/color/toHexString.es.js";
import { toRgbaString as zr } from "./utilities/color/toRgbaString.es.js";
import { setAlpha as Kr } from "./utilities/color/setAlpha.es.js";
import { toColorObject as Zr } from "./utilities/color/toColorObject.es.js";
import { getReadableColor as te } from "./utilities/color/getReadableColor.es.js";
import { toShortRgba as ee } from "./utilities/color/toShortRgba.es.js";
import { moveItemInArray as ne } from "./utilities/moveItemInArray.es.js";
import { getBackgroundColorStyles as ae } from "./utilities/react/getBackgroundColorStyles.es.js";
import { getBorderStyles as ie } from "./utilities/react/getBorderStyles.es.js";
import { getRadiusStyles as le } from "./utilities/react/getRadiusStyles.es.js";
import { joinClassNames as ue } from "./utilities/react/joinClassNames.es.js";
var r = /* @__PURE__ */ ((t) => (t.Main = "main", t.Basics = "basics", t.Layout = "layout", t.Style = "style", t.Security = "security", t.Targets = "targets", t))(r || {});
const o = (t) => t, n = (t) => t;
export {
  kt as AllTextStylePlugins,
  bt as AllTextStyles,
  s as Attachments,
  A as BUTTON_PLUGIN,
  $ as BlockButtonStyles,
  i as BlockInjectButton,
  l as BlockItemWrapper,
  Dt as BlockStyles,
  Rr as BorderStyle,
  R as ButtonPlugin,
  rt as Custom1Plugin,
  ot as Custom2Plugin,
  pt as Custom3Plugin,
  u as DEFAULT_DRAGGING_TOOLTIP,
  f as DEFAULT_DRAG_TOOLTIP,
  P as DownloadButton,
  C as ELEMENT_BUTTON,
  Cr as GutterSpacing,
  gt as Heading1Plugin,
  mt as Heading2Plugin,
  xt as Heading3Plugin,
  ft as Heading4Plugin,
  st as ImageCaptionPlugin,
  Pt as ImageTitlePlugin,
  T as LinkPlugin,
  Ir as Margin,
  ct as PARAGRAPH_CLASSES,
  Mr as Padding,
  yt as ParagraphMarkupElement,
  Tt as ParagraphMarkupElementNode,
  Et as ParagraphPlugin,
  Rt as QuoteMarkupElementNode,
  Ct as QuotePlugin,
  kr as Radius,
  c as RichTextEditor,
  r as Sections,
  br as Security,
  wt as THEME_PREFIX,
  Ft as TextStylePluginsWithoutImage,
  Ht as TextStyles,
  Gt as TextStylesWithoutImage,
  Zt as addHttps,
  Fr as borderStyleMap,
  Wt as convertToRteValue,
  V as createButtonNode,
  I as createButtonPlugin,
  E as createLinkPlugin,
  ht as createParagraphPlugin,
  It as createQuotePlugin,
  vt as customCoordinatesGetterFactory,
  o as defineBlock,
  n as defineSettings,
  ae as getBackgroundColorStyles,
  nr as getBackgroundSettings,
  ir as getBorderRadiusSettings,
  mr as getBorderRadiusSlider,
  ar as getBorderSettings,
  ie as getBorderStyles,
  _t as getDefaultPluginsWithLinkChooser,
  xr as getExtendedBorderRadiusSettings,
  fr as getGutterSettings,
  Br as getMarginExtendedSettings,
  sr as getMarginSettings,
  Sr as getMarginSlider,
  hr as getPaddingExtendedSettings,
  yr as getPaddingSettings,
  Tr as getPaddingSlider,
  le as getRadiusStyles,
  te as getReadableColor,
  _r as getSecurityDownloadableSetting,
  Or as getSecurityGlobalControlId,
  Ur as getSecurityGlobalControlSetting,
  v as getUrlFromEditor,
  Gr as gutterSpacingStyleMap,
  Vt as hasRichTextValue,
  F as insertButton,
  jr as isDark,
  qt as isDownloadable,
  ue as joinClassNames,
  Jt as mapAppBridgeColorPaletteToFonduePalette,
  Kt as mapAppBridgeColorPalettesToFonduePalettes,
  Lr as marginStyleMap,
  ne as moveItemInArray,
  Dr as paddingStyleMap,
  Hr as radiusStyleMap,
  Kr as setAlpha,
  L as submitFloatingButton,
  Zr as toColorObject,
  Qr as toHex8String,
  Xr as toHexString,
  zr as toRgbaString,
  ee as toShortRgba,
  q as triggerFloatingButton,
  J as triggerFloatingButtonEdit,
  Y as triggerFloatingButtonInsert,
  H as unwrapButton,
  _ as upsertButton,
  w as upsertButtonText,
  tr as useAttachments,
  er as useDndSensors,
  k as withButton,
  W as wrapButton
};
