// @frontify/fondue/dist/components/AssetInput/AssetThumbnail.es.js
import { IconSize as n } from "../../foundation/Icon/IconSize.es.js";
import { merge as c } from "../../utilities/merge.es.js";
import e, { cloneElement as m } from "react";
import { AssetInputSize as s } from "./AssetInput.es.js";
import o from "../../foundation/Icon/Generated/IconMusicNote.es.js";
const u = (t, a) => {
  switch (!0) {
    case t === s.Large:
      return "tw-w-full tw-h-32";
    case a:
      return "tw-w-11 tw-h-11";
    default:
      return "tw-w-14 tw-h-full";
  }
}, w = ({
  asset: t,
  size: a,
  isActive: l = !1,
  isMultiAsset: r = !1
}) => {
  const i = t.type === "icon" || t.type === "audio" ? t.type : "default";
  return /* @__PURE__ */ e.createElement(
    "div",
    {
      className: c([
        "tw-flex tw-flex-none tw-items-center tw-justify-center tw-bg-black-5 dark:tw-bg-black-95",
        u(a, r),
        l ? "tw-text-black-100 dark:tw-text-white" : "tw-text-black-80 dark:tw-text-black-20"
      ]),
      "data-test-id": "asset-input-thumbnail"
    },
    i === "icon" && /* @__PURE__ */ e.createElement("span", { "data-test-id": "asset-input-thumbnail-icon" }, t.icon && m(t.icon, { size: n.Size24 })),
    i === "audio" && /* @__PURE__ */ e.createElement(o, { size: n.Size24, "data-test-id": "asset-input-thumbnail-audio" }),
    i === "default" && /* @__PURE__ */ e.createElement(
      "img",
      {
        src: t.src,
        alt: t.alt || "",
        className: "tw-max-h-full",
        "data-test-id": "asset-input-thumbnail-image"
      }
    )
  );
};
w.displayName = "FondueAssetThumbnail";
export {
  w as AssetThumbnail
};
