// @frontify/fondue/dist/components/AssetInput/AssetInput.es.js
import { useMemoizedId as S } from "../../hooks/useMemoizedId.es.js";
import { merge as s } from "../../utilities/merge.es.js";
import e, { useRef as B } from "react";
import { SelectedAsset as F } from "./SingleAsset/SelectedAsset.es.js";
import { MultiAssetPreview as b } from "./MultiAssetPreview/MultiAssetPreview.es.js";
import y from "../../foundation/Icon/Generated/IconArrowCircleUp.es.js";
import A from "../../foundation/Icon/Generated/IconImageStack.es.js";
import { Button as d } from "../Button/Button.es.js";
import { ButtonStyle as p, ButtonEmphasis as f } from "../Button/ButtonTypes.es.js";
var D = /* @__PURE__ */ ((t) => (t.Small = "Small", t.Large = "Large", t))(D || {});
const R = ({
  assets: t = [],
  numberOfLocations: u = 1,
  actions: n = [],
  size: w = "Small",
  isLoading: o = !1,
  hideSize: h = !1,
  hideExtension: g = !1,
  onLibraryClick: l,
  onUploadClick: r,
  onMultiAssetClick: a,
  acceptFileType: E
}) => {
  const i = t.length, m = B(null), v = S(), x = (N) => {
    const c = N.target.files;
    c && r && r(c);
  }, I = () => {
    m.current && m.current.click();
  };
  return (o || i === 1) && n ? /* @__PURE__ */ e.createElement(
    F,
    {
      asset: t[0],
      size: w,
      actions: n,
      isLoading: o,
      hideExtension: g,
      hideSize: h
    }
  ) : i > 1 && a ? /* @__PURE__ */ e.createElement(b, { assets: t, onClick: a, numberOfLocations: u }) : /* @__PURE__ */ e.createElement(
    "div",
    {
      "data-test-id": "asset-input-placeholder",
      className: s([
        "tw-grid tw-p-3 tw-border tw-border-dashed tw-border-black-10 tw-rounded",
        r && l ? "tw-grid-cols-2 tw-divide-x tw-divide-black-10" : "tw-grid-cols-1"
      ])
    },
    r && /* @__PURE__ */ e.createElement(
      "div",
      {
        className: s(["tw-flex tw-flex-col tw-h-8", l && "tw-pr-3"]),
        "data-test-id": "asset-input-upload"
      },
      /* @__PURE__ */ e.createElement(
        d,
        {
          style: p.Default,
          onClick: I,
          emphasis: f.Weak,
          icon: /* @__PURE__ */ e.createElement(y, null)
        },
        "Upload"
      ),
      /* @__PURE__ */ e.createElement(
        "input",
        {
          id: v,
          ref: m,
          className: "tw-hidden",
          type: "file",
          accept: E,
          multiple: !!a,
          onChange: x
        }
      )
    ),
    l && /* @__PURE__ */ e.createElement(
      "div",
      {
        className: s(["tw-flex tw-flex-col tw-h-8", r && "tw-pl-3"]),
        "data-test-id": "asset-input-library"
      },
      /* @__PURE__ */ e.createElement(
        d,
        {
          style: p.Default,
          onClick: l,
          emphasis: f.Weak,
          icon: /* @__PURE__ */ e.createElement(A, null)
        },
        "Browse"
      )
    )
  );
};
R.displayName = "FondueAssetInput";
export {
  R as AssetInput,
  D as AssetInputSize
};
