// @frontify/fondue/dist/components/RichTextEditor/serializer/markdown/options.es.js
import { getPluginType as L, ELEMENT_PARAGRAPH as T, ELEMENT_LINK as n, MARK_CODE as a, MARK_BOLD as _, MARK_ITALIC as N, MARK_STRIKETHROUGH as l, ELEMENT_IMAGE as s, ELEMENT_CODE_BLOCK as g, ELEMENT_HR as m, ELEMENT_UL as I, ELEMENT_OL as e, ELEMENT_LI as h, ELEMENT_LIC as k, ELEMENT_H5 as p, ELEMENT_H6 as A, ELEMENT_MENTION as R } from "@udecode/plate";
import { TextStyles as M } from "../../Plugins/TextStylePlugin/types.es.js";
const O = (E) => ({
  nodeTypes: {
    paragraph: L(E, T),
    link: L(E, n),
    blockQuote: L(E, M.quote),
    inlineCodeMark: L(E, a),
    strongMark: L(E, _),
    emphasisMark: L(E, N),
    deleteMark: L(E, l),
    image: L(E, s),
    codeBlock: L(E, g),
    thematicBreak: L(E, m),
    ulList: L(E, I),
    olList: L(E, e),
    listItem: L(E, h),
    listItemChild: L(E, k),
    heading: {
      1: L(E, M.heading1),
      2: L(E, M.heading2),
      3: L(E, M.heading3),
      4: L(E, M.heading4),
      5: L(E, p),
      6: L(E, A)
    },
    mention: L(E, R)
  },
  linkDestinationKey: "url"
});
export {
  O as options
};
