// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/plugins/LinkPlugin/FloatingLink/InsertLinkModal/InsertModal.es.js
import { jsxs as r, jsx as e } from "react/jsx-runtime";
import { FormControl as n, TextInput as d, Checkbox as v, Button as o, ButtonSize as c, ButtonStyle as b, ButtonEmphasis as k, IconCheckMark20 as f } from "@frontify/fondue";
import { LinkSelector as C } from "../../../shared/LinkSelector/LinkSelector.es.js";
const g = ({
  state: l,
  onTextChange: a,
  onUrlChange: t,
  onToggleTab: u,
  onCancel: h,
  onSave: m,
  isValidUrlOrEmpty: i,
  hasValues: w,
  testId: s,
  appBridge: p,
  children: x
}) => /* @__PURE__ */ r("div", { "data-test-id": s, className: "tw-bg-white tw-rounded tw-shadow tw-p-7 tw-min-w-[400px] tw-overflow-y-auto", children: [
  /* @__PURE__ */ e(
    n,
    {
      label: {
        children: "Text",
        htmlFor: "linkText",
        required: !0
      },
      children: /* @__PURE__ */ e(d, { id: "linkText", value: l.text, placeholder: "Link Text", onChange: a })
    }
  ),
  x,
  /* @__PURE__ */ r("div", { className: "tw-pt-5", children: [
    /* @__PURE__ */ e(
      n,
      {
        label: {
          children: "URL",
          htmlFor: "url",
          required: !0
        },
        children: /* @__PURE__ */ e(
          d,
          {
            id: "url",
            value: l.url,
            placeholder: "https://example.com",
            focusOnMount: !0,
            onChange: t
          }
        )
      }
    ),
    !i(l == null ? void 0 : l.url) && /* @__PURE__ */ e("div", { className: "tw-text-red-65 tw-mt-3", children: "Please enter a valid URL." })
  ] }),
  /* @__PURE__ */ e("div", { className: "tw-mt-3", children: /* @__PURE__ */ e(C, { url: l.url, appBridge: p, onUrlChange: t }) }),
  /* @__PURE__ */ e("div", { className: "tw-mt-3", children: /* @__PURE__ */ e(v, { value: "new-tab", label: "Open in new tab", state: l.newTab, onChange: u }) }),
  /* @__PURE__ */ e("div", { className: "tw-mt-3", children: /* @__PURE__ */ r("div", { className: "tw-pt-5 tw-flex tw-gap-x-3 tw-justify-end tw-border-t tw-border-t-black-10", children: [
    /* @__PURE__ */ e(
      o,
      {
        onClick: h,
        size: c.Medium,
        style: b.Default,
        emphasis: k.Default,
        children: "Cancel"
      }
    ),
    /* @__PURE__ */ e(
      o,
      {
        onClick: m,
        size: c.Medium,
        icon: /* @__PURE__ */ e(f, {}),
        disabled: !w || !i(l == null ? void 0 : l.url),
        children: "Save"
      }
    )
  ] }) })
] });
export {
  g as InsertModal
};
